{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ce46012",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 700,
   "id": "309bb13d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !conda info --envs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f20e4a78",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pathlib import Path\n",
    "import re\n",
    "from collections import defaultdict\n",
    "import matplotlib.pyplot as plt\n",
    "import pickle"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b98b95f4",
   "metadata": {},
   "source": [
    "# Create files with scientific and common names for all the amniota\n",
    "Use those list as regex patterns for searching inside the biosample and bioproject information"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d6f48df",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###  Download the `taxdump.tar.gz` from the NCBI ftp and place this in your bash profile\n",
    "```sh\n",
    "export TAXONKIT_DB=/home/mibu/Bif_Data/taxdump\n",
    "```\n",
    "Then, we can create a list of taxids of all the specia under 'Amniota' with their ranks and scientific names:\n",
    "\n",
    "```sh\n",
    " taxonkit list --ids 32524 | taxonkit filter -E species | taxonkit lineage -L -n > amniota_sp.list\n",
    "```\n",
    "`-L`: Only include the names, not the lineage. `list` doesn't accept the input from `filter` \n",
    "\n",
    "There is no option to retrieve \"common names\", but we can get that information from  `names.dmp`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "002f4ee7",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## One pipe:\n",
    "```bash\n",
    "names=\"taxdump/names.dmp\"\n",
    "\n",
    "taxonkit list --ids 32524 | taxonkit filter -E species | taxonkit lineage -L -n | cut -f1 | \\\n",
    "rg -f - $names | rg 'common name' | cut -d '|' -f2 | tr -d '\\t' > ~/common_names_ncbi.list\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d3bc3357",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(True, True)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample = Path(\"/run/media/mibu/LR-orico_mini/Recovery_ADATA/Umayor/Work/Blautia/Notebooks/biosample_descr_akk.tsv\")\n",
    "common_pats = Path(\"/home/mibu/common_names_ncbi.list\")\n",
    "common_pats.exists(), sample.exists()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3327f93a",
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_desc_all = Path(\"/run/media/mibu/LR-orico_mini/Recovery_ADATA/Umayor/Work/Blautia/Notebooks/biosample_descr_non_akk.tsv\")\n",
    "assert sample_desc_all.exists() == True"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e95b3c9",
   "metadata": {},
   "source": [
    "**Get only the species names, withot the taxids**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "71bb221e",
   "metadata": {},
   "outputs": [],
   "source": [
    "! cut -f2 ~/amniota_sp.list > ~/amniota_only_sp.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "00c8a85a",
   "metadata": {},
   "outputs": [],
   "source": [
    "sp_pats = Path(\"/home/mibu/amniota_only_sp.list\")\n",
    "assert sp_pats.exists() == True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "16d328e1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 585 entries, 0 to 584\n",
      "Data columns (total 13 columns):\n",
      " #   Column                       Non-Null Count  Dtype  \n",
      "---  ------                       --------------  -----  \n",
      " 0   Genome                       585 non-null    object \n",
      " 1   Completeness                 585 non-null    float64\n",
      " 2   Contamination                585 non-null    float64\n",
      " 3   classification               585 non-null    object \n",
      " 4   classification_method        585 non-null    object \n",
      " 5   sum_len                      585 non-null    int64  \n",
      " 6   N50                          585 non-null    int64  \n",
      " 7   16S ribosomal RNA > 1400 nt  304 non-null    float64\n",
      " 8   Total_tRNAs                  585 non-null    int64  \n",
      " 9   host                         8 non-null      object \n",
      " 10  NCBI_organism_name           212 non-null    object \n",
      " 11  NCBI_infraspecific_name      195 non-null    object \n",
      " 12  relation_to_type_material    40 non-null     object \n",
      "dtypes: float64(3), int64(3), object(7)\n",
      "memory usage: 59.5+ KB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(Mouse                  3\n",
       " Choloepus hoffmanni    1\n",
       " Diceros bicornis       1\n",
       " Bufo bufo              1\n",
       " Horse                  1\n",
       " Gallus gallus          1\n",
       " Name: host, dtype: int64,\n",
       " 577)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = pd.read_csv('/run/media/mibu/LR-orico_mini/Recovery_ADATA/Umayor/Work/Akker/Verrucomicrobia/QC/verrocu_seqs_summary.tsv', sep='\\t')\n",
    "merged_df.info()\n",
    "merged_df['host'].value_counts(), merged_df['host'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6d9e97dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "qc_path = Path('/run/media/mibu/LR-orico_mini/Recovery_ADATA/Umayor/Work/Akker/Verrucomicrobia/QC/')\n",
    "# merged_df = pd.read_csv(qc_path.joinpath('verrocu_seqs_summary_2.tsv'), sep='\\t')\n",
    "# merged_df.info()\n",
    "# merged_df['host'].value_counts(), merged_df['host'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "id": "db0ef767",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 384 entries, 2 to 562\n",
      "Data columns (total 14 columns):\n",
      " #   Column                       Non-Null Count  Dtype  \n",
      "---  ------                       --------------  -----  \n",
      " 0   Genome                       384 non-null    object \n",
      " 1   Completeness                 384 non-null    float64\n",
      " 2   Contamination                384 non-null    float64\n",
      " 3   classification               384 non-null    object \n",
      " 4   classification_method        384 non-null    object \n",
      " 5   sum_len                      384 non-null    int64  \n",
      " 6   N50                          384 non-null    int64  \n",
      " 7   16S ribosomal RNA > 1400 nt  153 non-null    float64\n",
      " 8   Total_tRNAs                  384 non-null    int64  \n",
      " 9   host                         0 non-null      object \n",
      " 10  NCBI_organism_name           65 non-null     object \n",
      " 11  NCBI_infraspecific_name      55 non-null     object \n",
      " 12  relation_to_type_material    36 non-null     object \n",
      " 13  type_of_sample               0 non-null      object \n",
      "dtypes: float64(3), int64(3), object(8)\n",
      "memory usage: 45.0+ KB\n",
      "Series([], Name: host, dtype: int64)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>classification</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>sum_len</th>\n",
       "      <th>N50</th>\n",
       "      <th>16S ribosomal RNA &gt; 1400 nt</th>\n",
       "      <th>Total_tRNAs</th>\n",
       "      <th>host</th>\n",
       "      <th>NCBI_organism_name</th>\n",
       "      <th>NCBI_infraspecific_name</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>type_of_sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000155695.1_ASM15569v1_genomic</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1.35</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>5775745</td>\n",
       "      <td>5514226</td>\n",
       "      <td>1.0</td>\n",
       "      <td>45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000242935.3_ASM24293v3_genomic</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.23</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7417673</td>\n",
       "      <td>7317842</td>\n",
       "      <td>1.0</td>\n",
       "      <td>52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Genome  Completeness  Contamination  \\\n",
       "2  GCA_000155695.1_ASM15569v1_genomic         100.0           1.35   \n",
       "3  GCA_000242935.3_ASM24293v3_genomic         100.0           0.23   \n",
       "\n",
       "                                      classification  \\\n",
       "2  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "3  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "                               classification_method  sum_len      N50  \\\n",
       "2  taxonomic classification defined by topology a...  5775745  5514226   \n",
       "3  taxonomic classification defined by topology a...  7417673  7317842   \n",
       "\n",
       "   16S ribosomal RNA > 1400 nt  Total_tRNAs host NCBI_organism_name  \\\n",
       "2                          1.0           45  NaN                NaN   \n",
       "3                          1.0           52  NaN                NaN   \n",
       "\n",
       "  NCBI_infraspecific_name relation_to_type_material type_of_sample  \n",
       "2                     NaN                       NaN            NaN  \n",
       "3                     NaN                       NaN            NaN  "
      ]
     },
     "execution_count": 245,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "non_akk_ncbi = merged_df[~merged_df['classification'].str.contains('g__Akkermansia')]#.loc[:, 'Genome'].str.extract('(GC[AF]_[0-9]{9}\\.[1-9]{1,2})').dropna().values\n",
    "non_akk_ncbi.info()\n",
    "print(non_akk_ncbi['host'].value_counts())\n",
    "non_akk_ncbi.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90d6703b",
   "metadata": {},
   "source": [
    "### Read the refseq and genbank summary tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c34a4da0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1158218 entries, 0 to 1158217\n",
      "Data columns (total 23 columns):\n",
      " #   Column                     Non-Null Count    Dtype \n",
      "---  ------                     --------------    ----- \n",
      " 0   Genome                     1158163 non-null  object\n",
      " 1   assembly_accession         1158218 non-null  object\n",
      " 2   bioproject                 1158214 non-null  object\n",
      " 3   biosample                  1158213 non-null  object\n",
      " 4   wgs_master                 1104090 non-null  object\n",
      " 5   refseq_category            1158218 non-null  object\n",
      " 6   taxid                      1158218 non-null  int64 \n",
      " 7   species_taxid              1158218 non-null  int64 \n",
      " 8   NCBI_organism_name         1158218 non-null  object\n",
      " 9   NCBI_infraspecific_name    998033 non-null   object\n",
      " 10  isolate                    182630 non-null   object\n",
      " 11  version_status             1158218 non-null  object\n",
      " 12  assembly_level             1158218 non-null  object\n",
      " 13  release_type               1158218 non-null  object\n",
      " 14  genome_rep                 1158218 non-null  object\n",
      " 15  seq_rel_date               1158218 non-null  object\n",
      " 16  asm_name                   1158217 non-null  object\n",
      " 17  submitter                  1158210 non-null  object\n",
      " 18  gbrs_paired_asm            1158218 non-null  object\n",
      " 19  paired_asm_comp            1158218 non-null  object\n",
      " 20  ftp_path                   1158218 non-null  object\n",
      " 21  excluded_from_refseq       726985 non-null   object\n",
      " 22  relation_to_type_material  32556 non-null    object\n",
      "dtypes: int64(2), object(21)\n",
      "memory usage: 203.2+ MB\n"
     ]
    }
   ],
   "source": [
    "ncbi_df = pd.read_feather('/run/media/mibu/LR-orico_mini/Recovery_ADATA/Umayor/Data/ncbi_summ_tables.feather')\n",
    "ncbi_df.info()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6b4d3d50",
   "metadata": {},
   "source": [
    "### Create indexes for the akkermasia and non-akkermasia assemblies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "9a1206f6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([['GCA_000436395.1'],\n",
       "        ['GCA_000437075.1'],\n",
       "        ['GCA_001917295.1'],\n",
       "        ['GCA_002362435.1'],\n",
       "        ['GCA_002492685.1']], dtype=object),\n",
       " (180, 1))"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "akk_ncbi = merged_df[merged_df['classification'].str.contains('g__Akkermansia')].loc[:, 'Genome'].str.extract('(GC[AF]_[0-9]{9}\\.[1-9]{1,2})').dropna().values\n",
    "akk_ncbi[:5], akk_ncbi.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "9424db05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([['GCA_000155695.1'],\n",
       "        ['GCA_000242935.3'],\n",
       "        ['GCA_000243495.3'],\n",
       "        ['GCA_000526255.1'],\n",
       "        ['GCA_000739615.1']], dtype=object),\n",
       " (384, 1))"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "non_akk_ncbi = merged_df[~merged_df['classification'].str.contains('g__Akkermansia')].loc[:, 'Genome'].str.extract('(GC[AF]_[0-9]{9}\\.[1-9]{1,2})').dropna().values\n",
    "non_akk_ncbi[:5], non_akk_ncbi.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc64cf32",
   "metadata": {},
   "source": [
    "### For BioSample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "c81ae1a1",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "biosample_descr_akk.tsv\n",
      "GCA_000436395.1 SAMEA3138704\n",
      "GCA_000437075.1 SAMEA3138738\n",
      "GCA_001917295.1 SAMN05717313\n",
      "GCA_002362435.1 SAMN06457624\n",
      "GCA_002492685.1 SAMN06455865\n",
      "GCA_002493465.1 SAMN06455314\n",
      "GCA_003514565.1 SAMN08019340\n",
      "GCA_008672035.1 SAMN10316569\n",
      "GCA_008672205.1 SAMN10316561\n",
      "GCA_009773845.1 SAMN12406434\n",
      "GCA_014871965.1 SAMN12905381\n",
      "GCA_014871995.1 SAMN12905379\n",
      "GCA_014872005.1 SAMN12905380\n",
      "GCA_014872035.1 SAMN12905378\n",
      "GCA_014872065.1 SAMN12905377\n",
      "GCA_014872075.1 SAMN12905375\n",
      "GCA_014872105.1 SAMN12905376\n",
      "GCA_014872125.1 SAMN12905373\n",
      "GCA_014872165.1 SAMN12905372\n",
      "GCA_900539575.1 SAMEA4890833\n",
      "GCA_900539895.1 SAMEA4890871\n",
      "GCA_900545155.1 SAMEA4891402\n",
      "GCA_900752515.1 SAMEA5279134\n",
      "GCA_900759685.1 SAMEA5279138\n",
      "GCA_900765745.1 SAMEA5279136\n",
      "GCA_905193715.1 SAMEA7847250\n",
      "GCA_905197035.1 SAMEA7847110\n",
      "GCA_905200125.1 SAMEA7848347\n",
      "GCA_905200945.1 SAMEA7847584\n",
      "GCA_905205055.1 SAMEA7848078\n",
      "GCA_905205915.1 SAMEA7846587\n",
      "GCA_905210405.1 SAMEA7847399\n",
      "GCA_905210615.1 SAMEA7847488\n",
      "GCF_000020225.1 SAMN00138213\n",
      "GCF_000723745.2 SAMEA3139045\n",
      "GCF_001578645.1 SAMN03956217\n",
      "GCF_001580195.1 SAMN03956447\n",
      "GCF_001647615.1 SAMN04606485\n",
      "GCF_001688765.2 SAMN04621615\n",
      "GCF_002161325.1 SAMN06473757\n",
      "GCF_002201495.1 SAMN03854119\n",
      "GCF_002884915.1 SAMN08162557\n",
      "GCF_002884975.1 SAMN08162556\n",
      "GCF_002884995.1 SAMN08162554\n",
      "GCF_002885015.1 SAMN08162551\n",
      "GCF_002885025.1 SAMN08162552\n",
      "GCF_002885055.1 SAMN08162550\n",
      "GCF_002885075.1 SAMN08162548\n",
      "GCF_002885095.1 SAMN08162547\n",
      "GCF_002885155.1 SAMN08162544\n",
      "GCF_002885165.1 SAMN08162543\n",
      "GCF_002885195.1 SAMN08162540\n",
      "GCF_002885215.1 SAMN08162539\n",
      "GCF_002885235.1 SAMN08162538\n",
      "GCF_002885255.1 SAMN08162534\n",
      "GCF_002885265.1 SAMN08162533\n",
      "GCF_002885295.1 SAMN08162531\n",
      "GCF_002885315.1 SAMN08162529\n",
      "GCF_002885335.1 SAMN08162528\n",
      "GCF_002885355.1 SAMN08162527\n",
      "GCF_002885375.1 SAMN08162526\n",
      "GCF_002885395.1 SAMN08162525\n",
      "GCF_002885415.1 SAMN08162524\n",
      "GCF_002885425.1 SAMN08162523\n",
      "GCF_002885455.1 SAMN08162522\n",
      "GCF_002885465.1 SAMN08162520\n",
      "GCF_002885515.1 SAMN08162555\n",
      "GCF_002885535.1 SAMN08162553\n",
      "GCF_002885575.1 SAMN08162542\n",
      "GCF_002885595.1 SAMN08162541\n",
      "GCF_002885615.1 SAMN08162537\n",
      "GCF_002885625.1 SAMN08162536\n",
      "GCF_002885655.1 SAMN08162535\n",
      "GCF_002885675.1 SAMN08162532\n",
      "GCF_002885695.1 SAMN08162530\n",
      "GCF_002885715.1 SAMN08162521\n",
      "GCF_003716915.1 SAMN07978491\n",
      "GCF_003716935.1 SAMN07980909\n",
      "GCF_003716955.1 SAMN07980958\n",
      "GCF_003716975.1 SAMN07980963\n",
      "GCF_004015005.1 SAMN08329147\n",
      "GCF_004015025.1 SAMN08329199\n",
      "GCF_004015045.1 SAMN08334970\n",
      "GCF_004015065.1 SAMN08334973\n",
      "GCF_004015085.1 SAMN08334978\n",
      "GCF_004015105.1 SAMN08334979\n",
      "GCF_004015125.1 SAMN08334983\n",
      "GCF_004015205.1 SAMN08334976\n",
      "GCF_004015225.1 SAMN08334977\n",
      "GCF_004015245.1 SAMN08328879\n",
      "GCF_004015265.1 SAMN08329008\n",
      "GCF_004015285.1 SAMN08334972\n",
      "GCF_004015305.1 SAMN08334975\n",
      "GCF_004015325.1 SAMN08334981\n",
      "GCF_004015345.1 SAMN08334982\n",
      "GCF_004104435.1 SAMN10186677\n",
      "GCF_004168105.1 SAMN10239579\n",
      "GCF_004319565.1 SAMN10963682\n",
      "GCF_008000975.1 SAMN12556229\n",
      "GCF_008422275.1 SAMN11943062\n",
      "GCF_008422615.1 SAMN11943045\n",
      "GCF_008422635.1 SAMN11943044\n",
      "GCF_008422665.1 SAMN11943043\n",
      "GCF_008422685.1 SAMN11943039\n",
      "GCF_008422695.1 SAMN11943041\n",
      "GCF_008422705.1 SAMN11943040\n",
      "GCF_008422715.1 SAMN11943042\n",
      "GCF_008422765.1 SAMN11943036\n",
      "GCF_008422775.1 SAMN11943037\n",
      "GCF_008422795.1 SAMN11943035\n",
      "GCF_008422805.1 SAMN11943034\n",
      "GCF_008422815.1 SAMN11943038\n",
      "GCF_008422865.1 SAMN11943032\n",
      "GCF_008422875.1 SAMN11943030\n",
      "GCF_008422885.1 SAMN11943033\n",
      "GCF_008422895.1 SAMN11943031\n",
      "GCF_008422925.1 SAMN11943029\n",
      "GCF_008422965.1 SAMN11943027\n",
      "GCF_008422985.1 SAMN11943026\n",
      "GCF_008422995.1 SAMN11943028\n",
      "GCF_008423015.1 SAMN11943024\n",
      "GCF_008423035.1 SAMN11943025\n",
      "GCF_008423065.1 SAMN11943023\n",
      "GCF_008423085.1 SAMN11943021\n",
      "GCF_008423095.1 SAMN11943022\n",
      "GCF_008423115.1 SAMN11943020\n",
      "GCF_008423135.1 SAMN11943019\n",
      "GCF_008423175.1 SAMN11943018\n",
      "GCF_008423185.1 SAMN11943016\n",
      "GCF_008423215.1 SAMN11943015\n",
      "GCF_008423245.1 SAMN11943014\n",
      "GCF_008423265.1 SAMN11943013\n",
      "GCF_008423275.1 SAMN11943011\n",
      "GCF_008423295.1 SAMN11943009\n",
      "GCF_008423325.1 SAMN11943008\n",
      "GCF_008423335.1 SAMN11943007\n",
      "GCF_008423365.1 SAMN11943006\n",
      "GCF_008423375.1 SAMN11943004\n",
      "GCF_008423385.1 SAMN11943003\n",
      "GCF_008423405.1 SAMN11943005\n",
      "GCF_008423445.1 SAMN11943002\n",
      "GCF_008423455.1 SAMN11943001\n",
      "GCF_009663785.1 SAMN10240011\n",
      "GCF_009731575.1 SAMD00192834\n",
      "GCF_010223015.1 SAMN08563913\n",
      "GCF_010223075.1 SAMN08563914\n",
      "GCF_010223095.1 SAMN08565495\n",
      "GCF_010223115.1 SAMN08565496\n",
      "GCF_010223135.1 SAMN08565497\n",
      "GCF_010223155.1 SAMN08565498\n",
      "GCF_010223175.1 SAMN08565499\n",
      "GCF_010223195.1 SAMN08565501\n",
      "GCF_010223255.1 SAMN08565500\n",
      "GCF_010223405.1 SAMN08565502\n",
      "GCF_010223575.1 SAMN08565503\n",
      "GCF_010229055.1 SAMN09287185\n",
      "GCF_010229495.1 SAMN09287631\n",
      "GCF_010229695.1 SAMN09287781\n",
      "GCF_010229955.1 SAMN09287782\n",
      "GCF_010230155.1 SAMN09287783\n",
      "GCF_010230375.1 SAMN09287787\n",
      "GCF_010230515.1 SAMN09288166\n",
      "GCF_010230725.1 SAMN09288168\n",
      "GCF_010230935.1 SAMN09288183\n",
      "GCF_010231195.1 SAMN09288185\n",
      "GCF_010509235.1 SAMN13925612\n",
      "GCF_016697425.1 SAMN16911570\n",
      "GCF_017504145.1 SAMN00138213\n",
      "GCF_017566105.1 SAMN14738317\n",
      "GCF_017570425.1 SAMN18309827\n",
      "GCF_017570485.1 SAMN18309825\n",
      "GCF_017570505.1 SAMN18309826\n",
      "GCF_017570525.1 SAMN18309824\n",
      "GCF_900184845.1 SAMEA104102601\n",
      "GCF_900184905.1 SAMEA104102612\n",
      "GCF_900184985.1 SAMEA104102607\n",
      "GCF_900185025.1 SAMEA104102606\n",
      "GCF_001683795.1 SAMN03840437\n",
      "GCF_900097105.1 SAMEA4378782\n",
      "GCF_900184965.1 SAMEA104102608\n",
      "biosample_descr_non_akk.tsv\n",
      "GCA_000155695.1 SAMN02436317\n",
      "GCA_000242935.3 SAMN02261410\n",
      "GCA_000243495.3 SAMN02261400\n",
      "GCA_000526255.1 SAMN02584925\n",
      "GCA_000739615.1 SAMD00036766\n",
      "GCA_000739655.1 SAMD00036768\n",
      "GCA_000972765.1 SAMN03452270\n",
      "GCA_001872735.1 SAMN04328238\n",
      "GCA_001897195.1 SAMN05660525\n",
      "GCA_002290545.1 SAMN06820743\n",
      "GCA_002291675.1 SAMN06455474\n",
      "GCA_002293145.1 SAMN06451665\n",
      "GCA_002304445.1 SAMN06820747\n",
      "GCA_002307065.1 SAMN06454786\n",
      "GCA_002323615.1 SAMN06456429\n",
      "GCA_002323895.1 SAMN06452620\n",
      "GCA_002323955.1 SAMN06452625\n",
      "GCA_002323995.1 SAMN06456430\n",
      "GCA_002344525.1 SAMN06451526\n",
      "GCA_002346535.1 SAMN06456355\n",
      "GCA_002364575.1 SAMN06451190\n",
      "GCA_002376875.1 SAMN06455295\n",
      "GCA_002388665.1 SAMN06455775\n",
      "GCA_002396485.1 SAMN06452429\n",
      "GCA_002420185.1 SAMN06457676\n",
      "GCA_002420305.1 SAMN06455115\n",
      "GCA_002420315.1 SAMN06455114\n",
      "GCA_002422425.1 SAMN06453817\n",
      "GCA_002428665.1 SAMN06452880\n",
      "GCA_002429005.1 SAMN06455369\n",
      "GCA_002429065.1 SAMN06455374\n",
      "GCA_002469895.1 SAMN06457656\n",
      "GCA_002500665.1 SAMN06451839\n",
      "GCA_002501065.1 SAMN06455358\n",
      "GCA_002686085.1 SAMN07620270\n",
      "GCA_002699365.1 SAMN07618560\n",
      "GCA_002713695.1 SAMN07619560\n",
      "GCA_002726915.1 SAMN07620205\n",
      "GCA_002729725.1 SAMN07619857\n",
      "GCA_002730975.1 SAMN07619570\n",
      "GCA_002771405.1 SAMN06659651\n",
      "GCA_002784415.1 SAMN06659844\n",
      "GCA_002791315.1 SAMN06660124\n",
      "GCA_003054655.1 SAMN08724137\n",
      "GCA_003054665.1 SAMN08724138\n",
      "GCA_003054695.1 SAMN08724136\n",
      "GCA_003131465.1 SAMN08179500\n",
      "GCA_003131485.1 SAMN08179498\n",
      "GCA_003134735.1 SAMN08179325\n",
      "GCA_003134765.1 SAMN08179315\n",
      "GCA_003134785.1 SAMN08179313\n",
      "GCA_003140255.1 SAMN08179124\n",
      "GCA_003142615.1 SAMN08180119\n",
      "GCA_003142955.1 SAMN08180081\n",
      "GCA_003151515.1 SAMN08179892\n",
      "GCA_003151575.1 SAMN08179889\n",
      "GCA_003164675.1 SAMN08179934\n",
      "GCA_003164715.1 SAMN08179932\n",
      "GCA_003167365.1 SAMN08178918\n",
      "GCA_003167555.1 SAMN08178892\n",
      "GCA_003167635.1 SAMN08178888\n",
      "GCA_003169695.1 SAMN08179111\n",
      "GCA_003169855.1 SAMN08179103\n",
      "GCA_003169975.1 SAMN08179093\n",
      "GCA_003170395.1 SAMN08179092\n",
      "GCA_003219195.1 SAMN08911969\n",
      "GCA_003219545.1 SAMN08911982\n",
      "GCA_003226295.1 SAMN09042165\n",
      "GCA_003558065.1 SAMN08574537\n",
      "GCA_003559525.1 SAMN08574493\n",
      "GCA_004293065.1 SAMN10167093\n",
      "GCA_004293105.1 SAMN10167089\n",
      "GCA_004293445.1 SAMN10167078\n",
      "GCA_004294095.1 SAMN10167050\n",
      "GCA_004321905.1 SAMN10720107\n",
      "GCA_005809105.1 SAMN11369443\n",
      "GCA_007119975.1 SAMN10606099\n",
      "GCA_007125965.1 SAMN10607148\n",
      "GCA_007129275.1 SAMN10606984\n",
      "GCA_007130045.1 SAMN10610685\n",
      "GCA_007133065.1 SAMN10610532\n",
      "GCA_007134215.1 SAMN10610477\n",
      "GCA_008080695.1 SAMN11532986\n",
      "GCA_008932965.1 SAMN09684548\n",
      "GCA_009772895.1 SAMN10737522\n",
      "GCA_009773355.1 SAMN11533007\n",
      "GCA_009885995.1 SAMN12036116\n",
      "GCA_009886045.1 SAMN12036114\n",
      "GCA_009936795.1 SAMN13828881\n",
      "GCA_009937095.1 SAMN13828870\n",
      "GCA_010500615.1 SAMN14001152\n",
      "GCA_010681825.2 SAMN14056161\n",
      "GCA_011047805.1 SAMN09639370\n",
      "GCA_011054375.1 SAMN09639233\n",
      "GCA_011058495.1 SAMN09639278\n",
      "GCA_011327725.1 SAMN09639624\n",
      "GCA_011327785.1 SAMN09639621\n",
      "GCA_011333855.1 SAMN09639970\n",
      "GCA_011362745.1 SAMN09639488\n",
      "GCA_011367605.1 SAMN09639679\n",
      "GCA_011368065.1 SAMN09639645\n",
      "GCA_011368315.1 SAMN09639610\n",
      "GCA_011525695.1 SAMN13738158\n",
      "GCA_012270205.1 SAMN12098052\n",
      "GCA_012517065.1 SAMN13894314\n",
      "GCA_012517205.1 SAMN13894308\n",
      "GCA_012959985.1 SAMN10967654\n",
      "GCA_012960505.1 SAMN10967627\n",
      "GCA_013041925.1 SAMN14538091\n",
      "GCA_013215165.1 SAMN14675768\n",
      "GCA_013391475.1 SAMN15363799\n",
      "GCA_013820875.1 SAMN15052335\n",
      "GCA_014193315.1 SAMD00166100\n",
      "GCA_014239365.1 SAMN14533092\n",
      "GCA_014381645.1 SAMN15784325\n",
      "GCA_014382585.1 SAMN15784277\n",
      "GCA_014529625.1 SAMN12856402\n",
      "GCA_014529665.1 SAMN12856396\n",
      "GCA_014529675.1 SAMN12856399\n",
      "GCA_014879535.1 SAMN12995740\n",
      "GCA_014879575.1 SAMN12995737\n",
      "GCA_014879935.1 SAMN12995720\n",
      "GCA_015075305.1 SAMN15013958\n",
      "GCA_015075365.1 SAMN15013953\n",
      "GCA_015659695.1 SAMN09926294\n",
      "GCA_015767365.1 SAMN16793119\n",
      "GCA_016187925.1 SAMN15434943\n",
      "GCA_016190985.1 SAMN15435119\n",
      "GCA_016195385.1 SAMN15435534\n",
      "GCA_016199985.1 SAMN15435690\n",
      "GCA_016217545.1 SAMN15436431\n",
      "GCA_016222805.1 SAMN15436807\n",
      "GCA_016222825.1 SAMN15436806\n",
      "GCA_016235555.1 SAMN15436645\n",
      "GCA_016462965.1 SAMN16655404\n",
      "GCA_016463145.1 SAMN16655394\n",
      "GCA_016595455.1 SAMN17169595\n",
      "GCA_016595575.1 SAMN17169588\n",
      "GCA_016713245.1 SAMN16425973\n",
      "GCA_016716505.1 SAMN16426159\n",
      "GCA_016777945.1 SAMN16436173\n",
      "GCA_016793645.1 SAMN17180891\n",
      "GCA_016793885.1 SAMN17180892\n",
      "GCA_016795365.1 SAMN17180901\n",
      "GCA_016795385.1 SAMN17180896\n",
      "GCA_016795425.1 SAMN17180894\n",
      "GCA_016795485.1 SAMN17180900\n",
      "GCA_016795505.1 SAMN17180895\n",
      "GCA_016795565.1 SAMN17180884\n",
      "GCA_016811075.1 SAMN17711332\n",
      "GCA_016871495.1 SAMN10967018\n",
      "GCA_016871535.1 SAMN10967012\n",
      "GCA_016871575.1 SAMN10967009\n",
      "GCA_016871755.1 SAMN10966996\n",
      "GCA_016871805.1 SAMN10966993\n",
      "GCA_016928295.1 SAMN17242028\n",
      "GCA_016935555.1 SAMN17243333\n",
      "GCA_017305195.1 SAMN18061515\n",
      "GCA_017309115.1 SAMN18059914\n",
      "GCA_900547395.1 SAMEA4891626\n",
      "GCA_900760665.1 SAMEA5279772\n",
      "GCA_900763545.1 SAMEA5279771\n",
      "GCA_903820035.1 SAMEA6944815\n",
      "GCA_903822625.1 SAMEA6944443\n",
      "GCA_903823585.1 SAMEA6946406\n",
      "GCA_903823955.1 SAMEA6950342\n",
      "GCA_903825655.1 SAMEA6956414\n",
      "GCA_903825845.1 SAMEA6943583\n",
      "GCA_903826105.1 SAMEA6946973\n",
      "GCA_903827225.1 SAMEA6949337\n",
      "GCA_903827245.1 SAMEA6949399\n",
      "GCA_903827375.1 SAMEA6949914\n",
      "GCA_903827955.1 SAMEA6954115\n",
      "GCA_903828255.1 SAMEA6954361\n",
      "GCA_903829895.1 SAMEA6947886\n",
      "GCA_903829915.1 SAMEA6947972\n",
      "GCA_903829935.1 SAMEA6947795\n",
      "GCA_903830045.1 SAMEA6948458\n",
      "GCA_903830155.1 SAMEA6945856\n",
      "GCA_903830425.1 SAMEA6949252\n",
      "GCA_903830575.1 SAMEA6949930\n",
      "GCA_903832615.1 SAMEA6947231\n",
      "GCA_903832955.1 SAMEA6945032\n",
      "GCA_903833605.1 SAMEA6946468\n",
      "GCA_903834135.1 SAMEA6951084\n",
      "GCA_903835255.1 SAMEA6955984\n",
      "GCA_903835285.1 SAMEA6956244\n",
      "GCA_903836245.1 SAMEA6948268\n",
      "GCA_903839375.1 SAMEA6953453\n",
      "GCA_903840355.1 SAMEA6943673\n",
      "GCA_903840415.1 SAMEA6944424\n",
      "GCA_903841665.1 SAMEA6950344\n",
      "GCA_903842505.1 SAMEA6952138\n",
      "GCA_903844265.1 SAMEA6950560\n",
      "GCA_903844885.1 SAMEA6954470\n",
      "GCA_903844915.1 SAMEA6954364\n",
      "GCA_903845515.1 SAMEA6944124\n",
      "GCA_903846165.1 SAMEA6945849\n",
      "GCA_903848695.1 SAMEA6946744\n",
      "GCA_903849065.1 SAMEA6945146\n",
      "GCA_903849435.1 SAMEA6945900\n",
      "GCA_903851165.1 SAMEA6955154\n",
      "GCA_903851485.1 SAMEA6956481\n",
      "GCA_903851505.1 SAMEA6956381\n",
      "GCA_903854255.1 SAMEA6954854\n",
      "GCA_903854295.1 SAMEA6955214\n",
      "GCA_903854475.1 SAMEA6956008\n",
      "GCA_903855995.1 SAMEA6947847\n",
      "GCA_903856725.1 SAMEA6948856\n",
      "GCA_903856775.1 SAMEA6948787\n",
      "GCA_903859495.1 SAMEA6944204\n",
      "GCA_903859715.1 SAMEA6946987\n",
      "GCA_903859905.1 SAMEA6944535\n",
      "GCA_903862025.1 SAMEA6949848\n",
      "GCA_903862405.1 SAMEA6953741\n",
      "GCA_903862595.1 SAMEA6951930\n",
      "GCA_903862765.1 SAMEA6954964\n",
      "GCA_903865455.1 SAMEA6947796\n",
      "GCA_903865745.1 SAMEA6948611\n",
      "GCA_903867105.1 SAMEA6944701\n",
      "GCA_903869175.1 SAMEA6948331\n",
      "GCA_903869675.1 SAMEA6951122\n",
      "GCA_903870255.1 SAMEA6955290\n",
      "GCA_903870815.1 SAMEA6947619\n",
      "GCA_903870895.1 SAMEA6945026\n",
      "GCA_903871035.1 SAMEA6948255\n",
      "GCA_903871185.1 SAMEA6948669\n",
      "GCA_903871215.1 SAMEA6948735\n",
      "GCA_903872345.1 SAMEA6946728\n",
      "GCA_903872885.1 SAMEA6946428\n",
      "GCA_903873415.1 SAMEA6954104\n",
      "GCA_903873795.1 SAMEA6955794\n",
      "GCA_903873945.1 SAMEA6956650\n",
      "GCA_903874045.1 SAMEA6944314\n",
      "GCA_903875465.1 SAMEA6955522\n",
      "GCA_903875675.1 SAMEA6944266\n",
      "GCA_903875685.1 SAMEA6944104\n",
      "GCA_903875755.1 SAMEA6944461\n",
      "GCA_903876035.1 SAMEA6947827\n",
      "GCA_903877055.1 SAMEA6955873\n",
      "GCA_903877135.1 SAMEA6955875\n",
      "GCA_903878115.1 SAMEA6951628\n",
      "GCA_903881465.1 SAMEA6955893\n",
      "GCA_903882455.1 SAMEA6950452\n",
      "GCA_903882875.1 SAMEA6954865\n",
      "GCA_903883725.1 SAMEA6949408\n",
      "GCA_903884295.1 SAMEA6954921\n",
      "GCA_903885235.1 SAMEA6950004\n",
      "GCA_903885245.1 SAMEA6949997\n",
      "GCA_903886895.1 SAMEA6951096\n",
      "GCA_903888785.1 SAMEA6943597\n",
      "GCA_903888925.1 SAMEA6946963\n",
      "GCA_903891145.1 SAMEA6954686\n",
      "GCA_903891255.1 SAMEA6955469\n",
      "GCA_903892385.1 SAMEA6951567\n",
      "GCA_903893265.1 SAMEA6949830\n",
      "GCA_903893885.1 SAMEA6955959\n",
      "GCA_903896015.1 SAMEA6947793\n",
      "GCA_903896365.1 SAMEA6949987\n",
      "GCA_903896505.1 SAMEA6953829\n",
      "GCA_903896705.1 SAMEA6954332\n",
      "GCA_903898185.1 SAMEA6947576\n",
      "GCA_903898675.1 SAMEA6954297\n",
      "GCA_903900845.1 SAMEA6952525\n",
      "GCA_903901425.1 SAMEA6946097\n",
      "GCA_903902115.1 SAMEA6944320\n",
      "GCA_903902745.1 SAMEA6954005\n",
      "GCA_903903235.1 SAMEA6944679\n",
      "GCA_903903695.1 SAMEA6952195\n",
      "GCA_903905115.1 SAMEA6945919\n",
      "GCA_903905305.1 SAMEA6950709\n",
      "GCA_903906515.1 SAMEA6956830\n",
      "GCA_903906645.1 SAMEA6944681\n",
      "GCA_903907025.1 SAMEA6951751\n",
      "GCA_903907925.1 SAMEA6955093\n",
      "GCA_903907995.1 SAMEA6955912\n",
      "GCA_903910895.1 SAMEA6949277\n",
      "GCA_903912925.1 SAMEA6947578\n",
      "GCA_903913865.1 SAMEA6951691\n",
      "GCA_903915115.1 SAMEA6954449\n",
      "GCA_903915855.1 SAMEA6944368\n",
      "GCA_903916085.1 SAMEA6954269\n",
      "GCA_903916625.1 SAMEA6951599\n",
      "GCA_903917335.1 SAMEA6945143\n",
      "GCA_903918605.1 SAMEA6952246\n",
      "GCA_903918755.1 SAMEA6944391\n",
      "GCA_903918885.1 SAMEA6947962\n",
      "GCA_903919125.1 SAMEA6956003\n",
      "GCA_903919175.1 SAMEA6944400\n",
      "GCA_903919975.1 SAMEA6951887\n",
      "GCA_903921715.1 SAMEA6954395\n",
      "GCA_903922605.1 SAMEA6947267\n",
      "GCA_903923245.1 SAMEA6956407\n",
      "GCA_903923255.1 SAMEA6955238\n",
      "GCA_903925145.1 SAMEA6948093\n",
      "GCA_903925935.1 SAMEA6953871\n",
      "GCA_903928105.1 SAMEA6955050\n",
      "GCA_903930345.1 SAMEA6951950\n",
      "GCA_903930415.1 SAMEA6956850\n",
      "GCA_903933715.1 SAMEA6946304\n",
      "GCA_903934285.1 SAMEA6945188\n",
      "GCA_903939515.1 SAMEA6949779\n",
      "GCA_903940805.1 SAMEA6951443\n",
      "GCA_903943535.1 SAMEA6945174\n",
      "GCA_903944655.1 SAMEA6948502\n",
      "GCA_903945405.1 SAMEA6945843\n",
      "GCA_903945705.1 SAMEA6950841\n",
      "GCA_903950355.1 SAMEA6950144\n",
      "GCA_903951815.1 SAMEA6945554\n",
      "GCA_903952285.1 SAMEA6949685\n",
      "GCA_903960095.1 SAMEA6946833\n",
      "GCA_904419695.1 SAMEA7202372\n",
      "GCA_904424815.1 SAMEA7306389\n",
      "GCA_905182405.1 SAMEA7768543\n",
      "GCA_905182465.1 SAMEA7768562\n",
      "GCA_905182475.1 SAMEA7768561\n",
      "GCA_905193115.1 SAMEA7846742\n",
      "GCA_905198945.1 SAMEA7846640\n",
      "GCA_905205865.1 SAMEA7846715\n",
      "GCF_000019665.1 SAMN02604183\n",
      "GCF_000019965.1 SAMN00000285\n",
      "GCF_000025905.1 SAMN02598501\n",
      "GCF_000171235.2 SAMN02256391\n",
      "GCF_000172155.1 SAMN02436175\n",
      "GCF_000172555.1 SAMN00000631\n",
      "GCF_000173075.1 SAMN00000678\n",
      "GCF_000297415.1 SAMEA2272587\n",
      "GCF_000378105.1 SAMN02441342\n",
      "GCF_000379365.1 SAMN02256457\n",
      "GCF_000817245.1 SAMN02996356\n",
      "GCF_001580015.1 SAMN04456986\n",
      "GCF_001580045.1 SAMN04456987\n",
      "GCF_001613545.1 SAMD00048637\n",
      "GCF_001746835.1 SAMN04880612\n",
      "GCF_002310495.1 SAMN07628394\n",
      "GCF_002954445.1 SAMN06090799\n",
      "GCF_003054705.1 SAMN08724139\n",
      "GCF_003185655.1 SAMN08997837\n",
      "GCF_004118375.1 SAMN10780907\n",
      "GCF_004341915.1 SAMN10866388\n",
      "GCF_004364345.1 SAMN02745162\n",
      "GCF_004421155.1 SAMN04957793\n",
      "GCF_004421165.1 SAMN04957802\n",
      "GCF_004421175.1 SAMN04957800\n",
      "GCF_004421185.1 SAMN04957807\n",
      "GCF_004421195.1 SAMN04932749\n",
      "GCF_004421255.1 SAMN04957809\n",
      "GCF_005780165.1 SAMN11569541\n",
      "GCF_007475525.1 SAMN08451303\n",
      "GCF_007559335.1 SAMN10623051\n",
      "GCF_007992435.1 SAMD00172670\n",
      "GCF_011044495.1 SAMN14143996\n",
      "GCF_011044775.1 SAMN14102279\n",
      "GCF_012913485.1 SAMN14642216\n",
      "GCF_014201715.1 SAMN14908355\n",
      "GCF_014230085.1 SAMN15676579\n",
      "GCF_014230145.1 SAMN15676578\n",
      "GCF_014651635.1 SAMD00245575\n",
      "GCF_014651675.1 SAMD00245577\n",
      "GCF_014803405.1 SAMN16205841\n",
      "GCF_016595415.1 SAMN17169596\n",
      "GCF_016595435.1 SAMN17169591\n",
      "GCF_016595465.1 SAMN17169592\n",
      "GCF_016595495.1 SAMN17169594\n",
      "GCF_016595505.1 SAMN17169593\n",
      "GCF_016595525.1 SAMN17169590\n",
      "GCF_016595555.1 SAMN17169589\n",
      "GCF_017310505.1 SAMN17036178\n",
      "GCF_017310525.1 SAMN17036247\n",
      "GCF_017310545.1 SAMN17073650\n",
      "GCF_900104925.1 SAMN05444173\n",
      "GCF_900105685.1 SAMN05444156\n",
      "GCF_900141815.1 SAMN02745181\n",
      "GCF_900167535.1 SAMN02745166\n",
      "GCF_902143375.2 SAMEA5600707\n",
      "GCF_902143385.2 SAMEA5600708\n",
      "GCF_902726575.1 SAMEA6101992\n",
      "GCF_902726585.1 SAMEA6101991\n",
      "GCF_902726595.1 SAMEA6101989\n",
      "GCF_902726605.1 SAMEA6101987\n",
      "GCF_902728225.1 SAMEA6101988\n",
      "GCF_902728235.1 SAMEA6101985\n",
      "GCF_902728245.1 SAMEA6101986\n",
      "GCF_902728255.1 SAMEA6101990\n"
     ]
    }
   ],
   "source": [
    "# des_files = ['biosample_descr_akk.tsv', 'biosample_descr_non_akk.tsv']\n",
    "# idxs = [akk_ncbi, non_akk_ncbi]\n",
    "# for des_file,accs in zip(des_files, idxs):\n",
    "#     print(des_file)\n",
    "#     ! rm $des_file\n",
    "#     bs_descr = defaultdict()\n",
    "#     fh = open(des_file, 'a')\n",
    "#     # fh.write('Biosample\\tProject_Description\\tProject_Name\\tProject_Title\\n')\n",
    "#     for idx,bs in ncbi_df.set_index('assembly_accession').loc[accs.flatten(), :]['biosample'].iteritems():\n",
    "#         print(idx, bs)\n",
    "#         desc = ! esearch -db biosample -query $bs | efetch -format docsum | xtract -pattern DocumentSummary -element Title Attribute \n",
    "#         desc_ls = desc.get_spstr().split('\\t')\n",
    "#         desc_ls.insert(0, bs)\n",
    "#         desc_ls.insert(0, idx)\n",
    "#         bs_descr[idx] = desc_ls\n",
    "#         print('\\t'.join(desc_ls), file=fh)\n",
    "#     fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "54def8d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "180 biosample_descr_akk.tsv\n",
      "384 biosample_descr_non_akk.tsv\n"
     ]
    }
   ],
   "source": [
    "for f in des_files:\n",
    "    ! wc -l $f"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5cfe78c8",
   "metadata": {},
   "source": [
    "### For BioProject"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "e0b29d04",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000436395.1 PRJEB726\n",
      "GCA_000437075.1 PRJEB840\n",
      "GCA_001917295.1 PRJNA321218\n",
      "GCA_002362435.1 PRJNA348753\n",
      "GCA_002492685.1 PRJNA348753\n",
      "GCA_002493465.1 PRJNA348753\n",
      "GCA_003514565.1 PRJNA417962\n",
      "GCA_008672035.1 PRJNA492716\n",
      "GCA_008672205.1 PRJNA492716\n",
      "GCA_009773845.1 PRJNA544874\n",
      "GCA_014871965.1 PRJNA562766\n",
      "GCA_014871995.1 PRJNA562766\n",
      "GCA_014872005.1 PRJNA562766\n",
      "GCA_014872035.1 PRJNA562766\n",
      "GCA_014872065.1 PRJNA562766\n",
      "GCA_014872075.1 PRJNA562766\n",
      "GCA_014872105.1 PRJNA562766\n",
      "GCA_014872125.1 PRJNA562766\n",
      "GCA_014872165.1 PRJNA562766\n",
      "GCA_900539575.1 PRJEB26432\n",
      "GCA_900539895.1 PRJEB26432\n",
      "GCA_900545155.1 PRJEB26432\n",
      "GCA_900752515.1 PRJEB31003\n",
      "GCA_900759685.1 PRJEB31003\n",
      "GCA_900765745.1 PRJEB31003\n",
      "GCA_905193715.1 PRJEB37358\n",
      "GCA_905197035.1 PRJEB37358\n",
      "GCA_905200125.1 PRJEB37358\n",
      "GCA_905200945.1 PRJEB37358\n",
      "GCA_905205055.1 PRJEB37358\n",
      "GCA_905205915.1 PRJEB37358\n",
      "GCA_905210405.1 PRJEB37358\n",
      "GCA_905210615.1 PRJEB37358\n",
      "GCF_000020225.1 PRJNA224116\n",
      "GCF_000723745.2 PRJNA224116\n",
      "GCF_001578645.1 PRJNA224116\n",
      "GCF_001580195.1 PRJNA224116\n",
      "GCF_001647615.1 PRJNA224116\n",
      "GCF_001688765.2 PRJNA224116\n",
      "GCF_002161325.1 PRJNA224116\n",
      "GCF_002201495.1 PRJNA224116\n",
      "GCF_002884915.1 PRJNA224116\n",
      "GCF_002884975.1 PRJNA224116\n",
      "GCF_002884995.1 PRJNA224116\n",
      "GCF_002885015.1 PRJNA224116\n",
      "GCF_002885025.1 PRJNA224116\n",
      "GCF_002885055.1 PRJNA224116\n",
      "GCF_002885075.1 PRJNA224116\n",
      "GCF_002885095.1 PRJNA224116\n",
      "GCF_002885155.1 PRJNA224116\n",
      "GCF_002885165.1 PRJNA224116\n",
      "GCF_002885195.1 PRJNA224116\n",
      "GCF_002885215.1 PRJNA224116\n",
      "GCF_002885235.1 PRJNA224116\n",
      "GCF_002885255.1 PRJNA224116\n",
      "GCF_002885265.1 PRJNA224116\n",
      "GCF_002885295.1 PRJNA224116\n",
      "GCF_002885315.1 PRJNA224116\n",
      "GCF_002885335.1 PRJNA224116\n",
      "GCF_002885355.1 PRJNA224116\n",
      "GCF_002885375.1 PRJNA224116\n",
      "GCF_002885395.1 PRJNA224116\n",
      "GCF_002885415.1 PRJNA224116\n",
      "GCF_002885425.1 PRJNA224116\n",
      "GCF_002885455.1 PRJNA224116\n",
      "GCF_002885465.1 PRJNA224116\n",
      "GCF_002885515.1 PRJNA224116\n",
      "GCF_002885535.1 PRJNA224116\n",
      "GCF_002885575.1 PRJNA224116\n",
      "GCF_002885595.1 PRJNA224116\n",
      "GCF_002885615.1 PRJNA224116\n",
      "GCF_002885625.1 PRJNA224116\n",
      "GCF_002885655.1 PRJNA224116\n",
      "GCF_002885675.1 PRJNA224116\n",
      "GCF_002885695.1 PRJNA224116\n",
      "GCF_002885715.1 PRJNA224116\n",
      "GCF_003716915.1 PRJNA224116\n",
      "GCF_003716935.1 PRJNA224116\n",
      "GCF_003716955.1 PRJNA224116\n",
      "GCF_003716975.1 PRJNA224116\n",
      "GCF_004015005.1 PRJNA224116\n",
      "GCF_004015025.1 PRJNA224116\n",
      "GCF_004015045.1 PRJNA224116\n",
      "GCF_004015065.1 PRJNA224116\n",
      "GCF_004015085.1 PRJNA224116\n",
      "GCF_004015105.1 PRJNA224116\n",
      "GCF_004015125.1 PRJNA224116\n",
      "GCF_004015205.1 PRJNA224116\n",
      "GCF_004015225.1 PRJNA224116\n",
      "GCF_004015245.1 PRJNA224116\n",
      "GCF_004015265.1 PRJNA224116\n",
      "GCF_004015285.1 PRJNA224116\n",
      "GCF_004015305.1 PRJNA224116\n",
      "GCF_004015325.1 PRJNA224116\n",
      "GCF_004015345.1 PRJNA224116\n",
      "GCF_004104435.1 PRJNA224116\n",
      "GCF_004168105.1 PRJNA224116\n",
      "GCF_004319565.1 PRJNA224116\n",
      "GCF_008000975.1 PRJNA224116\n",
      "GCF_008422275.1 PRJNA224116\n",
      "GCF_008422615.1 PRJNA224116\n",
      "GCF_008422635.1 PRJNA224116\n",
      "GCF_008422665.1 PRJNA224116\n",
      "GCF_008422685.1 PRJNA224116\n",
      "GCF_008422695.1 PRJNA224116\n",
      "GCF_008422705.1 PRJNA224116\n",
      "GCF_008422715.1 PRJNA224116\n",
      "GCF_008422765.1 PRJNA224116\n",
      "GCF_008422775.1 PRJNA224116\n",
      "GCF_008422795.1 PRJNA224116\n",
      "GCF_008422805.1 PRJNA224116\n",
      "GCF_008422815.1 PRJNA224116\n",
      "GCF_008422865.1 PRJNA224116\n",
      "GCF_008422875.1 PRJNA224116\n",
      "GCF_008422885.1 PRJNA224116\n",
      "GCF_008422895.1 PRJNA224116\n",
      "GCF_008422925.1 PRJNA224116\n",
      "GCF_008422965.1 PRJNA224116\n",
      "GCF_008422985.1 PRJNA224116\n",
      "GCF_008422995.1 PRJNA224116\n",
      "GCF_008423015.1 PRJNA224116\n",
      "GCF_008423035.1 PRJNA224116\n",
      "GCF_008423065.1 PRJNA224116\n",
      "GCF_008423085.1 PRJNA224116\n",
      "GCF_008423095.1 PRJNA224116\n",
      "GCF_008423115.1 PRJNA224116\n",
      "GCF_008423135.1 PRJNA224116\n",
      "GCF_008423175.1 PRJNA224116\n",
      "GCF_008423185.1 PRJNA224116\n",
      "GCF_008423215.1 PRJNA224116\n",
      "GCF_008423245.1 PRJNA224116\n",
      "GCF_008423265.1 PRJNA224116\n",
      "GCF_008423275.1 PRJNA224116\n",
      "GCF_008423295.1 PRJNA224116\n",
      "GCF_008423325.1 PRJNA224116\n",
      "GCF_008423335.1 PRJNA224116\n",
      "GCF_008423365.1 PRJNA224116\n",
      "GCF_008423375.1 PRJNA224116\n",
      "GCF_008423385.1 PRJNA224116\n",
      "GCF_008423405.1 PRJNA224116\n",
      "GCF_008423445.1 PRJNA224116\n",
      "GCF_008423455.1 PRJNA224116\n",
      "GCF_009663785.1 PRJNA224116\n",
      "GCF_009731575.1 PRJNA224116\n",
      "GCF_010223015.1 PRJNA224116\n",
      "GCF_010223075.1 PRJNA224116\n",
      "GCF_010223095.1 PRJNA224116\n",
      "GCF_010223115.1 PRJNA224116\n",
      "GCF_010223135.1 PRJNA224116\n",
      "GCF_010223155.1 PRJNA224116\n",
      "GCF_010223175.1 PRJNA224116\n",
      "GCF_010223195.1 PRJNA224116\n",
      "GCF_010223255.1 PRJNA224116\n",
      "GCF_010223405.1 PRJNA224116\n",
      "GCF_010223575.1 PRJNA224116\n",
      "GCF_010229055.1 PRJNA224116\n",
      "GCF_010229495.1 PRJNA224116\n",
      "GCF_010229695.1 PRJNA224116\n",
      "GCF_010229955.1 PRJNA224116\n",
      "GCF_010230155.1 PRJNA224116\n",
      "GCF_010230375.1 PRJNA224116\n",
      "GCF_010230515.1 PRJNA224116\n",
      "GCF_010230725.1 PRJNA224116\n",
      "GCF_010230935.1 PRJNA224116\n",
      "GCF_010231195.1 PRJNA224116\n",
      "GCF_010509235.1 PRJNA224116\n",
      "GCF_016697425.1 PRJNA224116\n",
      "GCF_017504145.1 PRJNA224116\n",
      "GCF_017566105.1 PRJNA224116\n",
      "GCF_017570425.1 PRJNA224116\n",
      "GCF_017570485.1 PRJNA224116\n",
      "GCF_017570505.1 PRJNA224116\n",
      "GCF_017570525.1 PRJNA224116\n",
      "GCF_900184845.1 PRJNA224116\n",
      "GCF_900184905.1 PRJNA224116\n",
      "GCF_900184985.1 PRJNA224116\n",
      "GCF_900185025.1 PRJNA224116\n",
      "GCF_001683795.1 PRJNA224116\n",
      "GCF_900097105.1 PRJNA224116\n",
      "GCF_900184965.1 PRJNA224116\n"
     ]
    }
   ],
   "source": [
    "# ! rm bioproject_descr_akk.tsv\n",
    "# bp_descr = defaultdict()\n",
    "# fh = open('bioproject_descr_akk.tsv', 'a')\n",
    "# fh.write('Bioproject\\tProject_Description\\tProject_Name\\tProject_Title\\n')\n",
    "# for idx,bp in ncbi_df.set_index('assembly_accession').loc[akk_ncbi.flatten(), :]['bioproject'].iteritems():\n",
    "#     print(idx, bp)\n",
    "# #     i += 1\n",
    "# #     if i <100:\n",
    "#     desc = ! esearch -db bioproject -query $bp | efetch -format docsum |  xtract -pattern DocumentSummary -element Project_Name,Project_Title,Project_Description\n",
    "#     desc_ls = desc.get_spstr().split('\\t')\n",
    "#     desc_ls.insert(0, bp)\n",
    "#     bp_descr[idx] = desc_ls\n",
    "#     print('\\t'.join(desc_ls), file=fh)\n",
    "# fh.close()\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 537,
   "id": "c4d15555",
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_info_to_summary(df = merged_df, col_to_write='host', *acc_to_animal):\n",
    "    '''\n",
    "    Add 'host' or 'type_of_sample' to the summary df. This information comes from grepping \n",
    "    the scientific and common species names or any other regex against the metadata \n",
    "    from biosample and bioproject.\n",
    "    acc_to_anim : defaultdict/s with the form `genbank_acc:host_name`\n",
    "    col_to_write= any col_name, but here should be 'host' or 'type_of_sample'\n",
    "    '''\n",
    "    import pandas as pd\n",
    "    ncbi_genomes = df[df['Genome'].apply(lambda row: True if re.match('GCA|GCF', row) else False)]\n",
    "    for dic in acc_to_animal:\n",
    "        for i, row in ncbi_genomes.iterrows():\n",
    "            acc = row.str.extract( '(GC[AF]_[0-9]{9}\\.[1-9])')\n",
    "            if acc.loc['Genome',][0] in dic.keys():\n",
    "                df.loc[i, col_to_write] = dic[acc.loc['Genome',][0]]\n",
    "    df.loc[:, col_to_write] = df[col_to_write].str.lower()\n",
    "    print(df[col_to_write].value_counts())\n",
    "    return 0\n",
    "\n",
    "\n",
    "    \n",
    "def build_line_source(biosample_meta, pattern, file_pattern=False):\n",
    "    '''\n",
    "    Returns the number of the line and the matched word. (ripgrep result)\n",
    "    biosample_meta: samples description built with `efetch -db biosample`\n",
    "    pattern: Regex or a file with patterns. E.g.: \"'culture|soil|sludge|water|marine|biome'\", \n",
    "    notice the 2 quotes, which are needed for complex patterns to work.\n",
    "    file_pattern: Set it to True if a file with patterns is used. \n",
    "    '''\n",
    "    \n",
    "    import subprocess as sp\n",
    "    from collections import defaultdict\n",
    "    \n",
    "    pattern = f'-f {pattern}' if file_pattern else pattern\n",
    "    #uniq because one hit per line is enough\n",
    "    rg_cmd = f\"rg -woni {pattern} {biosample_meta} | sort -h | uniq\"\n",
    "    print(rg_cmd)\n",
    "    line_animal = sp.check_output(rg_cmd, shell=True, universal_newlines=True)\n",
    "    \n",
    "    return line_animal\n",
    "\n",
    "\n",
    "def build_acc_host(line_source, biosample_meta):\n",
    "    '''\n",
    "    For each line in the grip result (line_source), get the number of the line\n",
    "    and use it to  get the line from the metadata/biosample descriptions. Then, build a \n",
    "    dictionary from ncbi accession to source/biome/host_animal.\n",
    "    biosample_meta: samples description built with `efetch -db biosample`\n",
    "    '''\n",
    "    \n",
    "    with open(biosample_meta, 'r') as fh:\n",
    "        biosample_desc = fh.readlines()\n",
    "        \n",
    "    acc_to_source = defaultdict()\n",
    "    for l in line_source.strip().split('\\n'):\n",
    "        # python list, `biosample_desc`. starts from 0\n",
    "        n = int(l.split(':')[0]) - 1\n",
    "        acc = biosample_desc[n].split('\\t')[0]\n",
    "        acc_to_source[acc] = l.split(':')[1].lower()\n",
    "    print(len(acc_to_source))\n",
    "    print('index for the list in python:', n)\n",
    "    return acc_to_source\n",
    "\n",
    "def build_acc_host_fromdict(line_source_dict, biosample_meta, source):\n",
    "    '''\n",
    "    Accepts a dictionary with line numbers as keys and list of sources as values.\n",
    "    source: string, source to write in the table, e.g. 'marine water' \n",
    "    '''\n",
    "    \n",
    "    with open(biosample_meta, 'r') as fh:\n",
    "        biosample_desc = fh.readlines()\n",
    "        \n",
    "    acc_to_source = defaultdict()\n",
    "    for n in line_source_dict.keys():\n",
    "        # python list, `biosample_desc`. starts from 0\n",
    "        n = n - 1\n",
    "        acc = biosample_desc[n].split('\\t')[0]\n",
    "        acc_to_source[acc] = source\n",
    "    print(len(acc_to_source))\n",
    "    print('index for the list in python:', n)\n",
    "    return acc_to_source"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e6eee08c",
   "metadata": {},
   "source": [
    "### Search in bioproject metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "a2eb9013",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni 'stool|feces|gut|metagenome' bioproject_descr_akk.tsv | sort -h | uniq\n",
      "2:gut\n",
      "2:Gut\n",
      "2:stool\n",
      "3:gut\n",
      "3:Gut\n",
      "3:stool\n",
      "4:gut\n",
      "4:metagenome\n",
      "8:metagenome\n",
      "9:gut\n",
      "9:metagenome\n",
      "10:gut\n",
      "10\n",
      "29\n",
      "index for the list in python: 33\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'metagenome', 'stool'}"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ln_meta_indicative_bp =  build_line_source('bioproject_descr_akk.tsv', \"'stool|feces|gut|metagenome'\", \n",
    "                                file_pattern=False)\n",
    "print(ln_meta_indicative_bp[:100])\n",
    "acc_meta_indicative_bp = build_acc_host(ln_meta_indicative_bp, 'biosample_descr_akk.tsv')\n",
    "set(acc_meta_indicative_bp.values())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "031c13d1",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(None,\n",
       "            {'GCA_000437075.1': 'stool',\n",
       "             'GCA_001917295.1': 'stool',\n",
       "             'GCA_002362435.1': 'metagenome',\n",
       "             'GCA_008672035.1': 'metagenome',\n",
       "             'GCA_008672205.1': 'metagenome',\n",
       "             'GCA_009773845.1': 'metagenome',\n",
       "             'GCA_014871995.1': 'metagenome',\n",
       "             'GCA_014872005.1': 'metagenome',\n",
       "             'GCA_014872035.1': 'metagenome',\n",
       "             'GCA_014872065.1': 'metagenome',\n",
       "             'GCA_014872075.1': 'metagenome',\n",
       "             'GCA_014872105.1': 'metagenome',\n",
       "             'GCA_014872125.1': 'metagenome',\n",
       "             'GCA_014872165.1': 'metagenome',\n",
       "             'GCA_900539575.1': 'metagenome',\n",
       "             'GCA_900539895.1': 'metagenome',\n",
       "             'GCA_900545155.1': 'metagenome',\n",
       "             'GCA_900752515.1': 'metagenome',\n",
       "             'GCA_900759685.1': 'metagenome',\n",
       "             'GCA_900765745.1': 'metagenome',\n",
       "             'GCA_905193715.1': 'metagenome',\n",
       "             'GCA_905197035.1': 'metagenome',\n",
       "             'GCA_905200125.1': 'metagenome',\n",
       "             'GCA_905200945.1': 'metagenome',\n",
       "             'GCA_905205055.1': 'metagenome',\n",
       "             'GCA_905205915.1': 'metagenome',\n",
       "             'GCA_905210405.1': 'metagenome',\n",
       "             'GCA_905210615.1': 'metagenome',\n",
       "             'GCF_000020225.1': 'metagenome'})"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acc_meta_indicative_bp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "9888f96a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'GCF_000020225.1'}"
      ]
     },
     "execution_count": 171,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(acc_meta_indicative_bp.keys()) - set(acc_culture.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "bb36c1c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "mask = ~merged_akk['relation_to_type_material'].isna()\n",
    "merged_akk.loc[mask, 'type_of_sample'] = 'metagenome'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "58a4f0ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>classification</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>sum_len</th>\n",
       "      <th>N50</th>\n",
       "      <th>16S ribosomal RNA &gt; 1400 nt</th>\n",
       "      <th>Total_tRNAs</th>\n",
       "      <th>host</th>\n",
       "      <th>NCBI_organism_name</th>\n",
       "      <th>NCBI_infraspecific_name</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>type_of_sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>GCF_000020225.1_ASM2022v1_genomic</td>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2664102</td>\n",
       "      <td>2664102</td>\n",
       "      <td>3.0</td>\n",
       "      <td>54</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Akkermansia muciniphila ATCC BAA-835</td>\n",
       "      <td>strain=ATCC BAA-835</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>metagenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>451</th>\n",
       "      <td>GCF_008000975.1_ASM800097v1_genomic</td>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2664043</td>\n",
       "      <td>2664043</td>\n",
       "      <td>3.0</td>\n",
       "      <td>54</td>\n",
       "      <td>homo sapiens</td>\n",
       "      <td>Akkermansia muciniphila</td>\n",
       "      <td>strain=DSM 22959</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>metagenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>GCF_017504145.1_ASM1750414v1_genomic</td>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2664051</td>\n",
       "      <td>2664051</td>\n",
       "      <td>3.0</td>\n",
       "      <td>54</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Akkermansia muciniphila ATCC BAA-835</td>\n",
       "      <td>strain=ATCC BAA-835</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>metagenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>GCF_001683795.1_ASM168379v1_genomic</td>\n",
       "      <td>91.84</td>\n",
       "      <td>4.76</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>3157849</td>\n",
       "      <td>127311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>42</td>\n",
       "      <td>reticulated python</td>\n",
       "      <td>Akkermansia glycaniphila</td>\n",
       "      <td>strain=Pyt</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>metagenome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Genome  Completeness  Contamination  \\\n",
       "356     GCF_000020225.1_ASM2022v1_genomic         97.96           0.00   \n",
       "451   GCF_008000975.1_ASM800097v1_genomic         97.96           0.00   \n",
       "539  GCF_017504145.1_ASM1750414v1_genomic         97.96           0.00   \n",
       "580   GCF_001683795.1_ASM168379v1_genomic         91.84           4.76   \n",
       "\n",
       "                                        classification  \\\n",
       "356  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "451  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "539  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "580  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "                                 classification_method  sum_len      N50  \\\n",
       "356  taxonomic classification defined by topology a...  2664102  2664102   \n",
       "451  taxonomic classification defined by topology a...  2664043  2664043   \n",
       "539  taxonomic classification defined by topology a...  2664051  2664051   \n",
       "580  taxonomic classification defined by topology a...  3157849   127311   \n",
       "\n",
       "     16S ribosomal RNA > 1400 nt  Total_tRNAs                host  \\\n",
       "356                          3.0           54                 NaN   \n",
       "451                          3.0           54        homo sapiens   \n",
       "539                          3.0           54                 NaN   \n",
       "580                          NaN           42  reticulated python   \n",
       "\n",
       "                       NCBI_organism_name NCBI_infraspecific_name  \\\n",
       "356  Akkermansia muciniphila ATCC BAA-835     strain=ATCC BAA-835   \n",
       "451               Akkermansia muciniphila        strain=DSM 22959   \n",
       "539  Akkermansia muciniphila ATCC BAA-835     strain=ATCC BAA-835   \n",
       "580              Akkermansia glycaniphila              strain=Pyt   \n",
       "\n",
       "       relation_to_type_material type_of_sample  \n",
       "356  assembly from type material     metagenome  \n",
       "451  assembly from type material     metagenome  \n",
       "539  assembly from type material     metagenome  \n",
       "580  assembly from type material     metagenome  "
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_akk.loc[~merged_akk['relation_to_type_material'].isna()]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de746166",
   "metadata": {},
   "source": [
    "# Set `host`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "21a9f122",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 585 entries, 0 to 584\n",
      "Data columns (total 13 columns):\n",
      " #   Column                       Non-Null Count  Dtype  \n",
      "---  ------                       --------------  -----  \n",
      " 0   Genome                       585 non-null    object \n",
      " 1   Completeness                 585 non-null    float64\n",
      " 2   Contamination                585 non-null    float64\n",
      " 3   classification               585 non-null    object \n",
      " 4   classification_method        585 non-null    object \n",
      " 5   sum_len                      585 non-null    int64  \n",
      " 6   N50                          585 non-null    int64  \n",
      " 7   16S ribosomal RNA > 1400 nt  304 non-null    float64\n",
      " 8   Total_tRNAs                  585 non-null    int64  \n",
      " 9   host                         8 non-null      object \n",
      " 10  NCBI_organism_name           212 non-null    object \n",
      " 11  NCBI_infraspecific_name      195 non-null    object \n",
      " 12  relation_to_type_material    40 non-null     object \n",
      "dtypes: float64(3), int64(3), object(7)\n",
      "memory usage: 59.5+ KB\n",
      "Mouse                  3\n",
      "Choloepus hoffmanni    1\n",
      "Diceros bicornis       1\n",
      "Bufo bufo              1\n",
      "Horse                  1\n",
      "Gallus gallus          1\n",
      "Name: host, dtype: int64 577\n",
      "rg -woni -f /home/mibu/common_names_ncbi.list biosample_descr_akk.tsv | sort -h | uniq\n",
      "rg -woni -f /home/mibu/amniota_only_sp.list biosample_descr_akk.tsv | sort -h | uniq\n",
      "45\n",
      "index for the list in python: 177\n",
      "123\n",
      "index for the list in python: 172\n",
      "homo sapiens           113\n",
      "human                   17\n",
      "mus musculus             9\n",
      "mouse                    4\n",
      "rat                      3\n",
      "gallus gallus            2\n",
      "reticulated python       1\n",
      "horse                    1\n",
      "diceros bicornis         1\n",
      "siamang                  1\n",
      "choloepus hoffmanni      1\n",
      "bufo bufo                1\n",
      "Name: host, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>classification</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>sum_len</th>\n",
       "      <th>N50</th>\n",
       "      <th>16S ribosomal RNA &gt; 1400 nt</th>\n",
       "      <th>Total_tRNAs</th>\n",
       "      <th>host</th>\n",
       "      <th>NCBI_organism_name</th>\n",
       "      <th>NCBI_infraspecific_name</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DGYMR06203__metabat2_low_PE.047.contigs</td>\n",
       "      <td>91.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2417897</td>\n",
       "      <td>26278</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44</td>\n",
       "      <td>gallus gallus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>F157a_European_Toad__metabat2_high_PE.005.contigs</td>\n",
       "      <td>89.32</td>\n",
       "      <td>2.30</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2029501</td>\n",
       "      <td>8457</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40</td>\n",
       "      <td>bufo bufo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000155695.1_ASM15569v1_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>1.35</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>5775745</td>\n",
       "      <td>5514226</td>\n",
       "      <td>1.0</td>\n",
       "      <td>45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000242935.3_ASM24293v3_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.23</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7417673</td>\n",
       "      <td>7317842</td>\n",
       "      <td>1.0</td>\n",
       "      <td>52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000243495.3_ASM24349v3_genomic</td>\n",
       "      <td>98.63</td>\n",
       "      <td>0.23</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7073483</td>\n",
       "      <td>6196881</td>\n",
       "      <td>1.0</td>\n",
       "      <td>71</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>GCF_001683795.1_ASM168379v1_genomic</td>\n",
       "      <td>91.84</td>\n",
       "      <td>4.76</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>3157849</td>\n",
       "      <td>127311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>42</td>\n",
       "      <td>reticulated python</td>\n",
       "      <td>Akkermansia glycaniphila</td>\n",
       "      <td>strain=Pyt</td>\n",
       "      <td>assembly from type material</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>GCF_900097105.1_WK001_genomic</td>\n",
       "      <td>93.88</td>\n",
       "      <td>0.68</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2768865</td>\n",
       "      <td>1326</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Akkermansia glycaniphila</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>GCF_900184965.1_Mouse_2_MG.fasta_genomic</td>\n",
       "      <td>97.96</td>\n",
       "      <td>1.36</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2837147</td>\n",
       "      <td>181108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48</td>\n",
       "      <td>mouse</td>\n",
       "      <td>Akkermansia muciniphila</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>MGG23710</td>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2888964</td>\n",
       "      <td>250634</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48</td>\n",
       "      <td>mouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>584</th>\n",
       "      <td>MGG28571</td>\n",
       "      <td>96.60</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2937955</td>\n",
       "      <td>79240</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44</td>\n",
       "      <td>mouse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>585 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Genome  Completeness  \\\n",
       "0              DGYMR06203__metabat2_low_PE.047.contigs         91.12   \n",
       "1    F157a_European_Toad__metabat2_high_PE.005.contigs         89.32   \n",
       "2                   GCA_000155695.1_ASM15569v1_genomic        100.00   \n",
       "3                   GCA_000242935.3_ASM24293v3_genomic        100.00   \n",
       "4                   GCA_000243495.3_ASM24349v3_genomic         98.63   \n",
       "..                                                 ...           ...   \n",
       "580                GCF_001683795.1_ASM168379v1_genomic         91.84   \n",
       "581                      GCF_900097105.1_WK001_genomic         93.88   \n",
       "582           GCF_900184965.1_Mouse_2_MG.fasta_genomic         97.96   \n",
       "583                                           MGG23710         97.96   \n",
       "584                                           MGG28571         96.60   \n",
       "\n",
       "     Contamination                                     classification  \\\n",
       "0             0.00  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "1             2.30  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "2             1.35  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "3             0.23  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "4             0.23  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "..             ...                                                ...   \n",
       "580           4.76  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "581           0.68  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "582           1.36  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "583           0.00  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "584           0.00  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "                                 classification_method  sum_len      N50  \\\n",
       "0    taxonomic classification defined by topology a...  2417897    26278   \n",
       "1    taxonomic classification defined by topology a...  2029501     8457   \n",
       "2    taxonomic classification defined by topology a...  5775745  5514226   \n",
       "3    taxonomic classification defined by topology a...  7417673  7317842   \n",
       "4    taxonomic classification defined by topology a...  7073483  6196881   \n",
       "..                                                 ...      ...      ...   \n",
       "580  taxonomic classification defined by topology a...  3157849   127311   \n",
       "581  taxonomic classification defined by topology a...  2768865     1326   \n",
       "582  taxonomic classification defined by topology a...  2837147   181108   \n",
       "583  taxonomic classification defined by topology a...  2888964   250634   \n",
       "584  taxonomic classification defined by topology a...  2937955    79240   \n",
       "\n",
       "     16S ribosomal RNA > 1400 nt  Total_tRNAs                host  \\\n",
       "0                            NaN           44       gallus gallus   \n",
       "1                            NaN           40           bufo bufo   \n",
       "2                            1.0           45                 NaN   \n",
       "3                            1.0           52                 NaN   \n",
       "4                            1.0           71                 NaN   \n",
       "..                           ...          ...                 ...   \n",
       "580                          NaN           42  reticulated python   \n",
       "581                          0.0            0                 NaN   \n",
       "582                          NaN           48               mouse   \n",
       "583                          NaN           48               mouse   \n",
       "584                          NaN           44               mouse   \n",
       "\n",
       "           NCBI_organism_name NCBI_infraspecific_name  \\\n",
       "0                         NaN                     NaN   \n",
       "1                         NaN                     NaN   \n",
       "2                         NaN                     NaN   \n",
       "3                         NaN                     NaN   \n",
       "4                         NaN                     NaN   \n",
       "..                        ...                     ...   \n",
       "580  Akkermansia glycaniphila              strain=Pyt   \n",
       "581  Akkermansia glycaniphila                     NaN   \n",
       "582   Akkermansia muciniphila                     NaN   \n",
       "583                       NaN                     NaN   \n",
       "584                       NaN                     NaN   \n",
       "\n",
       "       relation_to_type_material  \n",
       "0                            NaN  \n",
       "1                            NaN  \n",
       "2                            NaN  \n",
       "3                            NaN  \n",
       "4                            NaN  \n",
       "..                           ...  \n",
       "580  assembly from type material  \n",
       "581                          NaN  \n",
       "582                          NaN  \n",
       "583                          NaN  \n",
       "584                          NaN  \n",
       "\n",
       "[585 rows x 13 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = pd.read_csv('/run/media/mibu/LR-orico_mini/Recovery_ADATA/Umayor/Work/Akker/Verrucomicrobia/QC/verrocu_seqs_summary.tsv', sep='\\t')\n",
    "merged_df.info()\n",
    "print(merged_df['host'].value_counts(), merged_df['host'].isna().sum())\n",
    "ln_comm_sp =  build_line_source('biosample_descr_akk.tsv', common_pats, file_pattern=True)\n",
    "ln_sp =  build_line_source('biosample_descr_akk.tsv', sp_pats, file_pattern=True)\n",
    "acc_comm_sp = build_acc_host(ln_comm_sp, 'biosample_descr_akk.tsv')\n",
    "acc_sp = build_acc_host(ln_sp, 'biosample_descr_akk.tsv')\n",
    "add_info_to_summary(merged_df, 'host', acc_comm_sp, acc_sp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "7e591af9",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df['host'] = merged_df['host'].str.lower()\n",
    "common_to_sp  = {'mice':'mus musculus','mouse':'mus musculus', 'human':'homo sapiens', 'jasper':'unknown'}\n",
    "merged_df['host'].replace(common_to_sp, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "6b076706",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "homo sapiens           130\n",
       "mus musculus            13\n",
       "rat                      3\n",
       "gallus gallus            2\n",
       "reticulated python       1\n",
       "horse                    1\n",
       "diceros bicornis         1\n",
       "siamang                  1\n",
       "choloepus hoffmanni      1\n",
       "bufo bufo                1\n",
       "Name: host, dtype: int64"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df['host'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c4f1ae7",
   "metadata": {},
   "source": [
    "# End `host`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f433f6f5",
   "metadata": {},
   "source": [
    "# Set `type_of_sample`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "cc6820dd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni 'culture|cultured|isolate|isolated|metagenome|monoisolate|uncultured' biosample_descr_akk.tsv | sort -h | uniq\n",
      "1:metagenome\n",
      "2:metagenome\n",
      "3:metagenome\n",
      "4:metagenome\n",
      "5:metagenome\n",
      "6:metagenome\n",
      "7:metagenome\n",
      "8:metagen\n",
      "80\n",
      "index for the list in python: 165\n",
      "{'uncultured', 'isolate', 'culture', 'metagenome'}\n"
     ]
    }
   ],
   "source": [
    "ln_culture =  build_line_source('biosample_descr_akk.tsv', \"'culture|cultured|isolate|isolated|metagenome|monoisolate|uncultured'\", \n",
    "                                file_pattern=False)\n",
    "print(ln_culture[:100])\n",
    "acc_culture = build_acc_host(ln_culture, 'biosample_descr_akk.tsv')\n",
    "print(set(acc_culture.values()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "a551717b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "culture       42\n",
      "metagenome    30\n",
      "isolate        5\n",
      "uncultured     3\n",
      "Name: type_of_sample, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>classification</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>sum_len</th>\n",
       "      <th>N50</th>\n",
       "      <th>16S ribosomal RNA &gt; 1400 nt</th>\n",
       "      <th>Total_tRNAs</th>\n",
       "      <th>host</th>\n",
       "      <th>NCBI_organism_name</th>\n",
       "      <th>NCBI_infraspecific_name</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>type_of_sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DGYMR06203__metabat2_low_PE.047.contigs</td>\n",
       "      <td>91.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2417897</td>\n",
       "      <td>26278</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44</td>\n",
       "      <td>gallus gallus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>F157a_European_Toad__metabat2_high_PE.005.contigs</td>\n",
       "      <td>89.32</td>\n",
       "      <td>2.30</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2029501</td>\n",
       "      <td>8457</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40</td>\n",
       "      <td>bufo bufo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000155695.1_ASM15569v1_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>1.35</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>5775745</td>\n",
       "      <td>5514226</td>\n",
       "      <td>1.0</td>\n",
       "      <td>45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000242935.3_ASM24293v3_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.23</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7417673</td>\n",
       "      <td>7317842</td>\n",
       "      <td>1.0</td>\n",
       "      <td>52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000243495.3_ASM24349v3_genomic</td>\n",
       "      <td>98.63</td>\n",
       "      <td>0.23</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7073483</td>\n",
       "      <td>6196881</td>\n",
       "      <td>1.0</td>\n",
       "      <td>71</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>GCF_001683795.1_ASM168379v1_genomic</td>\n",
       "      <td>91.84</td>\n",
       "      <td>4.76</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>3157849</td>\n",
       "      <td>127311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>42</td>\n",
       "      <td>reticulated python</td>\n",
       "      <td>Akkermansia glycaniphila</td>\n",
       "      <td>strain=Pyt</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>GCF_900097105.1_WK001_genomic</td>\n",
       "      <td>93.88</td>\n",
       "      <td>0.68</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2768865</td>\n",
       "      <td>1326</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Akkermansia glycaniphila</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>GCF_900184965.1_Mouse_2_MG.fasta_genomic</td>\n",
       "      <td>97.96</td>\n",
       "      <td>1.36</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2837147</td>\n",
       "      <td>181108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48</td>\n",
       "      <td>mus musculus</td>\n",
       "      <td>Akkermansia muciniphila</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>MGG23710</td>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2888964</td>\n",
       "      <td>250634</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48</td>\n",
       "      <td>mus musculus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>584</th>\n",
       "      <td>MGG28571</td>\n",
       "      <td>96.60</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2937955</td>\n",
       "      <td>79240</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44</td>\n",
       "      <td>mus musculus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>585 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Genome  Completeness  \\\n",
       "0              DGYMR06203__metabat2_low_PE.047.contigs         91.12   \n",
       "1    F157a_European_Toad__metabat2_high_PE.005.contigs         89.32   \n",
       "2                   GCA_000155695.1_ASM15569v1_genomic        100.00   \n",
       "3                   GCA_000242935.3_ASM24293v3_genomic        100.00   \n",
       "4                   GCA_000243495.3_ASM24349v3_genomic         98.63   \n",
       "..                                                 ...           ...   \n",
       "580                GCF_001683795.1_ASM168379v1_genomic         91.84   \n",
       "581                      GCF_900097105.1_WK001_genomic         93.88   \n",
       "582           GCF_900184965.1_Mouse_2_MG.fasta_genomic         97.96   \n",
       "583                                           MGG23710         97.96   \n",
       "584                                           MGG28571         96.60   \n",
       "\n",
       "     Contamination                                     classification  \\\n",
       "0             0.00  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "1             2.30  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "2             1.35  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "3             0.23  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "4             0.23  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "..             ...                                                ...   \n",
       "580           4.76  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "581           0.68  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "582           1.36  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "583           0.00  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "584           0.00  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "                                 classification_method  sum_len      N50  \\\n",
       "0    taxonomic classification defined by topology a...  2417897    26278   \n",
       "1    taxonomic classification defined by topology a...  2029501     8457   \n",
       "2    taxonomic classification defined by topology a...  5775745  5514226   \n",
       "3    taxonomic classification defined by topology a...  7417673  7317842   \n",
       "4    taxonomic classification defined by topology a...  7073483  6196881   \n",
       "..                                                 ...      ...      ...   \n",
       "580  taxonomic classification defined by topology a...  3157849   127311   \n",
       "581  taxonomic classification defined by topology a...  2768865     1326   \n",
       "582  taxonomic classification defined by topology a...  2837147   181108   \n",
       "583  taxonomic classification defined by topology a...  2888964   250634   \n",
       "584  taxonomic classification defined by topology a...  2937955    79240   \n",
       "\n",
       "     16S ribosomal RNA > 1400 nt  Total_tRNAs                host  \\\n",
       "0                            NaN           44       gallus gallus   \n",
       "1                            NaN           40           bufo bufo   \n",
       "2                            1.0           45                 NaN   \n",
       "3                            1.0           52                 NaN   \n",
       "4                            1.0           71                 NaN   \n",
       "..                           ...          ...                 ...   \n",
       "580                          NaN           42  reticulated python   \n",
       "581                          0.0            0                 NaN   \n",
       "582                          NaN           48        mus musculus   \n",
       "583                          NaN           48        mus musculus   \n",
       "584                          NaN           44        mus musculus   \n",
       "\n",
       "           NCBI_organism_name NCBI_infraspecific_name  \\\n",
       "0                         NaN                     NaN   \n",
       "1                         NaN                     NaN   \n",
       "2                         NaN                     NaN   \n",
       "3                         NaN                     NaN   \n",
       "4                         NaN                     NaN   \n",
       "..                        ...                     ...   \n",
       "580  Akkermansia glycaniphila              strain=Pyt   \n",
       "581  Akkermansia glycaniphila                     NaN   \n",
       "582   Akkermansia muciniphila                     NaN   \n",
       "583                       NaN                     NaN   \n",
       "584                       NaN                     NaN   \n",
       "\n",
       "       relation_to_type_material type_of_sample  \n",
       "0                            NaN            NaN  \n",
       "1                            NaN            NaN  \n",
       "2                            NaN            NaN  \n",
       "3                            NaN            NaN  \n",
       "4                            NaN            NaN  \n",
       "..                           ...            ...  \n",
       "580  assembly from type material            NaN  \n",
       "581                          NaN            NaN  \n",
       "582                          NaN            NaN  \n",
       "583                          NaN            NaN  \n",
       "584                          NaN            NaN  \n",
       "\n",
       "[585 rows x 14 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "add_info_to_summary(merged_df,'type_of_sample', acc_culture)\n",
    "# merged_df['type_of_sample'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "42069f39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(culture       47\n",
       " metagenome    33\n",
       " Name: type_of_sample, dtype: int64,\n",
       " 505)"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_to_sp  = {'isolate':'culture', 'uncultured':'metagenome'}\n",
    "merged_df['type_of_sample'].replace(common_to_sp, inplace=True)\n",
    "merged_df['type_of_sample'].value_counts(), merged_df['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b7fbafa",
   "metadata": {},
   "source": [
    "## Add `culture` to all the non-NAs values in `relation_to_type_material`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "1267bc59",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(culture       47\n",
       " metagenome    33\n",
       " Name: type_of_sample, dtype: int64,\n",
       " 121)"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_akk = merged_df[merged_df['classification'].str.contains('g__Akkermansia')].copy()\n",
    "merged_akk['type_of_sample'].value_counts(), merged_akk['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "1cb4de3d",
   "metadata": {},
   "outputs": [],
   "source": [
    "mask = ~merged_akk['relation_to_type_material'].isna()\n",
    "merged_akk.loc[mask, 'type_of_sample'] = 'culture'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "831f94d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>classification</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>sum_len</th>\n",
       "      <th>N50</th>\n",
       "      <th>16S ribosomal RNA &gt; 1400 nt</th>\n",
       "      <th>Total_tRNAs</th>\n",
       "      <th>host</th>\n",
       "      <th>NCBI_organism_name</th>\n",
       "      <th>NCBI_infraspecific_name</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>type_of_sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>GCF_000020225.1_ASM2022v1_genomic</td>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2664102</td>\n",
       "      <td>2664102</td>\n",
       "      <td>3.0</td>\n",
       "      <td>54</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Akkermansia muciniphila ATCC BAA-835</td>\n",
       "      <td>strain=ATCC BAA-835</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>culture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>451</th>\n",
       "      <td>GCF_008000975.1_ASM800097v1_genomic</td>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2664043</td>\n",
       "      <td>2664043</td>\n",
       "      <td>3.0</td>\n",
       "      <td>54</td>\n",
       "      <td>homo sapiens</td>\n",
       "      <td>Akkermansia muciniphila</td>\n",
       "      <td>strain=DSM 22959</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>culture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>GCF_017504145.1_ASM1750414v1_genomic</td>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2664051</td>\n",
       "      <td>2664051</td>\n",
       "      <td>3.0</td>\n",
       "      <td>54</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Akkermansia muciniphila ATCC BAA-835</td>\n",
       "      <td>strain=ATCC BAA-835</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>culture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>GCF_001683795.1_ASM168379v1_genomic</td>\n",
       "      <td>91.84</td>\n",
       "      <td>4.76</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>3157849</td>\n",
       "      <td>127311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>42</td>\n",
       "      <td>reticulated python</td>\n",
       "      <td>Akkermansia glycaniphila</td>\n",
       "      <td>strain=Pyt</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>culture</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Genome  Completeness  Contamination  \\\n",
       "356     GCF_000020225.1_ASM2022v1_genomic         97.96           0.00   \n",
       "451   GCF_008000975.1_ASM800097v1_genomic         97.96           0.00   \n",
       "539  GCF_017504145.1_ASM1750414v1_genomic         97.96           0.00   \n",
       "580   GCF_001683795.1_ASM168379v1_genomic         91.84           4.76   \n",
       "\n",
       "                                        classification  \\\n",
       "356  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "451  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "539  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "580  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "                                 classification_method  sum_len      N50  \\\n",
       "356  taxonomic classification defined by topology a...  2664102  2664102   \n",
       "451  taxonomic classification defined by topology a...  2664043  2664043   \n",
       "539  taxonomic classification defined by topology a...  2664051  2664051   \n",
       "580  taxonomic classification defined by topology a...  3157849   127311   \n",
       "\n",
       "     16S ribosomal RNA > 1400 nt  Total_tRNAs                host  \\\n",
       "356                          3.0           54                 NaN   \n",
       "451                          3.0           54        homo sapiens   \n",
       "539                          3.0           54                 NaN   \n",
       "580                          NaN           42  reticulated python   \n",
       "\n",
       "                       NCBI_organism_name NCBI_infraspecific_name  \\\n",
       "356  Akkermansia muciniphila ATCC BAA-835     strain=ATCC BAA-835   \n",
       "451               Akkermansia muciniphila        strain=DSM 22959   \n",
       "539  Akkermansia muciniphila ATCC BAA-835     strain=ATCC BAA-835   \n",
       "580              Akkermansia glycaniphila              strain=Pyt   \n",
       "\n",
       "       relation_to_type_material type_of_sample  \n",
       "356  assembly from type material        culture  \n",
       "451  assembly from type material        culture  \n",
       "539  assembly from type material        culture  \n",
       "580  assembly from type material        culture  "
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_akk.loc[~merged_akk['relation_to_type_material'].isna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "335a2876",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_akk.to_csv(qc_path.joinpath('merged_akk.tsv'), sep='\\t', index=False)c"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a12215ea",
   "metadata": {},
   "source": [
    "# End `type_of_sample`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4bc557ba",
   "metadata": {},
   "source": [
    "# Which asms are in the 'indicative' but no in the 'culture dict' ?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "85ed492b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni 'stool|feces|gut|fecal|biome|microbiome' biosample_descr_akk.tsv | sort -h | uniq\n",
      "1:gut\n",
      "2:gut\n",
      "3:gut\n",
      "4:gut\n",
      "5:gut\n",
      "6:gut\n",
      "8:biome\n",
      "8:feces\n",
      "8:gut\n",
      "9:biome\n",
      "9:feces\n",
      "9:gut\n",
      "10:feces\n",
      "11:feces\n",
      "11\n",
      "163\n",
      "index for the list in python: 177\n"
     ]
    }
   ],
   "source": [
    "ln_meta_indicative =  build_line_source('biosample_descr_akk.tsv', \"'stool|feces|gut|fecal|biome|microbiome'\", \n",
    "                                file_pattern=False)\n",
    "print(ln_meta_indicative[:100])\n",
    "acc_meta_indicative = build_acc_host(ln_meta_indicative, 'biosample_descr_akk.tsv')\n",
    "# acc_meta_indicative"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "416779ef",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "({'GCA_003514565.1', 'GCF_004319565.1'},\n",
       " {'GCF_001580195.1',\n",
       "  'GCF_001683795.1',\n",
       "  'GCF_002884915.1',\n",
       "  'GCF_002884995.1',\n",
       "  'GCF_002885015.1',\n",
       "  'GCF_002885025.1',\n",
       "  'GCF_002885055.1',\n",
       "  'GCF_002885075.1',\n",
       "  'GCF_002885155.1',\n",
       "  'GCF_002885165.1',\n",
       "  'GCF_002885195.1',\n",
       "  'GCF_002885215.1',\n",
       "  'GCF_002885235.1',\n",
       "  'GCF_002885255.1',\n",
       "  'GCF_002885295.1',\n",
       "  'GCF_002885315.1',\n",
       "  'GCF_002885335.1',\n",
       "  'GCF_002885355.1',\n",
       "  'GCF_002885375.1',\n",
       "  'GCF_002885395.1',\n",
       "  'GCF_002885415.1',\n",
       "  'GCF_002885425.1',\n",
       "  'GCF_002885455.1',\n",
       "  'GCF_002885465.1',\n",
       "  'GCF_002885515.1',\n",
       "  'GCF_002885535.1',\n",
       "  'GCF_002885575.1',\n",
       "  'GCF_002885595.1',\n",
       "  'GCF_002885615.1',\n",
       "  'GCF_002885625.1',\n",
       "  'GCF_002885655.1',\n",
       "  'GCF_002885675.1',\n",
       "  'GCF_002885695.1',\n",
       "  'GCF_002885715.1',\n",
       "  'GCF_004104435.1',\n",
       "  'GCF_004168105.1',\n",
       "  'GCF_008422275.1',\n",
       "  'GCF_008422615.1',\n",
       "  'GCF_008422635.1',\n",
       "  'GCF_008422665.1',\n",
       "  'GCF_008422685.1',\n",
       "  'GCF_008422695.1',\n",
       "  'GCF_008422705.1',\n",
       "  'GCF_008422715.1',\n",
       "  'GCF_008422765.1',\n",
       "  'GCF_008422775.1',\n",
       "  'GCF_008422795.1',\n",
       "  'GCF_008422805.1',\n",
       "  'GCF_008422815.1',\n",
       "  'GCF_008422865.1',\n",
       "  'GCF_008422875.1',\n",
       "  'GCF_008422885.1',\n",
       "  'GCF_008422895.1',\n",
       "  'GCF_008422925.1',\n",
       "  'GCF_008422965.1',\n",
       "  'GCF_008422985.1',\n",
       "  'GCF_008422995.1',\n",
       "  'GCF_008423015.1',\n",
       "  'GCF_008423035.1',\n",
       "  'GCF_008423065.1',\n",
       "  'GCF_008423085.1',\n",
       "  'GCF_008423095.1',\n",
       "  'GCF_008423115.1',\n",
       "  'GCF_008423135.1',\n",
       "  'GCF_008423175.1',\n",
       "  'GCF_008423185.1',\n",
       "  'GCF_008423215.1',\n",
       "  'GCF_008423245.1',\n",
       "  'GCF_008423265.1',\n",
       "  'GCF_008423275.1',\n",
       "  'GCF_008423295.1',\n",
       "  'GCF_008423325.1',\n",
       "  'GCF_008423335.1',\n",
       "  'GCF_008423365.1',\n",
       "  'GCF_008423375.1',\n",
       "  'GCF_008423385.1',\n",
       "  'GCF_008423405.1',\n",
       "  'GCF_008423445.1',\n",
       "  'GCF_008423455.1',\n",
       "  'GCF_016697425.1',\n",
       "  'GCF_017566105.1',\n",
       "  'GCF_017570425.1',\n",
       "  'GCF_017570485.1',\n",
       "  'GCF_017570505.1',\n",
       "  'GCF_017570525.1'})"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(acc_culture.keys()) - set(acc_meta_indicative.keys()), set(acc_meta_indicative.keys()) - set(acc_culture.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "82dec6ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(metagenome    74\n",
       " culture       50\n",
       " Name: type_of_sample, dtype: int64,\n",
       " 77)"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_akk['type_of_sample'].value_counts(), merged_akk['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28c67a9a",
   "metadata": {},
   "source": [
    "## Get all the asms that aren't `culture` but are \"indicative\" and set them to `metagenome`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "b8d87a52",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['GCA_000436395.1', 'GCA_000437075.1', 'GCA_001917295.1',\n",
       "       'GCA_002362435.1', 'GCA_002492685.1', 'GCA_002493465.1',\n",
       "       'GCA_003514565.1', 'GCA_008672035.1', 'GCA_008672205.1',\n",
       "       'GCA_009773845.1'], dtype=object)"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acc_not_culture = (merged_akk[~(merged_akk['type_of_sample'] == 'culture')]['Genome'].\n",
    "                   str.extract( '(GC[AF]_[0-9]{9}\\.[1-9])').dropna().values.flatten())\n",
    "acc_not_culture[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "24c1e001",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(116, 163)"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acc_meta_indicat_notcult = {asm:source for asm,source in acc_meta_indicative.items() if asm in acc_not_culture}\n",
    "len(acc_meta_indicat_notcult), len(acc_meta_indicative)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "facee62b",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'GCA_000436395.1': 'gut',\n",
       " 'GCA_000437075.1': 'gut',\n",
       " 'GCA_001917295.1': 'gut',\n",
       " 'GCA_002362435.1': 'gut',\n",
       " 'GCA_002492685.1': 'gut',\n",
       " 'GCA_002493465.1': 'gut',\n",
       " 'GCA_008672035.1': 'gut',\n",
       " 'GCA_008672205.1': 'gut',\n",
       " 'GCA_009773845.1': 'feces',\n",
       " 'GCA_014871965.1': 'gut',\n",
       " 'GCA_014871995.1': 'gut',\n",
       " 'GCA_014872005.1': 'gut',\n",
       " 'GCA_014872035.1': 'gut',\n",
       " 'GCA_014872065.1': 'gut',\n",
       " 'GCA_014872075.1': 'gut',\n",
       " 'GCA_014872105.1': 'gut',\n",
       " 'GCA_014872125.1': 'gut',\n",
       " 'GCA_014872165.1': 'gut',\n",
       " 'GCA_900539575.1': 'gut',\n",
       " 'GCA_900539895.1': 'gut',\n",
       " 'GCA_900545155.1': 'gut',\n",
       " 'GCA_900752515.1': 'stool',\n",
       " 'GCA_900759685.1': 'stool',\n",
       " 'GCA_900765745.1': 'stool',\n",
       " 'GCA_905193715.1': 'gut',\n",
       " 'GCA_905197035.1': 'gut',\n",
       " 'GCA_905200125.1': 'gut',\n",
       " 'GCA_905200945.1': 'gut',\n",
       " 'GCA_905205055.1': 'gut',\n",
       " 'GCA_905205915.1': 'gut',\n",
       " 'GCA_905210405.1': 'gut',\n",
       " 'GCA_905210615.1': 'gut',\n",
       " 'GCF_001580195.1': 'feces',\n",
       " 'GCF_002884915.1': 'gut',\n",
       " 'GCF_002884995.1': 'gut',\n",
       " 'GCF_002885015.1': 'gut',\n",
       " 'GCF_002885025.1': 'gut',\n",
       " 'GCF_002885055.1': 'gut',\n",
       " 'GCF_002885075.1': 'gut',\n",
       " 'GCF_002885155.1': 'gut',\n",
       " 'GCF_002885165.1': 'gut',\n",
       " 'GCF_002885195.1': 'gut',\n",
       " 'GCF_002885215.1': 'gut',\n",
       " 'GCF_002885235.1': 'gut',\n",
       " 'GCF_002885255.1': 'gut',\n",
       " 'GCF_002885295.1': 'gut',\n",
       " 'GCF_002885315.1': 'gut',\n",
       " 'GCF_002885335.1': 'gut',\n",
       " 'GCF_002885355.1': 'gut',\n",
       " 'GCF_002885375.1': 'gut',\n",
       " 'GCF_002885395.1': 'gut',\n",
       " 'GCF_002885415.1': 'gut',\n",
       " 'GCF_002885425.1': 'gut',\n",
       " 'GCF_002885455.1': 'gut',\n",
       " 'GCF_002885465.1': 'gut',\n",
       " 'GCF_002885515.1': 'gut',\n",
       " 'GCF_002885535.1': 'gut',\n",
       " 'GCF_002885575.1': 'gut',\n",
       " 'GCF_002885595.1': 'gut',\n",
       " 'GCF_002885615.1': 'gut',\n",
       " 'GCF_002885625.1': 'gut',\n",
       " 'GCF_002885655.1': 'gut',\n",
       " 'GCF_002885675.1': 'gut',\n",
       " 'GCF_002885695.1': 'gut',\n",
       " 'GCF_002885715.1': 'gut',\n",
       " 'GCF_004104435.1': 'gut',\n",
       " 'GCF_004168105.1': 'stool',\n",
       " 'GCF_008422275.1': 'fecal',\n",
       " 'GCF_008422615.1': 'fecal',\n",
       " 'GCF_008422635.1': 'fecal',\n",
       " 'GCF_008422665.1': 'fecal',\n",
       " 'GCF_008422685.1': 'fecal',\n",
       " 'GCF_008422695.1': 'fecal',\n",
       " 'GCF_008422705.1': 'fecal',\n",
       " 'GCF_008422715.1': 'fecal',\n",
       " 'GCF_008422765.1': 'fecal',\n",
       " 'GCF_008422775.1': 'fecal',\n",
       " 'GCF_008422795.1': 'fecal',\n",
       " 'GCF_008422805.1': 'fecal',\n",
       " 'GCF_008422815.1': 'fecal',\n",
       " 'GCF_008422865.1': 'fecal',\n",
       " 'GCF_008422875.1': 'fecal',\n",
       " 'GCF_008422885.1': 'fecal',\n",
       " 'GCF_008422895.1': 'fecal',\n",
       " 'GCF_008422925.1': 'fecal',\n",
       " 'GCF_008422965.1': 'fecal',\n",
       " 'GCF_008422985.1': 'fecal',\n",
       " 'GCF_008422995.1': 'fecal',\n",
       " 'GCF_008423015.1': 'fecal',\n",
       " 'GCF_008423035.1': 'fecal',\n",
       " 'GCF_008423065.1': 'fecal',\n",
       " 'GCF_008423085.1': 'fecal',\n",
       " 'GCF_008423095.1': 'fecal',\n",
       " 'GCF_008423115.1': 'fecal',\n",
       " 'GCF_008423135.1': 'fecal',\n",
       " 'GCF_008423175.1': 'fecal',\n",
       " 'GCF_008423185.1': 'fecal',\n",
       " 'GCF_008423215.1': 'fecal',\n",
       " 'GCF_008423245.1': 'fecal',\n",
       " 'GCF_008423265.1': 'fecal',\n",
       " 'GCF_008423275.1': 'fecal',\n",
       " 'GCF_008423295.1': 'fecal',\n",
       " 'GCF_008423325.1': 'fecal',\n",
       " 'GCF_008423335.1': 'fecal',\n",
       " 'GCF_008423365.1': 'fecal',\n",
       " 'GCF_008423375.1': 'fecal',\n",
       " 'GCF_008423385.1': 'fecal',\n",
       " 'GCF_008423405.1': 'fecal',\n",
       " 'GCF_008423445.1': 'fecal',\n",
       " 'GCF_008423455.1': 'fecal',\n",
       " 'GCF_016697425.1': 'gut',\n",
       " 'GCF_017566105.1': 'stool',\n",
       " 'GCF_017570425.1': 'gut',\n",
       " 'GCF_017570485.1': 'gut',\n",
       " 'GCF_017570505.1': 'gut',\n",
       " 'GCF_017570525.1': 'gut'}"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acc_meta_indicat_notcult"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "524e0d99",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gut           66\n",
      "culture       50\n",
      "fecal         43\n",
      "stool          5\n",
      "feces          2\n",
      "metagenome     1\n",
      "Name: type_of_sample, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>classification</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>sum_len</th>\n",
       "      <th>N50</th>\n",
       "      <th>16S ribosomal RNA &gt; 1400 nt</th>\n",
       "      <th>Total_tRNAs</th>\n",
       "      <th>host</th>\n",
       "      <th>NCBI_organism_name</th>\n",
       "      <th>NCBI_infraspecific_name</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>type_of_sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DGYMR06203__metabat2_low_PE.047.contigs</td>\n",
       "      <td>91.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2417897</td>\n",
       "      <td>26278</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44</td>\n",
       "      <td>gallus gallus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>F157a_European_Toad__metabat2_high_PE.005.contigs</td>\n",
       "      <td>89.32</td>\n",
       "      <td>2.30</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2029501</td>\n",
       "      <td>8457</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40</td>\n",
       "      <td>bufo bufo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCA_000436395.1_MGS154_genomic</td>\n",
       "      <td>95.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2721828</td>\n",
       "      <td>74342</td>\n",
       "      <td>NaN</td>\n",
       "      <td>46</td>\n",
       "      <td>homo sapiens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gut</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCA_000437075.1_MGS344_genomic</td>\n",
       "      <td>97.25</td>\n",
       "      <td>1.36</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>3024253</td>\n",
       "      <td>170254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>homo sapiens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gut</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCA_001917295.1_ASM191729v1_genomic</td>\n",
       "      <td>97.28</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2966498</td>\n",
       "      <td>183307</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47</td>\n",
       "      <td>homo sapiens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gut</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>GCF_001683795.1_ASM168379v1_genomic</td>\n",
       "      <td>91.84</td>\n",
       "      <td>4.76</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>3157849</td>\n",
       "      <td>127311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>42</td>\n",
       "      <td>reticulated python</td>\n",
       "      <td>Akkermansia glycaniphila</td>\n",
       "      <td>strain=Pyt</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>culture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>GCF_900097105.1_WK001_genomic</td>\n",
       "      <td>93.88</td>\n",
       "      <td>0.68</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2768865</td>\n",
       "      <td>1326</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Akkermansia glycaniphila</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>GCF_900184965.1_Mouse_2_MG.fasta_genomic</td>\n",
       "      <td>97.96</td>\n",
       "      <td>1.36</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2837147</td>\n",
       "      <td>181108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48</td>\n",
       "      <td>mus musculus</td>\n",
       "      <td>Akkermansia muciniphila</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>MGG23710</td>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2888964</td>\n",
       "      <td>250634</td>\n",
       "      <td>NaN</td>\n",
       "      <td>48</td>\n",
       "      <td>mus musculus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>584</th>\n",
       "      <td>MGG28571</td>\n",
       "      <td>96.60</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2937955</td>\n",
       "      <td>79240</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44</td>\n",
       "      <td>mus musculus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>201 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Genome  Completeness  \\\n",
       "0              DGYMR06203__metabat2_low_PE.047.contigs         91.12   \n",
       "1    F157a_European_Toad__metabat2_high_PE.005.contigs         89.32   \n",
       "5                       GCA_000436395.1_MGS154_genomic         95.07   \n",
       "6                       GCA_000437075.1_MGS344_genomic         97.25   \n",
       "13                 GCA_001917295.1_ASM191729v1_genomic         97.28   \n",
       "..                                                 ...           ...   \n",
       "580                GCF_001683795.1_ASM168379v1_genomic         91.84   \n",
       "581                      GCF_900097105.1_WK001_genomic         93.88   \n",
       "582           GCF_900184965.1_Mouse_2_MG.fasta_genomic         97.96   \n",
       "583                                           MGG23710         97.96   \n",
       "584                                           MGG28571         96.60   \n",
       "\n",
       "     Contamination                                     classification  \\\n",
       "0             0.00  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "1             2.30  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "5             0.00  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "6             1.36  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "13            0.00  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "..             ...                                                ...   \n",
       "580           4.76  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "581           0.68  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "582           1.36  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "583           0.00  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "584           0.00  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "                                 classification_method  sum_len     N50  \\\n",
       "0    taxonomic classification defined by topology a...  2417897   26278   \n",
       "1    taxonomic classification defined by topology a...  2029501    8457   \n",
       "5    taxonomic classification defined by topology a...  2721828   74342   \n",
       "6    taxonomic classification defined by topology a...  3024253  170254   \n",
       "13   taxonomic classification defined by topology a...  2966498  183307   \n",
       "..                                                 ...      ...     ...   \n",
       "580  taxonomic classification defined by topology a...  3157849  127311   \n",
       "581  taxonomic classification defined by topology a...  2768865    1326   \n",
       "582  taxonomic classification defined by topology a...  2837147  181108   \n",
       "583  taxonomic classification defined by topology a...  2888964  250634   \n",
       "584  taxonomic classification defined by topology a...  2937955   79240   \n",
       "\n",
       "     16S ribosomal RNA > 1400 nt  Total_tRNAs                host  \\\n",
       "0                            NaN           44       gallus gallus   \n",
       "1                            NaN           40           bufo bufo   \n",
       "5                            NaN           46        homo sapiens   \n",
       "6                            NaN           50        homo sapiens   \n",
       "13                           NaN           47        homo sapiens   \n",
       "..                           ...          ...                 ...   \n",
       "580                          NaN           42  reticulated python   \n",
       "581                          0.0            0                 NaN   \n",
       "582                          NaN           48        mus musculus   \n",
       "583                          NaN           48        mus musculus   \n",
       "584                          NaN           44        mus musculus   \n",
       "\n",
       "           NCBI_organism_name NCBI_infraspecific_name  \\\n",
       "0                         NaN                     NaN   \n",
       "1                         NaN                     NaN   \n",
       "5                         NaN                     NaN   \n",
       "6                         NaN                     NaN   \n",
       "13                        NaN                     NaN   \n",
       "..                        ...                     ...   \n",
       "580  Akkermansia glycaniphila              strain=Pyt   \n",
       "581  Akkermansia glycaniphila                     NaN   \n",
       "582   Akkermansia muciniphila                     NaN   \n",
       "583                       NaN                     NaN   \n",
       "584                       NaN                     NaN   \n",
       "\n",
       "       relation_to_type_material type_of_sample  \n",
       "0                            NaN            NaN  \n",
       "1                            NaN            NaN  \n",
       "5                            NaN            gut  \n",
       "6                            NaN            gut  \n",
       "13                           NaN            gut  \n",
       "..                           ...            ...  \n",
       "580  assembly from type material        culture  \n",
       "581                          NaN            NaN  \n",
       "582                          NaN            NaN  \n",
       "583                          NaN            NaN  \n",
       "584                          NaN            NaN  \n",
       "\n",
       "[201 rows x 14 columns]"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "add_info_to_summary(merged_akk,'type_of_sample', acc_meta_indicat_notcult)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "b779f52c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(gut           66\n",
       " culture       50\n",
       " fecal         43\n",
       " stool          5\n",
       " feces          2\n",
       " metagenome     1\n",
       " Name: type_of_sample, dtype: int64,\n",
       " 34)"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_akk['type_of_sample'].value_counts(), merged_akk['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "30db8aba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "metagenome    117\n",
       "culture        50\n",
       "Name: type_of_sample, dtype: int64"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_to_sp  = {'gut':'metagenome', 'feces':'metagenome','fecal':'metagenome',\n",
    "                 'stool':'metagenome'}\n",
    "merged_akk['type_of_sample'].replace(common_to_sp, inplace=True)\n",
    "merged_akk['type_of_sample'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "d9c30efd",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_akk.to_csv(qc_path.joinpath('merged_akk.tsv'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9bc2fca2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "2fa4482f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[34, 35, 37, 38, 39, 40, 41, 42, 44, 45, 46, 47, 48, 50, 51, 52, 53, 54, 55, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 96, 97, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180] 100\n"
     ]
    }
   ],
   "source": [
    "# build_acc_host(line_biome_all, sample_desc_all)\n",
    "ln = {int(el.split(':')[0]) for el in ln_culture.strip().split('\\n')}\n",
    "ln_notpresent = [el for el in range(1,181) if el not in ln]\n",
    "print(ln_notpresent, len(ln_notpresent))\n",
    "# for el in line_biome_all.split('\\n'):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "634cad53",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "180 181\n"
     ]
    }
   ],
   "source": [
    "with open('biosample_descr_akk.tsv', 'r') as fh:\n",
    "    biosample_akk_desc = fh.readlines()\n",
    "\n",
    "with open('bioproject_descr_akk.tsv', 'r') as fh:\n",
    "    bioproj_akk_desc = fh.readlines()\n",
    "print(len(biosample_akk_desc), len(bioproj_akk_desc))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "1bf6b5ce",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCF_000020225.1\tSAMN00138213\tATCC BAA-835\n",
      "\n",
      "GCF_000723745.2\tSAMEA3139045\tGCA_000723745\tEuropean Nucleotide Archive\tERC000011\tEBI\tEuropean Bioinformatics Institute\t2014-11-20T12:20:15Z\t2014-11-20T11:56:31Z\tpublic\tERS610980\tERS610980\tBioSample entry for genome collection GCA_000723745\tUrmite\n",
      "\n",
      "GCF_001580195.1\tSAMN03956447\tfeces\tHomo sapiens\tKLE1798\n",
      "\n",
      "GCF_001647615.1\tSAMN04606485\tGI tract\tKLE1605\tHomo sapiens\tHigh-Quality Draft\tHigh-Quality Draft\tVelvet v. 1.1.04\t128x\tIllumina\n",
      "\n",
      "GCF_001688765.2\tSAMN04621615\tYL44\tMus musculus\t2015-02-26\tSwitzerland: Bern\twhole organism\n",
      "\n",
      "GCF_002161325.1\tSAMN06473757\thealthy\tmissing\tnot applicable\tmissing\tnot applicable\tGallus gallus\tnot applicable\tnot applicable\tcommensal\tcaecum\tAn78\tanaerobe\tcaecum\t2016\tnot applicable\tmissing\n",
      "\n",
      "GCF_002201495.1\tSAMN03854119\tYL44\tSwitzerland: Zurich\tnot applicable\tnot applicable\tnot applicable\torganic feature\tintestine environment\t47 N 8 E\torganic material\tnot applicable\tMus musculus\tDSM:26127\n",
      "\n",
      "GCF_002884915.1\tSAMN08162557\tGP24\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002884995.1\tSAMN08162554\tGP30\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885015.1\tSAMN08162551\tGP12\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885025.1\tSAMN08162552\tGP15\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885055.1\tSAMN08162550\tGP11\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885075.1\tSAMN08162548\tGP08\tMus musculus\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885155.1\tSAMN08162544\tGP43\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885165.1\tSAMN08162543\tGP41\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885195.1\tSAMN08162540\tGP36\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885215.1\tSAMN08162539\tGP29\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885235.1\tSAMN08162538\tGP21\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885255.1\tSAMN08162534\tGP03\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885295.1\tSAMN08162531\tGP35\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885315.1\tSAMN08162529\tGP27\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885335.1\tSAMN08162528\tGP26\tMus musculus\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885355.1\tSAMN08162527\tGP23\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885375.1\tSAMN08162526\tGP20\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885395.1\tSAMN08162525\tGP19\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885415.1\tSAMN08162524\tGP16\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885425.1\tSAMN08162523\tGP14\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885455.1\tSAMN08162522\tGP13\tMus musculus\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885465.1\tSAMN08162520\tGP02\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885515.1\tSAMN08162555\tGP42\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885535.1\tSAMN08162553\tGP25\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885575.1\tSAMN08162542\tGP39\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885595.1\tSAMN08162541\tGP38\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885615.1\tSAMN08162537\tGP06\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885625.1\tSAMN08162536\tGP05\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885655.1\tSAMN08162535\tGP04\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885675.1\tSAMN08162532\tGP40\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885695.1\tSAMN08162530\tGP28\tHomo sapiens\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_002885715.1\tSAMN08162521\tGP09\tMus musculus\tgut microbiota\t2015\tChina\tassembly\n",
      "\n",
      "GCF_004104435.1\tSAMN10186677\tCBA5201\tmissing\tmissing\tmissing\thuman-gut\tmissing\tHomo sapiens\tmissing\tmissing\tmissing\tmissing\thuman feces\n",
      "\n",
      "GCF_004168105.1\tSAMN10239579\tam_0171\tHomo sapiens\tstool\t2015-01\tUSA: Cambridge\tOpenBiome Stol\n",
      "\n",
      "GCF_008422275.1\tSAMN11943062\tBIOML-A17\t2014-07-04\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422615.1\tSAMN11943045\tBIOML-A45\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422635.1\tSAMN11943044\tBIOML-A44\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422665.1\tSAMN11943043\tBIOML-A43\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422685.1\tSAMN11943039\tBIOML-A39\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422695.1\tSAMN11943041\tBIOML-A41\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422705.1\tSAMN11943040\tBIOML-A40\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422715.1\tSAMN11943042\tBIOML-A42\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422765.1\tSAMN11943036\tBIOML-A36\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422775.1\tSAMN11943037\tBIOML-A37\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422795.1\tSAMN11943035\tBIOML-A35\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422805.1\tSAMN11943034\tBIOML-A34\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422815.1\tSAMN11943038\tBIOML-A38\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422865.1\tSAMN11943032\tBIOML-A32\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422875.1\tSAMN11943030\tBIOML-A30\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422885.1\tSAMN11943033\tBIOML-A33\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422895.1\tSAMN11943031\tBIOML-A31\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422925.1\tSAMN11943029\tBIOML-A29\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422965.1\tSAMN11943027\tBIOML-A27\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422985.1\tSAMN11943026\tBIOML-A26\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008422995.1\tSAMN11943028\tBIOML-A28\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423015.1\tSAMN11943024\tBIOML-A24\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423035.1\tSAMN11943025\tBIOML-A25\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423065.1\tSAMN11943023\tBIOML-A23\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423085.1\tSAMN11943021\tBIOML-A21\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423095.1\tSAMN11943022\tBIOML-A22\t2016-05-12\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423115.1\tSAMN11943020\tBIOML-A20\t2016-01-11\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423135.1\tSAMN11943019\tBIOML-A19\t2016-01-11\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423175.1\tSAMN11943018\tBIOML-A18\t2016-01-11\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423185.1\tSAMN11943016\tBIOML-A16\t2016-01-11\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423215.1\tSAMN11943015\tBIOML-A15\t2016-01-11\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423245.1\tSAMN11943014\tBIOML-A14\t2016-01-11\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423265.1\tSAMN11943013\tBIOML-A13\t2016-01-11\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423275.1\tSAMN11943011\tBIOML-A11\t2016-01-11\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423295.1\tSAMN11943009\tBIOML-A9\t2016-01-11\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423325.1\tSAMN11943008\tBIOML-A8\t2016-01-11\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423335.1\tSAMN11943007\tBIOML-A7\t2016-01-11\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423365.1\tSAMN11943006\tBIOML-A6\t2014-12-18\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423375.1\tSAMN11943004\tBIOML-A4\t2014-12-18\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423385.1\tSAMN11943003\tBIOML-A3\t2014-12-18\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423405.1\tSAMN11943005\tBIOML-A5\t2014-12-18\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423445.1\tSAMN11943002\tBIOML-A2\t2014-12-18\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_008423455.1\tSAMN11943001\tBIOML-A1\t2014-12-18\tbodily fluid material biome [ENVO:02000019]\texcreta material [ENVO:02000022]\tfecal material [ENVO:00002003]\tUSA:Boston\tHomo sapiens\tmissing\t42.36 N 71.06 W\tmissing\tmissing\tfecal material\n",
      "\n",
      "GCF_009663785.1\tSAMN10240011\tD10-15\t2016-06-28\tintestine enviorment\tintestine\tanimal manure\tChina:Beijing\tob/ob mouse\tmGAM medium at 37 centigrade\t39.59 N 116.23 E\t1\tno applicable\n",
      "\n",
      "GCF_009731575.1\tSAMD00192834\tAkkermansia_muciniphila_JCM30893\tAKMU\tJCM 30893\tmissing\tmissing\tmissing\tmissing\tmissing\tmissing\tA_muchiniphila_JCM30893\tmissing\tmissing\tmissing\tHomo sapiens\n",
      "\n",
      "GCF_016697425.1\tSAMN16911570\tYL44\t2019-09\tmice microbiota\tfeces\tmice gut\tFrance:Paris\tMus musculus\t29051233\tnot applicable\t1\tOMM12\n",
      "\n",
      "GCF_017504145.1\tSAMN00138213\tATCC BAA-835\n",
      "\n",
      "GCF_017566105.1\tSAMN14738317\tGGCC_0220\tfeces\tHomo sapiens\t2019-02\tUSA: North Carolina\tstool sample\n",
      "\n",
      "GCF_017570425.1\tSAMN18309827\tKGMB02009\t2017-05-18\thuman\tgut\tfeces\tSouth Korea:Gyeonggi-do, Seongnam-si\tHomo sapiens\tTSA supplemented with blood\t37.35 N 127.15 E\t1\t10.1099/ijs.0.02873-4\n",
      "\n",
      "GCF_017570485.1\tSAMN18309825\tKGMB01989\t2017-05-18\thuman\tgut\tfeces\tSouth Korea:Gyeonggi-do, Seongnam-si\tHomo sapiens\tTSA supplemented with blood\t37.35 N 127.13 E\t1\t10.1099/ijs.0.02873-2\n",
      "\n",
      "GCF_017570505.1\tSAMN18309826\tKGMB01990\t2017-05-18\thuman\tgut\tfeces\tSouth Korea:Gyeonggi-do, Seongnam-si\tHomo sapiens\tTSA supplemented with blood\t37.35 N 127.14 E\t1\t10.1099/ijs.0.02873-3\n",
      "\n",
      "GCF_017570525.1\tSAMN18309824\tKGMB01988\t2017-05-18\thuman\tgut\tfeces\tSouth Korea:Gyeonggi-do, Seongnam-si\tHomo sapiens\tTSA supplemented with blood\t37.35 N 127.12 E\t1\t10.1099/ijs.0.02873-1\n",
      "\n",
      "GCF_900184845.1\tSAMEA104102601\t2017-07-28\t2017-05-29\tERC000011\tSAMEA104102601\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\t2017-07-28T17:03:29Z\t2017-05-29T10:55:56Z\tpublic\tChimpanzee_1.fasta\tAkkermansia muciniphila\tChimpanzee_1.fasta\n",
      "\n",
      "GCF_900184905.1\tSAMEA104102612\t2017-07-28\t2017-05-29\tERC000011\tSAMEA104102612\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\t2017-07-28T17:03:29Z\t2017-05-29T10:55:57Z\tpublic\tSiamang.fasta\tAkkermansia muciniphila\tSiamang.fasta\n",
      "\n",
      "GCF_900184985.1\tSAMEA104102607\t2017-07-28\t2017-05-29\tERC000011\tSAMEA104102607\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\t2017-07-28T17:03:29Z\t2017-05-29T10:55:56Z\tpublic\tMouse_1_89.fasta\tAkkermansia muciniphila\tMouse_1_89.fasta\n",
      "\n",
      "GCF_900185025.1\tSAMEA104102606\t2017-07-28\t2017-05-29\tERC000011\tSAMEA104102606\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\t2017-07-28T17:03:29Z\t2017-05-29T10:55:56Z\tpublic\tMG2_JASPER.fasta\tAkkermansia muciniphila\tMG2_JASPER.fasta\n",
      "\n",
      "GCF_001683795.1\tSAMN03840437\tPyt\treticulated python\tfeces\t2011-10-05\tNetherlands: Burgers Zoo, Arnhem\tfaeces\t52.01 N 5.90 E\ttype strain of Akkermansia glycaniphila\n",
      "\n",
      "GCF_900097105.1\tSAMEA4378782\t2016-09-02\t2016-08-17\tERC000011\tSAMEA4378782\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\t2016-09-02T17:03:18Z\t2016-08-17T06:39:26Z\tpublic\tAPytT\tAPytT\n",
      "\n",
      "GCF_900184965.1\tSAMEA104102608\t2017-07-28\t2017-05-29\tERC000011\tSAMEA104102608\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\t2017-07-28T17:03:29Z\t2017-05-29T10:55:56Z\tpublic\tMouse_2_MG.fasta\tAkkermansia muciniphila\tMouse_2_MG.fasta\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for ln in ln_notpresent:\n",
    "    print(biosample_akk_desc[ln-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "c82bf167",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[7, 34, 35, 77, 78, 79, 80, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 168, 174, 176, 177, 179, 180] 34\n"
     ]
    }
   ],
   "source": [
    "# concatenate both strings\n",
    "ln = {int(el.split(':')[0]) for el in (ln_sp + ln_comm_sp).strip().split('\\n')}\n",
    "ln_notpresent = [el for el in range(1,181) if el not in ln]\n",
    "print(ln_notpresent, len(ln_notpresent))\n",
    "# for el in line_biome_all.split('\\n'):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "bff0cae2",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_003514565.1\tSAMN08019340\tUBA8862\tmetagenome\tnot applicable\tnot applicable\tmetagenomic assembly\tCLC de novo assembler\t4.4.1\tBWA (BWA-MEM)\t0.7.12-r1039\t27.77x\tCheckM\t1.0.6\t97.96%\t0.00%\tmetagenome\ttrue\ttrue\tmetagenome\tThis BioSample is a metagenomic assembly obtained from the metagenome reads: SRR606249.\n",
      "\n",
      "GCF_000020225.1\tSAMN00138213\tATCC BAA-835\n",
      "\n",
      "GCF_000723745.2\tSAMEA3139045\tGCA_000723745\tEuropean Nucleotide Archive\tERC000011\tEBI\tEuropean Bioinformatics Institute\t2014-11-20T12:20:15Z\t2014-11-20T11:56:31Z\tpublic\tERS610980\tERS610980\tBioSample entry for genome collection GCA_000723745\tUrmite\n",
      "\n",
      "GCF_003716915.1\tSAMN07978491\tEB-AMDK-1\tKorean adult feces\t2017-07-20\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_003716935.1\tSAMN07980909\tEB-AMDK-3\tKorean adult feces\t2017-07-20\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_003716955.1\tSAMN07980958\tEB-AMDK-4\tKorean adult feces\t2017-07-20\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_003716975.1\tSAMN07980963\tEB-AMDK-2\tKorean adult feces\t2017-07-20\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_010223015.1\tSAMN08563913\tEB-AMDK-27\tKorean adult feces\t2018-02-02\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_010223075.1\tSAMN08563914\tEB-AMDK-28\tKorean adult feces\t2018-02-02\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_010223095.1\tSAMN08565495\tEB-AMDK-29\tKorean adult feces\t2018-02-02\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_010223115.1\tSAMN08565496\tEB-AMDK-30\tKorean adult feces\t2018-02-02\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_010223135.1\tSAMN08565497\tEB-AMDK-31\tKorean adult feces\t2018-02-02\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_010223155.1\tSAMN08565498\tKorean adult feces\t2018-02-02\tSouth Korea: Seoul\tcell culture\tEB-AMDK-33\n",
      "\n",
      "GCF_010223175.1\tSAMN08565499\tEB-AMDK-34\tKorean adult feces\t2018-02-02\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_010223195.1\tSAMN08565501\tEB-AMDK-36\tKorean adult feces\t2018-02-02\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_010223255.1\tSAMN08565500\tEB-AMDK-35\tKorean adult feces\t2018-02-02\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_010223405.1\tSAMN08565502\tEB-AMDK-37\tKorean adult feces\t2018-02-02\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_010223575.1\tSAMN08565503\tEB-AMDK-38\tKorean adult feces\t2018-02-02\tSouth Korea: Seoul\tcell culture\n",
      "\n",
      "GCF_010229055.1\tSAMN09287185\tEB-AMDK-5\tKorean adult feces\t2018-02-08\tSouth Korea: Ilsan\tcell culture\n",
      "\n",
      "GCF_010229495.1\tSAMN09287631\tEB-AMDK-23\tKorean adult feces\t2018-02-08\tSouth Korea: Ilsan\tcell culture\n",
      "\n",
      "GCF_010229695.1\tSAMN09287781\tEB-AMDK-24\tKorean adult feces\t2018-02-08\tSouth Korea: Ilsan\tcell culture\n",
      "\n",
      "GCF_010229955.1\tSAMN09287782\tEB-AMDK-25\tKorean adult feces\t2018-02-08\tSouth Korea: Ilsan\tcell culture\n",
      "\n",
      "GCF_010230155.1\tSAMN09287783\tEB-AMDK-26\tKorean adult feces\t2018-02-08\tSouth Korea: Ilsan\tcell culture\n",
      "\n",
      "GCF_010230375.1\tSAMN09287787\tEB-AMDK-40\tKorean adult feces\t2018-02-08\tSouth Korea: Ilsan\tcell culture\n",
      "\n",
      "GCF_010230515.1\tSAMN09288166\tEB-AMDK-46\tKorean adult feces\t2018-02-08\tSouth Korea: Ilsan\tcell culture\n",
      "\n",
      "GCF_010230725.1\tSAMN09288168\tEB-AMDK-47\tKorean adult feces\t2018-02-08\tSouth Korea: Ilsan\tcell culture\n",
      "\n",
      "GCF_010230935.1\tSAMN09288183\tEB-AMDK-48\tKorean adult feces\t2018-02-08\tSouth Korea: Ilsan\tcell culture\n",
      "\n",
      "GCF_010231195.1\tSAMN09288185\tEB-AMDK-49\tKorean adult feces\t2018-02-08\tSouth Korea: Ilsan\tcell culture\n",
      "\n",
      "GCF_017504145.1\tSAMN00138213\tATCC BAA-835\n",
      "\n",
      "GCF_900184845.1\tSAMEA104102601\t2017-07-28\t2017-05-29\tERC000011\tSAMEA104102601\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\t2017-07-28T17:03:29Z\t2017-05-29T10:55:56Z\tpublic\tChimpanzee_1.fasta\tAkkermansia muciniphila\tChimpanzee_1.fasta\n",
      "\n",
      "GCF_900184985.1\tSAMEA104102607\t2017-07-28\t2017-05-29\tERC000011\tSAMEA104102607\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\t2017-07-28T17:03:29Z\t2017-05-29T10:55:56Z\tpublic\tMouse_1_89.fasta\tAkkermansia muciniphila\tMouse_1_89.fasta\n",
      "\n",
      "GCF_900185025.1\tSAMEA104102606\t2017-07-28\t2017-05-29\tERC000011\tSAMEA104102606\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\t2017-07-28T17:03:29Z\t2017-05-29T10:55:56Z\tpublic\tMG2_JASPER.fasta\tAkkermansia muciniphila\tMG2_JASPER.fasta\n",
      "\n",
      "GCF_900097105.1\tSAMEA4378782\t2016-09-02\t2016-08-17\tERC000011\tSAMEA4378782\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\t2016-09-02T17:03:18Z\t2016-08-17T06:39:26Z\tpublic\tAPytT\tAPytT\n",
      "\n",
      "GCF_900184965.1\tSAMEA104102608\t2017-07-28\t2017-05-29\tERC000011\tSAMEA104102608\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\tWAGENINGEN UNIVERSITY, LABORATORY OF SYSTEMS AND SYNTHETIC BIOLOGY.\t2017-07-28T17:03:29Z\t2017-05-29T10:55:56Z\tpublic\tMouse_2_MG.fasta\tAkkermansia muciniphila\tMouse_2_MG.fasta\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for ln in ln_notpresent:\n",
    "    print(biosample_akk_desc[ln-1])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7dd69730",
   "metadata": {},
   "source": [
    "# The same, but for BioProject"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f3f2fd7",
   "metadata": {},
   "source": [
    "## Read the manually edited file and check BioProjects"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "7c1da83a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 201 entries, 0 to 200\n",
      "Data columns (total 14 columns):\n",
      " #   Column                       Non-Null Count  Dtype  \n",
      "---  ------                       --------------  -----  \n",
      " 0   Genome                       201 non-null    object \n",
      " 1   Completeness                 201 non-null    float64\n",
      " 2   Contamination                201 non-null    float64\n",
      " 3   classification               201 non-null    object \n",
      " 4   classification_method        201 non-null    object \n",
      " 5   sum_len                      201 non-null    int64  \n",
      " 6   N50                          201 non-null    int64  \n",
      " 7   16S ribosomal RNA > 1400 nt  151 non-null    float64\n",
      " 8   Total_tRNAs                  201 non-null    int64  \n",
      " 9   host                         179 non-null    object \n",
      " 10  NCBI_organism_name           147 non-null    object \n",
      " 11  NCBI_infraspecific_name      140 non-null    object \n",
      " 12  relation_to_type_material    4 non-null      object \n",
      " 13  type_of_sample               194 non-null    object \n",
      "dtypes: float64(3), int64(3), object(8)\n",
      "memory usage: 22.1+ KB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(homo sapiens           155\n",
       " mus musculus            13\n",
       " rat                      3\n",
       " gallus gallus            2\n",
       " reticulated python       1\n",
       " horse                    1\n",
       " diceros bicornis         1\n",
       " siamang                  1\n",
       " bufo bufo                1\n",
       " choloepus hoffmanni      1\n",
       " Name: host, dtype: int64,\n",
       " 431,\n",
       " 7)"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_akk = pd.read_csv(qc_path.joinpath('merged_akk_hand_edited.csv'), sep='\\t')\n",
    "merged_akk.info()\n",
    "merged_akk['host'].value_counts(), merged_df['host'].isna().sum(), merged_akk['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec5b1420",
   "metadata": {},
   "source": [
    "# Set `type_of_sample`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "b7a81b9e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni 'culture|cultured|isolate|isolated|metagenome|monoisolate|uncultured' bioproject_descr_akk.tsv | sort -h | uniq\n",
      "4:metagenome\n",
      "8:metagenome\n",
      "9:metagenome\n",
      "10:metagenome\n",
      "12:metagenome\n",
      "13:metagenome\n",
      "14:metagenome\n",
      "15:me\n",
      "27\n",
      "index for the list in python: 33\n",
      "{'uncultured', 'metagenome'}\n"
     ]
    }
   ],
   "source": [
    "ln_culture_bp =  build_line_source('bioproject_descr_akk.tsv', \"'culture|cultured|isolate|isolated|metagenome|monoisolate|uncultured'\", \n",
    "                                file_pattern=False)\n",
    "print(ln_culture_bp[:100])\n",
    "acc_culture_bp = build_acc_host(ln_culture_bp, 'biosample_descr_akk.tsv')\n",
    "print(set(acc_culture_bp.values()))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf0f3963",
   "metadata": {},
   "source": [
    "**Which bioprojects line numbers aren't present in the `ln_culture`?**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "5df0fbbf",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30\n",
      "PRJNA562766\tHuman gut metagenome\tFecal microbiome analysis of patients with chronic obstructive pulmonary disease\tStudy analysed metagenomes derived from fecal samples from 28 individuals with chronic obstructive pulmonary disease (COPD) and 29 healthy controls. Paired metabolomic data was also generated for the same samples.\n",
      "\n",
      "PRJEB26432\tUncultured human gut species\tA new genomic blueprint of the human gut microbiota\tThe composition of the human gut microbiota is linked to health and disease, but knowledge of individual microbial species is needed to decipher their biological roles. Despite extensive culturing and sequencing efforts, the complete bacterial repertoire of the human gut microbiota remains undefined. Here we identify 1,952 uncultured candidate bacterial species by reconstructing 92,143 metagenome-assembled genomes from 11,850 human gut microbiomes. These uncultured genomes substantially expand the known species repertoire of the collective human gut microbiota, with a 281% increase in phylogenetic diversity. Although the newly identified species are less prevalent in well-studied populations compared to reference isolate genomes, they improve classification of understudied African and South American samples by more than 200%. These candidate species encode hundreds of newly identified biosynthetic gene clusters and possess a distinctive functional capacity that might explain their elusive nature. Our work expands the known diversity of uncultured gut bacteria, which provides unprecedented resolution for taxonomic and functional characterization of the intestinal microbiota.\n",
      "\n",
      "PRJEB26432\tUncultured human gut species\tA new genomic blueprint of the human gut microbiota\tThe composition of the human gut microbiota is linked to health and disease, but knowledge of individual microbial species is needed to decipher their biological roles. Despite extensive culturing and sequencing efforts, the complete bacterial repertoire of the human gut microbiota remains undefined. Here we identify 1,952 uncultured candidate bacterial species by reconstructing 92,143 metagenome-assembled genomes from 11,850 human gut microbiomes. These uncultured genomes substantially expand the known species repertoire of the collective human gut microbiota, with a 281% increase in phylogenetic diversity. Although the newly identified species are less prevalent in well-studied populations compared to reference isolate genomes, they improve classification of understudied African and South American samples by more than 200%. These candidate species encode hundreds of newly identified biosynthetic gene clusters and possess a distinctive functional capacity that might explain their elusive nature. Our work expands the known diversity of uncultured gut bacteria, which provides unprecedented resolution for taxonomic and functional characterization of the intestinal microbiota.\n",
      "\n",
      "PRJEB26432\tUncultured human gut species\tA new genomic blueprint of the human gut microbiota\tThe composition of the human gut microbiota is linked to health and disease, but knowledge of individual microbial species is needed to decipher their biological roles. Despite extensive culturing and sequencing efforts, the complete bacterial repertoire of the human gut microbiota remains undefined. Here we identify 1,952 uncultured candidate bacterial species by reconstructing 92,143 metagenome-assembled genomes from 11,850 human gut microbiomes. These uncultured genomes substantially expand the known species repertoire of the collective human gut microbiota, with a 281% increase in phylogenetic diversity. Although the newly identified species are less prevalent in well-studied populations compared to reference isolate genomes, they improve classification of understudied African and South American samples by more than 200%. These candidate species encode hundreds of newly identified biosynthetic gene clusters and possess a distinctive functional capacity that might explain their elusive nature. Our work expands the known diversity of uncultured gut bacteria, which provides unprecedented resolution for taxonomic and functional characterization of the intestinal microbiota.\n",
      "\n",
      "PRJEB26432\tUncultured human gut species\tA new genomic blueprint of the human gut microbiota\tThe composition of the human gut microbiota is linked to health and disease, but knowledge of individual microbial species is needed to decipher their biological roles. Despite extensive culturing and sequencing efforts, the complete bacterial repertoire of the human gut microbiota remains undefined. Here we identify 1,952 uncultured candidate bacterial species by reconstructing 92,143 metagenome-assembled genomes from 11,850 human gut microbiomes. These uncultured genomes substantially expand the known species repertoire of the collective human gut microbiota, with a 281% increase in phylogenetic diversity. Although the newly identified species are less prevalent in well-studied populations compared to reference isolate genomes, they improve classification of understudied African and South American samples by more than 200%. These candidate species encode hundreds of newly identified biosynthetic gene clusters and possess a distinctive functional capacity that might explain their elusive nature. Our work expands the known diversity of uncultured gut bacteria, which provides unprecedented resolution for taxonomic and functional characterization of the intestinal microbiota.\n",
      "\n",
      "PRJEB26432\tUncultured human gut species\tA new genomic blueprint of the human gut microbiota\tThe composition of the human gut microbiota is linked to health and disease, but knowledge of individual microbial species is needed to decipher their biological roles. Despite extensive culturing and sequencing efforts, the complete bacterial repertoire of the human gut microbiota remains undefined. Here we identify 1,952 uncultured candidate bacterial species by reconstructing 92,143 metagenome-assembled genomes from 11,850 human gut microbiomes. These uncultured genomes substantially expand the known species repertoire of the collective human gut microbiota, with a 281% increase in phylogenetic diversity. Although the newly identified species are less prevalent in well-studied populations compared to reference isolate genomes, they improve classification of understudied African and South American samples by more than 200%. These candidate species encode hundreds of newly identified biosynthetic gene clusters and possess a distinctive functional capacity that might explain their elusive nature. Our work expands the known diversity of uncultured gut bacteria, which provides unprecedented resolution for taxonomic and functional characterization of the intestinal microbiota.\n",
      "\n",
      "PRJEB26432\tUncultured human gut species\tA new genomic blueprint of the human gut microbiota\tThe composition of the human gut microbiota is linked to health and disease, but knowledge of individual microbial species is needed to decipher their biological roles. Despite extensive culturing and sequencing efforts, the complete bacterial repertoire of the human gut microbiota remains undefined. Here we identify 1,952 uncultured candidate bacterial species by reconstructing 92,143 metagenome-assembled genomes from 11,850 human gut microbiomes. These uncultured genomes substantially expand the known species repertoire of the collective human gut microbiota, with a 281% increase in phylogenetic diversity. Although the newly identified species are less prevalent in well-studied populations compared to reference isolate genomes, they improve classification of understudied African and South American samples by more than 200%. These candidate species encode hundreds of newly identified biosynthetic gene clusters and possess a distinctive functional capacity that might explain their elusive nature. Our work expands the known diversity of uncultured gut bacteria, which provides unprecedented resolution for taxonomic and functional characterization of the intestinal microbiota.\n",
      "\n",
      "PRJEB26432\tUncultured human gut species\tA new genomic blueprint of the human gut microbiota\tThe composition of the human gut microbiota is linked to health and disease, but knowledge of individual microbial species is needed to decipher their biological roles. Despite extensive culturing and sequencing efforts, the complete bacterial repertoire of the human gut microbiota remains undefined. Here we identify 1,952 uncultured candidate bacterial species by reconstructing 92,143 metagenome-assembled genomes from 11,850 human gut microbiomes. These uncultured genomes substantially expand the known species repertoire of the collective human gut microbiota, with a 281% increase in phylogenetic diversity. Although the newly identified species are less prevalent in well-studied populations compared to reference isolate genomes, they improve classification of understudied African and South American samples by more than 200%. These candidate species encode hundreds of newly identified biosynthetic gene clusters and possess a distinctive functional capacity that might explain their elusive nature. Our work expands the known diversity of uncultured gut bacteria, which provides unprecedented resolution for taxonomic and functional characterization of the intestinal microbiota.\n",
      "\n",
      "PRJEB26432\tUncultured human gut species\tA new genomic blueprint of the human gut microbiota\tThe composition of the human gut microbiota is linked to health and disease, but knowledge of individual microbial species is needed to decipher their biological roles. Despite extensive culturing and sequencing efforts, the complete bacterial repertoire of the human gut microbiota remains undefined. Here we identify 1,952 uncultured candidate bacterial species by reconstructing 92,143 metagenome-assembled genomes from 11,850 human gut microbiomes. These uncultured genomes substantially expand the known species repertoire of the collective human gut microbiota, with a 281% increase in phylogenetic diversity. Although the newly identified species are less prevalent in well-studied populations compared to reference isolate genomes, they improve classification of understudied African and South American samples by more than 200%. These candidate species encode hundreds of newly identified biosynthetic gene clusters and possess a distinctive functional capacity that might explain their elusive nature. Our work expands the known diversity of uncultured gut bacteria, which provides unprecedented resolution for taxonomic and functional characterization of the intestinal microbiota.\n",
      "\n",
      "PRJEB26432\tUncultured human gut species\tA new genomic blueprint of the human gut microbiota\tThe composition of the human gut microbiota is linked to health and disease, but knowledge of individual microbial species is needed to decipher their biological roles. Despite extensive culturing and sequencing efforts, the complete bacterial repertoire of the human gut microbiota remains undefined. Here we identify 1,952 uncultured candidate bacterial species by reconstructing 92,143 metagenome-assembled genomes from 11,850 human gut microbiomes. These uncultured genomes substantially expand the known species repertoire of the collective human gut microbiota, with a 281% increase in phylogenetic diversity. Although the newly identified species are less prevalent in well-studied populations compared to reference isolate genomes, they improve classification of understudied African and South American samples by more than 200%. These candidate species encode hundreds of newly identified biosynthetic gene clusters and possess a distinctive functional capacity that might explain their elusive nature. Our work expands the known diversity of uncultured gut bacteria, which provides unprecedented resolution for taxonomic and functional characterization of the intestinal microbiota.\n",
      "\n",
      "PRJEB31003\tGlobal Human Gut MAG (HGM) Dataset\tProkaryotic genomes for 2058 new species assembled from globally distributed human gut metagenomes\tLargely due to challenges cultivating microbes under laboratory conditions, many species in the human gut microbiome do not have a genome sequence. To address this problem, we reconstructed 60,664 prokaryotic draft genomes from 3,810 fecal metagenomes from geographically and phenotypically diverse human subjects. These genomes provide reference points for 2,058 previously unknown uncultured species and nearly double the phylogenetic diversity of sequenced gut Bacteria. New species comprise 33% of species richness and 28% of species abundance per individual on average and are more common in humans living rural lifestyles. A metagenome-wide association study of ten human diseases revealed thousands of new species-level biomarkers, particularly for colorectal cancer, ankylosing spondylitis, and obesity. Our analysis also uncovered novel genomic and ecological features of uncultured species, including reduced genome size, metabolic auxotrophies, and increased richness in high-diversity communities. We anticipate that this dataset and associated tools will drive new culturing efforts and enable further discoveries from the human gut microbiome.\n",
      "\n",
      "PRJEB31003\tGlobal Human Gut MAG (HGM) Dataset\tProkaryotic genomes for 2058 new species assembled from globally distributed human gut metagenomes\tLargely due to challenges cultivating microbes under laboratory conditions, many species in the human gut microbiome do not have a genome sequence. To address this problem, we reconstructed 60,664 prokaryotic draft genomes from 3,810 fecal metagenomes from geographically and phenotypically diverse human subjects. These genomes provide reference points for 2,058 previously unknown uncultured species and nearly double the phylogenetic diversity of sequenced gut Bacteria. New species comprise 33% of species richness and 28% of species abundance per individual on average and are more common in humans living rural lifestyles. A metagenome-wide association study of ten human diseases revealed thousands of new species-level biomarkers, particularly for colorectal cancer, ankylosing spondylitis, and obesity. Our analysis also uncovered novel genomic and ecological features of uncultured species, including reduced genome size, metabolic auxotrophies, and increased richness in high-diversity communities. We anticipate that this dataset and associated tools will drive new culturing efforts and enable further discoveries from the human gut microbiome.\n",
      "\n",
      "PRJEB31003\tGlobal Human Gut MAG (HGM) Dataset\tProkaryotic genomes for 2058 new species assembled from globally distributed human gut metagenomes\tLargely due to challenges cultivating microbes under laboratory conditions, many species in the human gut microbiome do not have a genome sequence. To address this problem, we reconstructed 60,664 prokaryotic draft genomes from 3,810 fecal metagenomes from geographically and phenotypically diverse human subjects. These genomes provide reference points for 2,058 previously unknown uncultured species and nearly double the phylogenetic diversity of sequenced gut Bacteria. New species comprise 33% of species richness and 28% of species abundance per individual on average and are more common in humans living rural lifestyles. A metagenome-wide association study of ten human diseases revealed thousands of new species-level biomarkers, particularly for colorectal cancer, ankylosing spondylitis, and obesity. Our analysis also uncovered novel genomic and ecological features of uncultured species, including reduced genome size, metabolic auxotrophies, and increased richness in high-diversity communities. We anticipate that this dataset and associated tools will drive new culturing efforts and enable further discoveries from the human gut microbiome.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n",
      "PRJEB37358\tUncultured novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tRecovery of novel prokaryotic genomes associated with metabolic diseases such as T2D and T2D-related comorbidities\tMetabolic diseases such as type 2 diabetes (T2D) and T2D-related comorbidities (eg. obesity, metabolic syndrome, pre-diabetes and impaired glucose tolerance) are major public health issues worldwide. Recent experiments in animal and human studies have generated growing evidence for the association of gut microbiome and development of these diseases. Using next generation sequencing, large amounts of metagenomic data have been generated from T2D and control subjects. However, the disease-associated prokaryotic repertoire of the human gut microbiota is still undefined. Here we identified 389 uncultured novel prokaryotic species, including 388 bacteria and 1 archaea, by reconstructing 95871 metagenome-assembled genomes (MAGs) from 5814 disease-associated human gut microbiomes and de-replicating the MAGs against a comprehensive human gut library (ERP108418). Amongst these uncultured genomes, 302 are novel species and 87 are related to known species but have never been found in the human gut before. Our work expands the knowledge of the diversity of uncultured human gut bacteria associated with metabolic diseases, and can be used as a valuable reference library to help improve the species detection in future disease-associated metagenomics studies.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "bp_notin_ln_cult = [int(n.split(':')[0]) for n in ln_culture_bp.strip().split('\\n') \n",
    "                if n not in ln_culture.strip().split('\\n')]\n",
    "print(len(bp_notin_ln_cult))\n",
    "for ln in bp_notin_ln_cult:\n",
    "    print(bioproj_akk_desc[ln-1])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9e9db170",
   "metadata": {},
   "source": [
    "# Make a table with assembly-biosample-bioproject equivalencies\n",
    "Then, we can use this mappings to get the assemblies easily"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "b6579629",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1158218 entries, 0 to 1158217\n",
      "Data columns (total 23 columns):\n",
      " #   Column                     Non-Null Count    Dtype \n",
      "---  ------                     --------------    ----- \n",
      " 0   Genome                     1158163 non-null  object\n",
      " 1   assembly_accession         1158218 non-null  object\n",
      " 2   bioproject                 1158214 non-null  object\n",
      " 3   biosample                  1158213 non-null  object\n",
      " 4   wgs_master                 1104090 non-null  object\n",
      " 5   refseq_category            1158218 non-null  object\n",
      " 6   taxid                      1158218 non-null  int64 \n",
      " 7   species_taxid              1158218 non-null  int64 \n",
      " 8   NCBI_organism_name         1158218 non-null  object\n",
      " 9   NCBI_infraspecific_name    998033 non-null   object\n",
      " 10  isolate                    182630 non-null   object\n",
      " 11  version_status             1158218 non-null  object\n",
      " 12  assembly_level             1158218 non-null  object\n",
      " 13  release_type               1158218 non-null  object\n",
      " 14  genome_rep                 1158218 non-null  object\n",
      " 15  seq_rel_date               1158218 non-null  object\n",
      " 16  asm_name                   1158217 non-null  object\n",
      " 17  submitter                  1158210 non-null  object\n",
      " 18  gbrs_paired_asm            1158218 non-null  object\n",
      " 19  paired_asm_comp            1158218 non-null  object\n",
      " 20  ftp_path                   1158218 non-null  object\n",
      " 21  excluded_from_refseq       726985 non-null   object\n",
      " 22  relation_to_type_material  32556 non-null    object\n",
      "dtypes: int64(2), object(21)\n",
      "memory usage: 203.2+ MB\n"
     ]
    }
   ],
   "source": [
    "ncbi_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "89e97b30",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>classification</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>sum_len</th>\n",
       "      <th>N50</th>\n",
       "      <th>16S ribosomal RNA &gt; 1400 nt</th>\n",
       "      <th>Total_tRNAs</th>\n",
       "      <th>host</th>\n",
       "      <th>NCBI_organism_name</th>\n",
       "      <th>NCBI_infraspecific_name</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>type_of_sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DGYMR06203__metabat2_low_PE.047.contigs</td>\n",
       "      <td>91.12</td>\n",
       "      <td>0.0</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2417897</td>\n",
       "      <td>26278</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44</td>\n",
       "      <td>gallus gallus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>F157a_European_Toad__metabat2_high_PE.005.contigs</td>\n",
       "      <td>89.32</td>\n",
       "      <td>2.3</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2029501</td>\n",
       "      <td>8457</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40</td>\n",
       "      <td>bufo bufo</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              Genome  Completeness  \\\n",
       "0            DGYMR06203__metabat2_low_PE.047.contigs         91.12   \n",
       "1  F157a_European_Toad__metabat2_high_PE.005.contigs         89.32   \n",
       "\n",
       "   Contamination                                     classification  \\\n",
       "0            0.0  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "1            2.3  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "                               classification_method  sum_len    N50  \\\n",
       "0  taxonomic classification defined by topology a...  2417897  26278   \n",
       "1  taxonomic classification defined by topology a...  2029501   8457   \n",
       "\n",
       "   16S ribosomal RNA > 1400 nt  Total_tRNAs           host NCBI_organism_name  \\\n",
       "0                          NaN           44  gallus gallus                NaN   \n",
       "1                          NaN           40      bufo bufo                NaN   \n",
       "\n",
       "  NCBI_infraspecific_name relation_to_type_material type_of_sample  \n",
       "0                     NaN                       NaN            NaN  \n",
       "1                     NaN                       NaN            NaN  "
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "bbc09c46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>biosample</th>\n",
       "      <th>bioproject</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000155695.1</td>\n",
       "      <td>SAMN02436317</td>\n",
       "      <td>PRJNA19391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000242935.3</td>\n",
       "      <td>SAMN02261410</td>\n",
       "      <td>PRJNA63271</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  assembly_accession     biosample  bioproject\n",
       "0    GCA_000155695.1  SAMN02436317  PRJNA19391\n",
       "1    GCA_000242935.3  SAMN02261410  PRJNA63271"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_accs = (merged_df.loc[:,'Genome'].str.extract( '(GC[AF]_[0-9]{9}\\.[1-9])').\n",
    "            dropna().values.flatten())\n",
    "asm_biosam_bioproj = (ncbi_df.set_index('assembly_accession').loc[all_accs, :].\n",
    "                      loc[:, ['biosample', 'bioproject']].reset_index())\n",
    "asm_biosam_bioproj.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "ff3159a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>biosample</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bioproject</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PRJEB31003</th>\n",
       "      <td>GCA_900752515.1</td>\n",
       "      <td>SAMEA5279134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB31003</th>\n",
       "      <td>GCA_900759685.1</td>\n",
       "      <td>SAMEA5279138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB31003</th>\n",
       "      <td>GCA_900760665.1</td>\n",
       "      <td>SAMEA5279772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB31003</th>\n",
       "      <td>GCA_900763545.1</td>\n",
       "      <td>SAMEA5279771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB31003</th>\n",
       "      <td>GCA_900765745.1</td>\n",
       "      <td>SAMEA5279136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB37358</th>\n",
       "      <td>GCA_905193115.1</td>\n",
       "      <td>SAMEA7846742</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB37358</th>\n",
       "      <td>GCA_905193715.1</td>\n",
       "      <td>SAMEA7847250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB37358</th>\n",
       "      <td>GCA_905197035.1</td>\n",
       "      <td>SAMEA7847110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB37358</th>\n",
       "      <td>GCA_905198945.1</td>\n",
       "      <td>SAMEA7846640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB37358</th>\n",
       "      <td>GCA_905200125.1</td>\n",
       "      <td>SAMEA7848347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB37358</th>\n",
       "      <td>GCA_905200945.1</td>\n",
       "      <td>SAMEA7847584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB37358</th>\n",
       "      <td>GCA_905205055.1</td>\n",
       "      <td>SAMEA7848078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB37358</th>\n",
       "      <td>GCA_905205865.1</td>\n",
       "      <td>SAMEA7846715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB37358</th>\n",
       "      <td>GCA_905205915.1</td>\n",
       "      <td>SAMEA7846587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB37358</th>\n",
       "      <td>GCA_905210405.1</td>\n",
       "      <td>SAMEA7847399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB37358</th>\n",
       "      <td>GCA_905210615.1</td>\n",
       "      <td>SAMEA7847488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJNA562766</th>\n",
       "      <td>GCA_014871965.1</td>\n",
       "      <td>SAMN12905381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJNA562766</th>\n",
       "      <td>GCA_014871995.1</td>\n",
       "      <td>SAMN12905379</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJNA562766</th>\n",
       "      <td>GCA_014872005.1</td>\n",
       "      <td>SAMN12905380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJNA562766</th>\n",
       "      <td>GCA_014872035.1</td>\n",
       "      <td>SAMN12905378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJNA562766</th>\n",
       "      <td>GCA_014872065.1</td>\n",
       "      <td>SAMN12905377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJNA562766</th>\n",
       "      <td>GCA_014872075.1</td>\n",
       "      <td>SAMN12905375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJNA562766</th>\n",
       "      <td>GCA_014872105.1</td>\n",
       "      <td>SAMN12905376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJNA562766</th>\n",
       "      <td>GCA_014872125.1</td>\n",
       "      <td>SAMN12905373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJNA562766</th>\n",
       "      <td>GCA_014872165.1</td>\n",
       "      <td>SAMN12905372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB26432</th>\n",
       "      <td>GCA_900539575.1</td>\n",
       "      <td>SAMEA4890833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB26432</th>\n",
       "      <td>GCA_900539895.1</td>\n",
       "      <td>SAMEA4890871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB26432</th>\n",
       "      <td>GCA_900545155.1</td>\n",
       "      <td>SAMEA4891402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRJEB26432</th>\n",
       "      <td>GCA_900547395.1</td>\n",
       "      <td>SAMEA4891626</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            assembly_accession     biosample\n",
       "bioproject                                  \n",
       "PRJEB31003     GCA_900752515.1  SAMEA5279134\n",
       "PRJEB31003     GCA_900759685.1  SAMEA5279138\n",
       "PRJEB31003     GCA_900760665.1  SAMEA5279772\n",
       "PRJEB31003     GCA_900763545.1  SAMEA5279771\n",
       "PRJEB31003     GCA_900765745.1  SAMEA5279136\n",
       "PRJEB37358     GCA_905193115.1  SAMEA7846742\n",
       "PRJEB37358     GCA_905193715.1  SAMEA7847250\n",
       "PRJEB37358     GCA_905197035.1  SAMEA7847110\n",
       "PRJEB37358     GCA_905198945.1  SAMEA7846640\n",
       "PRJEB37358     GCA_905200125.1  SAMEA7848347\n",
       "PRJEB37358     GCA_905200945.1  SAMEA7847584\n",
       "PRJEB37358     GCA_905205055.1  SAMEA7848078\n",
       "PRJEB37358     GCA_905205865.1  SAMEA7846715\n",
       "PRJEB37358     GCA_905205915.1  SAMEA7846587\n",
       "PRJEB37358     GCA_905210405.1  SAMEA7847399\n",
       "PRJEB37358     GCA_905210615.1  SAMEA7847488\n",
       "PRJNA562766    GCA_014871965.1  SAMN12905381\n",
       "PRJNA562766    GCA_014871995.1  SAMN12905379\n",
       "PRJNA562766    GCA_014872005.1  SAMN12905380\n",
       "PRJNA562766    GCA_014872035.1  SAMN12905378\n",
       "PRJNA562766    GCA_014872065.1  SAMN12905377\n",
       "PRJNA562766    GCA_014872075.1  SAMN12905375\n",
       "PRJNA562766    GCA_014872105.1  SAMN12905376\n",
       "PRJNA562766    GCA_014872125.1  SAMN12905373\n",
       "PRJNA562766    GCA_014872165.1  SAMN12905372\n",
       "PRJEB26432     GCA_900539575.1  SAMEA4890833\n",
       "PRJEB26432     GCA_900539895.1  SAMEA4890871\n",
       "PRJEB26432     GCA_900545155.1  SAMEA4891402\n",
       "PRJEB26432     GCA_900547395.1  SAMEA4891626"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "asms = {bioproj_akk_desc[ln-1].split('\\t')[0] for ln in bp_notin_ln_cult}\n",
    "asm_biosam_bioproj.set_index('bioproject').loc[asms, :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "1ae01ba9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosample</th>\n",
       "      <th>bioproject</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assembly_accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_009731575.1</th>\n",
       "      <td>SAMD00192834</td>\n",
       "      <td>PRJNA224116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001688765.2</th>\n",
       "      <td>SAMN04621615</td>\n",
       "      <td>PRJNA224116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_002161325.1</th>\n",
       "      <td>SAMN06473757</td>\n",
       "      <td>PRJNA224116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_002201495.1</th>\n",
       "      <td>SAMN03854119</td>\n",
       "      <td>PRJNA224116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_009663785.1</th>\n",
       "      <td>SAMN10240011</td>\n",
       "      <td>PRJNA224116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001647615.1</th>\n",
       "      <td>SAMN04606485</td>\n",
       "      <td>PRJNA224116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000723745.2</th>\n",
       "      <td>SAMEA3139045</td>\n",
       "      <td>PRJNA224116</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       biosample   bioproject\n",
       "assembly_accession                           \n",
       "GCF_009731575.1     SAMD00192834  PRJNA224116\n",
       "GCF_001688765.2     SAMN04621615  PRJNA224116\n",
       "GCF_002161325.1     SAMN06473757  PRJNA224116\n",
       "GCF_002201495.1     SAMN03854119  PRJNA224116\n",
       "GCF_009663785.1     SAMN10240011  PRJNA224116\n",
       "GCF_001647615.1     SAMN04606485  PRJNA224116\n",
       "GCF_000723745.2     SAMEA3139045  PRJNA224116"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idxs = set(merged_akk[merged_akk['type_of_sample'].isna()]['Genome'].str.\n",
    " extract( '(GC[AF]_[0-9]{9}\\.[1-9])').dropna().values.flatten()) \n",
    "\n",
    "asm_biosam_bioproj.set_index('assembly_accession').loc[idxs, :]#.str.extract( '(GC[AF]_[0-9]{9}\\.[1-9])').dropna().values.flatten()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "id": "f10d886a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SAMD00192834\n",
      "AMJCM30893\tAkkermansia_muciniphila_JCM30893\tAKMU\tJCM 30893\tmissing\tmissing\tmissing\tmissing\tmissing\tmissing\tA_muchiniphila_JCM30893\tmissing\tmissing\tmissing\tHomo sapiens\tAkkermansia muciniphila\n",
      "SAMN04621615\n",
      "Microbe sample from Akkermansia muciniphila\tYL44\tMus musculus\t2015-02-26\tSwitzerland: Bern\twhole organism\tAkkermansia muciniphila\n",
      "SAMN06473757\n",
      "MIGS Cultured Bacterial/Archaeal sample from Akkermansia muciniphila\thealthy\tmissing\tnot applicable\tmissing\tnot applicable\tGallus gallus\tnot applicable\tnot applicable\tcommensal\tcaecum\tAn78\tanaerobe\tcaecum\t2016\tnot applicable\tmissing\tAkkermansia muciniphila\n",
      "SAMN03854119\n",
      "Akkermansia muciniphila YL44\tYL44\tSwitzerland: Zurich\tnot applicable\tnot applicable\tnot applicable\torganic feature\tintestine environment\t47 N 8 E\torganic material\tnot applicable\tMus musculus\tDSM:26127\tAkkermansia muciniphila\n",
      "SAMN10240011\n",
      "MIGS Cultured Bacterial/Archaeal sample from Akkermansia muciniphila\tD10-15\t2016-06-28\tintestine enviorment\tintestine\tanimal manure\tChina:Beijing\tob/ob mouse\tmGAM medium at 37 centigrade\t39.59 N 116.23 E\t1\tno applicable\tAkkermansia muciniphila\n",
      "SAMN04606485\n",
      "Sample from Akkermansia sp. KLE1605\tGI tract\tKLE1605\tHomo sapiens\tHigh-Quality Draft\tHigh-Quality Draft\tVelvet v. 1.1.04\t128x\tIllumina\tAkkermansia sp. KLE1605\n",
      "SAMEA3139045\n",
      "GCA_000723745\tGCA_000723745\tEuropean Nucleotide Archive\tERC000011\tEBI\tEuropean Bioinformatics Institute\t2014-11-20T12:20:15Z\t2014-11-20T11:56:31Z\tpublic\tERS610980\tERS610980\tBioSample entry for genome collection GCA_000723745\tUrmite\tAkkermansia muciniphila\n"
     ]
    }
   ],
   "source": [
    "ls = asm_biosam_bioproj.set_index('assembly_accession').loc[idxs, :].loc[:,'biosample'].to_list()\n",
    "for bp in ls:\n",
    "    print(bp)\n",
    "    ! esearch -db biosample -query $bp | efetch -format xml | xtract -pattern BioSampleSet -element  Title  Attribute  OrganismName\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "57269f7d",
   "metadata": {},
   "source": [
    "# Annotation for the non-akkermansia assemblies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 430,
   "id": "de78ac41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(Series([], Name: host, dtype: int64),\n",
       " Series([], Name: type_of_sample, dtype: int64),\n",
       " 384)"
      ]
     },
     "execution_count": 430,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_non_akk = merged_df[~merged_df['classification'].str.contains('g__Akkermansia')].copy()\n",
    "(merged_non_akk['host'].value_counts(), \n",
    " merged_non_akk['type_of_sample'].value_counts(), merged_non_akk['type_of_sample'].isna().sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 431,
   "id": "120daf54",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>classification</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>sum_len</th>\n",
       "      <th>N50</th>\n",
       "      <th>16S ribosomal RNA &gt; 1400 nt</th>\n",
       "      <th>Total_tRNAs</th>\n",
       "      <th>host</th>\n",
       "      <th>NCBI_organism_name</th>\n",
       "      <th>NCBI_infraspecific_name</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>type_of_sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000155695.1_ASM15569v1_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>1.35</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>5775745</td>\n",
       "      <td>5514226</td>\n",
       "      <td>1.0</td>\n",
       "      <td>45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000242935.3_ASM24293v3_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.23</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7417673</td>\n",
       "      <td>7317842</td>\n",
       "      <td>1.0</td>\n",
       "      <td>52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000243495.3_ASM24349v3_genomic</td>\n",
       "      <td>98.63</td>\n",
       "      <td>0.23</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7073483</td>\n",
       "      <td>6196881</td>\n",
       "      <td>1.0</td>\n",
       "      <td>71</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Genome  Completeness  Contamination  \\\n",
       "2  GCA_000155695.1_ASM15569v1_genomic        100.00           1.35   \n",
       "3  GCA_000242935.3_ASM24293v3_genomic        100.00           0.23   \n",
       "4  GCA_000243495.3_ASM24349v3_genomic         98.63           0.23   \n",
       "\n",
       "                                      classification  \\\n",
       "2  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "3  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "4  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "                               classification_method  sum_len      N50  \\\n",
       "2  taxonomic classification defined by topology a...  5775745  5514226   \n",
       "3  taxonomic classification defined by topology a...  7417673  7317842   \n",
       "4  taxonomic classification defined by topology a...  7073483  6196881   \n",
       "\n",
       "   16S ribosomal RNA > 1400 nt  Total_tRNAs host NCBI_organism_name  \\\n",
       "2                          1.0           45  NaN                NaN   \n",
       "3                          1.0           52  NaN                NaN   \n",
       "4                          1.0           71  NaN                NaN   \n",
       "\n",
       "  NCBI_infraspecific_name relation_to_type_material type_of_sample  \n",
       "2                     NaN                       NaN            NaN  \n",
       "3                     NaN                       NaN            NaN  \n",
       "4                     NaN                       NaN            NaN  "
      ]
     },
     "execution_count": 431,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_non_akk.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 432,
   "id": "83191f33",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni -f /home/mibu/common_names_ncbi.list biosample_descr_non_akk.tsv | sort -h | uniq\n",
      "rg -woni -f /home/mibu/amniota_only_sp.list biosample_descr_non_akk.tsv | sort -h | uniq\n",
      "7\n",
      "index for the list in python: 318\n",
      "1\n",
      "index for the list in python: 311\n",
      "human            7\n",
      "gallus gallus    1\n",
      "Name: host, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>classification</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>sum_len</th>\n",
       "      <th>N50</th>\n",
       "      <th>16S ribosomal RNA &gt; 1400 nt</th>\n",
       "      <th>Total_tRNAs</th>\n",
       "      <th>host</th>\n",
       "      <th>NCBI_organism_name</th>\n",
       "      <th>NCBI_infraspecific_name</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>type_of_sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000155695.1_ASM15569v1_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>1.35</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>5775745</td>\n",
       "      <td>5514226</td>\n",
       "      <td>1.0</td>\n",
       "      <td>45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000242935.3_ASM24293v3_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.23</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7417673</td>\n",
       "      <td>7317842</td>\n",
       "      <td>1.0</td>\n",
       "      <td>52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000243495.3_ASM24349v3_genomic</td>\n",
       "      <td>98.63</td>\n",
       "      <td>0.23</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7073483</td>\n",
       "      <td>6196881</td>\n",
       "      <td>1.0</td>\n",
       "      <td>71</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCA_000526255.1_ASM52625v1_genomic</td>\n",
       "      <td>98.65</td>\n",
       "      <td>0.68</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2465519</td>\n",
       "      <td>1285429</td>\n",
       "      <td>1.0</td>\n",
       "      <td>48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_000739615.1_ASM73961v1_genomic</td>\n",
       "      <td>99.86</td>\n",
       "      <td>2.72</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7795129</td>\n",
       "      <td>203667</td>\n",
       "      <td>1.0</td>\n",
       "      <td>52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>558</th>\n",
       "      <td>GCF_902726605.1_LCC19_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>2.03</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>4878525</td>\n",
       "      <td>109520</td>\n",
       "      <td>1.0</td>\n",
       "      <td>48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lentimonas sp. CC19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>GCF_902728225.1_LCC11_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>2.03</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>4879546</td>\n",
       "      <td>130068</td>\n",
       "      <td>1.0</td>\n",
       "      <td>49</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lentimonas sp. CC11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>GCF_902728235.1_LCC4_PacBio_genomic</td>\n",
       "      <td>99.32</td>\n",
       "      <td>2.03</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>4871565</td>\n",
       "      <td>3986899</td>\n",
       "      <td>3.0</td>\n",
       "      <td>53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lentimonas sp. CC4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>561</th>\n",
       "      <td>GCF_902728245.1_LCC21_genomic</td>\n",
       "      <td>99.16</td>\n",
       "      <td>2.03</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>4884521</td>\n",
       "      <td>148072</td>\n",
       "      <td>1.0</td>\n",
       "      <td>47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lentimonas sp. CC21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>562</th>\n",
       "      <td>GCF_902728255.1_LCC8_genomic</td>\n",
       "      <td>99.16</td>\n",
       "      <td>2.03</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>4886160</td>\n",
       "      <td>168040</td>\n",
       "      <td>1.0</td>\n",
       "      <td>47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lentimonas sp. CC8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>384 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Genome  Completeness  Contamination  \\\n",
       "2     GCA_000155695.1_ASM15569v1_genomic        100.00           1.35   \n",
       "3     GCA_000242935.3_ASM24293v3_genomic        100.00           0.23   \n",
       "4     GCA_000243495.3_ASM24349v3_genomic         98.63           0.23   \n",
       "7     GCA_000526255.1_ASM52625v1_genomic         98.65           0.68   \n",
       "8     GCA_000739615.1_ASM73961v1_genomic         99.86           2.72   \n",
       "..                                   ...           ...            ...   \n",
       "558        GCF_902726605.1_LCC19_genomic        100.00           2.03   \n",
       "559        GCF_902728225.1_LCC11_genomic        100.00           2.03   \n",
       "560  GCF_902728235.1_LCC4_PacBio_genomic         99.32           2.03   \n",
       "561        GCF_902728245.1_LCC21_genomic         99.16           2.03   \n",
       "562         GCF_902728255.1_LCC8_genomic         99.16           2.03   \n",
       "\n",
       "                                        classification  \\\n",
       "2    d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "3    d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "4    d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "7    d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "8    d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "..                                                 ...   \n",
       "558  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "559  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "560  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "561  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "562  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "                                 classification_method  sum_len      N50  \\\n",
       "2    taxonomic classification defined by topology a...  5775745  5514226   \n",
       "3    taxonomic classification defined by topology a...  7417673  7317842   \n",
       "4    taxonomic classification defined by topology a...  7073483  6196881   \n",
       "7    taxonomic classification defined by topology a...  2465519  1285429   \n",
       "8    taxonomic classification defined by topology a...  7795129   203667   \n",
       "..                                                 ...      ...      ...   \n",
       "558  taxonomic classification defined by topology a...  4878525   109520   \n",
       "559  taxonomic classification defined by topology a...  4879546   130068   \n",
       "560  taxonomic classification defined by topology a...  4871565  3986899   \n",
       "561  taxonomic classification defined by topology a...  4884521   148072   \n",
       "562  taxonomic classification defined by topology a...  4886160   168040   \n",
       "\n",
       "     16S ribosomal RNA > 1400 nt  Total_tRNAs host   NCBI_organism_name  \\\n",
       "2                            1.0           45  NaN                  NaN   \n",
       "3                            1.0           52  NaN                  NaN   \n",
       "4                            1.0           71  NaN                  NaN   \n",
       "7                            1.0           48  NaN                  NaN   \n",
       "8                            1.0           52  NaN                  NaN   \n",
       "..                           ...          ...  ...                  ...   \n",
       "558                          1.0           48  NaN  Lentimonas sp. CC19   \n",
       "559                          1.0           49  NaN  Lentimonas sp. CC11   \n",
       "560                          3.0           53  NaN   Lentimonas sp. CC4   \n",
       "561                          1.0           47  NaN  Lentimonas sp. CC21   \n",
       "562                          1.0           47  NaN   Lentimonas sp. CC8   \n",
       "\n",
       "    NCBI_infraspecific_name relation_to_type_material type_of_sample  \n",
       "2                       NaN                       NaN            NaN  \n",
       "3                       NaN                       NaN            NaN  \n",
       "4                       NaN                       NaN            NaN  \n",
       "7                       NaN                       NaN            NaN  \n",
       "8                       NaN                       NaN            NaN  \n",
       "..                      ...                       ...            ...  \n",
       "558                     NaN                       NaN            NaN  \n",
       "559                     NaN                       NaN            NaN  \n",
       "560                     NaN                       NaN            NaN  \n",
       "561                     NaN                       NaN            NaN  \n",
       "562                     NaN                       NaN            NaN  \n",
       "\n",
       "[384 rows x 14 columns]"
      ]
     },
     "execution_count": 432,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ln_comm_sp =  build_line_source('biosample_descr_non_akk.tsv', common_pats, file_pattern=True)\n",
    "ln_sp =  build_line_source('biosample_descr_non_akk.tsv', sp_pats, file_pattern=True)\n",
    "acc_comm_sp = build_acc_host(ln_comm_sp, 'biosample_descr_non_akk.tsv')\n",
    "acc_sp = build_acc_host(ln_sp, 'biosample_descr_non_akk.tsv')\n",
    "add_info_to_summary(merged_non_akk, 'host', acc_comm_sp, acc_sp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 433,
   "id": "cab83c70",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(human            7\n",
       " gallus gallus    1\n",
       " Name: host, dtype: int64,\n",
       " Series([], Name: type_of_sample, dtype: int64),\n",
       " 384)"
      ]
     },
     "execution_count": 433,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(merged_non_akk['host'].value_counts(), \n",
    " merged_non_akk['type_of_sample'].value_counts(), merged_non_akk['type_of_sample'].isna().sum())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "88e0f39b",
   "metadata": {},
   "source": [
    "# End `host`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d8dcee9f",
   "metadata": {},
   "source": [
    "# Set `type_of_sample`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 434,
   "id": "cd200b3e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni 'isolate|isolated|culture|cultured|cultivation' biosample_descr_non_akk.tsv | sort -h | uniq\n",
      "'44:culture\\n45:culture\\n46:culture\\n111:culture\\n158:culture\\n334:culture\\n336:culture\\n337:culture\\n338:culture\\n339:culture\\n350:isolate\\n352:culture\\n353:culture\\n356:culture\\n357:culture\\n360:culture\\n363:isolate'\n",
      "22\n",
      "index for the list in python: 369\n",
      "{'isolated', 'isolate', 'culture'} 22\n"
     ]
    }
   ],
   "source": [
    "regex = (\"'isolate|isolated|culture|cultured|cultivation'\")\n",
    "ln_culture_non_akk = build_line_source('biosample_descr_non_akk.tsv', regex, file_pattern=False)\n",
    "print(repr(ln_culture_non_akk[:200]))\n",
    "acc_culture_non_akk = build_acc_host(ln_culture_non_akk, 'biosample_descr_non_akk.tsv')\n",
    "print(set(acc_culture_non_akk.values()), len(acc_culture_non_akk))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 435,
   "id": "54888c35",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(None,\n",
       "            {'GCA_003054655.1': 'culture',\n",
       "             'GCA_003054665.1': 'culture',\n",
       "             'GCA_003054695.1': 'culture',\n",
       "             'GCA_013391475.1': 'culture',\n",
       "             'GCA_017305195.1': 'culture',\n",
       "             'GCF_001746835.1': 'culture',\n",
       "             'GCF_002954445.1': 'culture',\n",
       "             'GCF_003054705.1': 'culture',\n",
       "             'GCF_003185655.1': 'culture',\n",
       "             'GCF_004118375.1': 'culture',\n",
       "             'GCF_007559335.1': 'isolate',\n",
       "             'GCF_011044495.1': 'culture',\n",
       "             'GCF_011044775.1': 'culture',\n",
       "             'GCF_014230085.1': 'culture',\n",
       "             'GCF_014230145.1': 'culture',\n",
       "             'GCF_014803405.1': 'culture',\n",
       "             'GCF_016595465.1': 'isolated',\n",
       "             'GCF_016595505.1': 'isolated',\n",
       "             'GCF_016595555.1': 'isolated',\n",
       "             'GCF_017310505.1': 'isolate',\n",
       "             'GCF_017310525.1': 'isolate',\n",
       "             'GCF_017310545.1': 'isolate'})"
      ]
     },
     "execution_count": 435,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acc_culture_non_akk"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a18e4fdf",
   "metadata": {},
   "source": [
    "## Search for terms that are indicative of metagenomic samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 436,
   "id": "d66c6da0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni 'metagenome|metagenomic|microbiome|biome|gut|bucal|respiratory|mud|forestsoil|sludge|water|marine|ocean|sediment|seawater|freshwater|freshwaters|feces|fecal|uncultured' biosample_descr_non_akk.tsv | sort -h | uniq\n",
      "'7:freshwater\\n7:Freshwater\\n7:water\\n8:metagenome\\n8:metagenomic\\n9:metagenome\\n9:metagenomic\\n10:biome\\n10:freshwater\\n10:Freshwater\\n10:metagenome\\n10:metagenomic\\n11:marine\\n11:metagenome\\n11:metagenomic\\n11:wate'\n",
      "346\n",
      "index for the list in python: 383\n",
      "{'gut', 'uncultured', 'freshwater', 'metagenome', 'mud', 'seawater', 'metagenomic', 'microbiome', 'water', 'ocean', 'sludge', 'marine', 'sediment', 'feces'} 346\n"
     ]
    }
   ],
   "source": [
    "regex = (\"'metagenome|metagenomic|microbiome|biome|gut|bucal|respiratory|mud|forest\"\n",
    "         \"soil|sludge|water|marine|ocean|sediment|seawater|freshwater|freshwaters|feces|fecal|uncultured'\")\n",
    "ln_meta_non_akk = build_line_source('biosample_descr_non_akk.tsv', regex, file_pattern=False)\n",
    "print(repr(ln_meta_non_akk[:200]))\n",
    "acc_meta_non_akk = build_acc_host(ln_meta_non_akk, 'biosample_descr_non_akk.tsv')\n",
    "print(set(acc_meta_non_akk.values()), len(acc_meta_non_akk))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a8d358f",
   "metadata": {},
   "source": [
    "### Which lines are in common between culture and metagenomic indicative samples?\n",
    "Check those descriptions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 437,
   "id": "fedcefdf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "346 22\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{111, 158, 334, 336, 338, 339, 350, 357, 360, 363, 365, 367, 368, 369}"
      ]
     },
     "execution_count": 437,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_meta = {int(n.split(':')[0]) for n in ln_meta_non_akk.strip().split('\\n')}\n",
    "n_culture = {int(n.split(':')[0]) for n in ln_culture_non_akk.strip().split('\\n')}\n",
    "print(len(n_meta), len(n_culture))\n",
    "n_meta & n_culture"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 438,
   "id": "4af7f6c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('biosample_descr_non_akk.tsv', 'r') as fh:\n",
    "    biosample_non_akk_desc = fh.readlines()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 439,
   "id": "c9780b36",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{111, 158, 334, 336, 338, 339, 350, 357, 360, 363, 365, 367, 368, 369}"
      ]
     },
     "execution_count": 439,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(n_meta & n_culture)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 440,
   "id": "ddd83918",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCF_007559335.1\tSAMN10623051\t53C-WASEF\tfreshwater creek\t2018-04-17\tAustria: Eugendorf\tisolate\t560 m\t47.86 N 13.14 E\ttype strain of Rariglobus hedericola\n",
      "\n",
      "GCF_014230145.1\tSAMN15676578\tJCM23202\tseawater\t2020-05-30\tJapan:Palau\tcell culture\ttype strain of Pelagicoccus albus\n",
      "\n",
      "GCF_014803405.1\tSAMN16205841\tNFK12\tsediment\t2020-06-11\tChina: Weihai\tmix culture\ttype strain of Pelagicoccus enzymogenes\n",
      "\n",
      "GCF_016595465.1\tSAMN17169592\tJCM 18052\t2020-06-01\t0\t0\tActivated sludge\tnot collected\tnot collected\tnot collected\tnot collected\tnot collected\t1\tLuteolibacter yonseiensis sp. nov., isolated from activated sludge using algal metabolites\ttype strain of Luteolibacter yonseiensis\n",
      "\n",
      "GCF_016595505.1\tSAMN17169593\tKCTC 13126\t2020-09-30\t0\t0\tSeawater\tnot collected\tnot collected\tnot collected\tnot collected\tnot collected\t1\tPelagicoccus mobilis gen. nov., sp. nov., Pelagicoccus albus sp. nov. and Pelagicoccus litoralis sp. nov., three novel members of subdivision 4 within the phylum 'Verrucomicrobia', isolated from seawater by in situ cultivation.\ttype strain of Pelagicoccus mobilis\n",
      "\n",
      "GCF_001746835.1\tSAMN04880612\tIG16b\tlimnic surface water\t2012-08-30\tGermany: Wolfenbuettel\tcell culture\ttype strain of Lacunisphaera limnophila\n",
      "\n",
      "GCA_013391475.1\tSAMN15363799\tN1E253\tsediment\t2018-09-11\tChina: Weihai\tmix culture\n",
      "\n",
      "GCF_002954445.1\tSAMN06090799\tSAORIC-165\tdeep sea water\t2011-02-27\tPacific Ocean: Northwestern Pacific Ocean station K2\tculture\t2000\t10\ttype strain of Rubritalea profundi\n",
      "\n",
      "GCF_016595555.1\tSAMN17169589\tKCTC 12870\t2020-09-30\t0\t0\tAcid-treated sediment\tnot collected\tChina: Weihai\tnot collected\tnot collected\tnot collected\t1\tCerasicoccus arenae gen. nov., sp. nov., a carotenoid-producing marine representative of the family Puniceicoccaceae within the phylum 'Verrucomicrobia', isolated from marine sand.\ttype strain of Cerasicoccus arenae\n",
      "\n",
      "GCF_003185655.1\tSAMN08997837\tWN38\tsediment\t2017-09-30\tChina:Weihai\tpure culture\ttype strain of Coraliomargarita sinensis\n",
      "\n",
      "GCF_004118375.1\tSAMN10780907\tTWA-58\twater\t2018-06-24\tTaiwan\tcell culture\ttype material of Oleiharenicola lentus\n",
      "\n",
      "GCF_017310505.1\tSAMN17036178\tIT6\tmud volcano\t2017\tItaly\tisolate\n",
      "\n",
      "GCF_017310525.1\tSAMN17036247\tIT5\tmud volcano\t2017\tItaly\tisolate\n",
      "\n",
      "GCA_017305195.1\tSAMN18061515\t2014-10-30\tbiomaterial containment unit [ENVO:01001816]\tbioreactor biome [ENVO:00002123]\tanthropogenic environmental material [ENVO:0010001]\tSouth Africa: Capetown\t33.95 S 18.46 E\tbiofilm fraction from bioreactor innoculated with Activated Sludge Tailings Effluent Remediation (ASTER); culture conditions: molasses with increasing thiocyanate until 250 mg/L maintanence and no molasses\tSCN18_30_10_14_R3_B_61_17\tThis BioSample is a metagenomic assembly obtained from the bioreactor metagenome BioSample SAMN17179661\t1\t1\tmetagenomic assembly\tbioreactor metagenome\tbiofilm\tmolasses with increasing thiocyanate until 250 mg/L maintanence and no molasses\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for n in (n_meta & n_culture):\n",
    "    n = n - 1\n",
    "    print(biosample_non_akk_desc[n])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "baa4fe49",
   "metadata": {},
   "source": [
    "**From above, `GCA_013391475.1\tSAMN15363799` , `GCA_017305195.1\tSAMN18061515` and `GCF_014803405.1\tSAMN16205841` are metagenomic, the rest seems to be `culture`, so all except these 3 have to be taken off from `acc_meta_non_akk`**\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 441,
   "id": "da1d1c5a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['GCF_007559335.1', 'GCF_014230145.1', 'GCF_016595465.1', 'GCF_016595505.1', 'GCF_001746835.1', 'GCF_002954445.1', 'GCF_016595555.1', 'GCF_003185655.1', 'GCF_004118375.1', 'GCF_017310505.1', 'GCF_017310525.1']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "335"
      ]
     },
     "execution_count": 441,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_asm = ['GCA_013391475.1', 'GCA_017305195.1', 'GCF_014803405.1']\n",
    "asms = []\n",
    "for n in (n_meta & n_culture):\n",
    "    n = n - 1\n",
    "    acc = biosample_non_akk_desc[n].split('\\t')[0]\n",
    "    if biosample_non_akk_desc[n].split('\\t')[0] in meta_asm:\n",
    "        continue\n",
    "    else:\n",
    "        asms.append(acc)\n",
    "print(asms)\n",
    "for asm in asms:\n",
    "    acc_meta_non_akk.pop(asm)\n",
    "len(acc_meta_non_akk)#3346 is the original  length"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 525,
   "id": "d4c94189",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metagenome    332\n",
      "culture        15\n",
      "isolate         4\n",
      "isolated        3\n",
      "Name: type_of_sample, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 525,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "add_info_to_summary(merged_non_akk,'type_of_sample', acc_culture_non_akk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 526,
   "id": "db2afc72",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(metagenome    332\n",
       " culture        22\n",
       " Name: type_of_sample, dtype: int64,\n",
       " 30)"
      ]
     },
     "execution_count": 526,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_to_sp  = {'isolate':'culture', 'isolated':'culture'}\n",
    "merged_non_akk['type_of_sample'].replace(common_to_sp, inplace=True)\n",
    "merged_non_akk['type_of_sample'].value_counts(), merged_non_akk['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4651ab6",
   "metadata": {},
   "source": [
    "### Add these \"sources\" to `type_of_sample`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 444,
   "id": "a6d40e6d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "water          198\n",
      "metagenomic     59\n",
      "sediment        30\n",
      "culture         19\n",
      "sludge          15\n",
      "microbiome       8\n",
      "ocean            5\n",
      "seawater         5\n",
      "metagenome       5\n",
      "marine           3\n",
      "gut              3\n",
      "uncultured       2\n",
      "feces            1\n",
      "mud              1\n",
      "Name: type_of_sample, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>classification</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>sum_len</th>\n",
       "      <th>N50</th>\n",
       "      <th>16S ribosomal RNA &gt; 1400 nt</th>\n",
       "      <th>Total_tRNAs</th>\n",
       "      <th>host</th>\n",
       "      <th>NCBI_organism_name</th>\n",
       "      <th>NCBI_infraspecific_name</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>type_of_sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCA_000155695.1_ASM15569v1_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>1.35</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>5775745</td>\n",
       "      <td>5514226</td>\n",
       "      <td>1.0</td>\n",
       "      <td>45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCA_000242935.3_ASM24293v3_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.23</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7417673</td>\n",
       "      <td>7317842</td>\n",
       "      <td>1.0</td>\n",
       "      <td>52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCA_000243495.3_ASM24349v3_genomic</td>\n",
       "      <td>98.63</td>\n",
       "      <td>0.23</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7073483</td>\n",
       "      <td>6196881</td>\n",
       "      <td>1.0</td>\n",
       "      <td>71</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCA_000526255.1_ASM52625v1_genomic</td>\n",
       "      <td>98.65</td>\n",
       "      <td>0.68</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2465519</td>\n",
       "      <td>1285429</td>\n",
       "      <td>1.0</td>\n",
       "      <td>48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCA_000739615.1_ASM73961v1_genomic</td>\n",
       "      <td>99.86</td>\n",
       "      <td>2.72</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>7795129</td>\n",
       "      <td>203667</td>\n",
       "      <td>1.0</td>\n",
       "      <td>52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>558</th>\n",
       "      <td>GCF_902726605.1_LCC19_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>2.03</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>4878525</td>\n",
       "      <td>109520</td>\n",
       "      <td>1.0</td>\n",
       "      <td>48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lentimonas sp. CC19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>GCF_902728225.1_LCC11_genomic</td>\n",
       "      <td>100.00</td>\n",
       "      <td>2.03</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>4879546</td>\n",
       "      <td>130068</td>\n",
       "      <td>1.0</td>\n",
       "      <td>49</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lentimonas sp. CC11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>GCF_902728235.1_LCC4_PacBio_genomic</td>\n",
       "      <td>99.32</td>\n",
       "      <td>2.03</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>4871565</td>\n",
       "      <td>3986899</td>\n",
       "      <td>3.0</td>\n",
       "      <td>53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lentimonas sp. CC4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>561</th>\n",
       "      <td>GCF_902728245.1_LCC21_genomic</td>\n",
       "      <td>99.16</td>\n",
       "      <td>2.03</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>4884521</td>\n",
       "      <td>148072</td>\n",
       "      <td>1.0</td>\n",
       "      <td>47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lentimonas sp. CC21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>562</th>\n",
       "      <td>GCF_902728255.1_LCC8_genomic</td>\n",
       "      <td>99.16</td>\n",
       "      <td>2.03</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>4886160</td>\n",
       "      <td>168040</td>\n",
       "      <td>1.0</td>\n",
       "      <td>47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lentimonas sp. CC8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>water</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>384 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Genome  Completeness  Contamination  \\\n",
       "2     GCA_000155695.1_ASM15569v1_genomic        100.00           1.35   \n",
       "3     GCA_000242935.3_ASM24293v3_genomic        100.00           0.23   \n",
       "4     GCA_000243495.3_ASM24349v3_genomic         98.63           0.23   \n",
       "7     GCA_000526255.1_ASM52625v1_genomic         98.65           0.68   \n",
       "8     GCA_000739615.1_ASM73961v1_genomic         99.86           2.72   \n",
       "..                                   ...           ...            ...   \n",
       "558        GCF_902726605.1_LCC19_genomic        100.00           2.03   \n",
       "559        GCF_902728225.1_LCC11_genomic        100.00           2.03   \n",
       "560  GCF_902728235.1_LCC4_PacBio_genomic         99.32           2.03   \n",
       "561        GCF_902728245.1_LCC21_genomic         99.16           2.03   \n",
       "562         GCF_902728255.1_LCC8_genomic         99.16           2.03   \n",
       "\n",
       "                                        classification  \\\n",
       "2    d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "3    d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "4    d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "7    d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "8    d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "..                                                 ...   \n",
       "558  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "559  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "560  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "561  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "562  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "                                 classification_method  sum_len      N50  \\\n",
       "2    taxonomic classification defined by topology a...  5775745  5514226   \n",
       "3    taxonomic classification defined by topology a...  7417673  7317842   \n",
       "4    taxonomic classification defined by topology a...  7073483  6196881   \n",
       "7    taxonomic classification defined by topology a...  2465519  1285429   \n",
       "8    taxonomic classification defined by topology a...  7795129   203667   \n",
       "..                                                 ...      ...      ...   \n",
       "558  taxonomic classification defined by topology a...  4878525   109520   \n",
       "559  taxonomic classification defined by topology a...  4879546   130068   \n",
       "560  taxonomic classification defined by topology a...  4871565  3986899   \n",
       "561  taxonomic classification defined by topology a...  4884521   148072   \n",
       "562  taxonomic classification defined by topology a...  4886160   168040   \n",
       "\n",
       "     16S ribosomal RNA > 1400 nt  Total_tRNAs host   NCBI_organism_name  \\\n",
       "2                            1.0           45  NaN                  NaN   \n",
       "3                            1.0           52  NaN                  NaN   \n",
       "4                            1.0           71  NaN                  NaN   \n",
       "7                            1.0           48  NaN                  NaN   \n",
       "8                            1.0           52  NaN                  NaN   \n",
       "..                           ...          ...  ...                  ...   \n",
       "558                          1.0           48  NaN  Lentimonas sp. CC19   \n",
       "559                          1.0           49  NaN  Lentimonas sp. CC11   \n",
       "560                          3.0           53  NaN   Lentimonas sp. CC4   \n",
       "561                          1.0           47  NaN  Lentimonas sp. CC21   \n",
       "562                          1.0           47  NaN   Lentimonas sp. CC8   \n",
       "\n",
       "    NCBI_infraspecific_name relation_to_type_material type_of_sample  \n",
       "2                       NaN                       NaN            NaN  \n",
       "3                       NaN                       NaN            NaN  \n",
       "4                       NaN                       NaN            NaN  \n",
       "7                       NaN                       NaN            NaN  \n",
       "8                       NaN                       NaN            NaN  \n",
       "..                      ...                       ...            ...  \n",
       "558                     NaN                       NaN          water  \n",
       "559                     NaN                       NaN          water  \n",
       "560                     NaN                       NaN          water  \n",
       "561                     NaN                       NaN          water  \n",
       "562                     NaN                       NaN          water  \n",
       "\n",
       "[384 rows x 14 columns]"
      ]
     },
     "execution_count": 444,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "add_info_to_summary(merged_non_akk,'type_of_sample', acc_meta_non_akk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 526,
   "id": "be47e489",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(metagenome    332\n",
       " culture        22\n",
       " Name: type_of_sample, dtype: int64,\n",
       " 30)"
      ]
     },
     "execution_count": 526,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_to_sp  = {'isolate':'culture', 'isolated':'culture'}\n",
    "merged_non_akk['type_of_sample'].replace(common_to_sp, inplace=True)\n",
    "merged_non_akk['type_of_sample'].value_counts(), merged_non_akk['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 496,
   "id": "f053966d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "metagenome    335\n",
       "culture        19\n",
       "Name: type_of_sample, dtype: int64"
      ]
     },
     "execution_count": 496,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_non_akk['type_of_sample'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14e28312",
   "metadata": {},
   "source": [
    "## When adding metagenome after `culture`, we get 19 `culture` instead of 22, check why those are changed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 538,
   "id": "bb9488f2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metagenome    332\n",
      "culture        15\n",
      "isolate         4\n",
      "isolated        3\n",
      "Name: type_of_sample, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 538,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "add_info_to_summary(merged_non_akk,'type_of_sample', acc_culture_non_akk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 539,
   "id": "bfcc98fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(metagenome    332\n",
       " culture        22\n",
       " Name: type_of_sample, dtype: int64,\n",
       " 30)"
      ]
     },
     "execution_count": 539,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_to_sp  = {'isolate':'culture', 'isolated':'culture'}\n",
    "merged_non_akk['type_of_sample'].replace(common_to_sp, inplace=True)\n",
    "merged_non_akk['type_of_sample'].value_counts(), merged_non_akk['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e3afab8",
   "metadata": {},
   "source": [
    "## Create a dictionary with line numbers as keys and list of sources as values and use it with `build_acc_host_fromdict()`\n",
    "\n",
    "All the sources should be lower case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 527,
   "id": "f7faf487",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'biome',\n",
       " 'feces',\n",
       " 'freshwater',\n",
       " 'freshwaters',\n",
       " 'gut',\n",
       " 'marine',\n",
       " 'metagenome',\n",
       " 'metagenomic',\n",
       " 'microbiome',\n",
       " 'mud',\n",
       " 'ocean',\n",
       " 'seawater',\n",
       " 'sediment',\n",
       " 'sludge',\n",
       " 'uncultured',\n",
       " 'water'}"
      ]
     },
     "execution_count": 527,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ln_meta_non_akk_dic = defaultdict(list)\n",
    "for l in ln_meta_non_akk.strip().split('\\n'):\n",
    "    ln_meta_non_akk_dic[int(l.split(':')[0])].append(l.split(':')[1].lower() )\n",
    "# print unique sources\n",
    "{v for ls in ln_meta_non_akk_dic.values() for v in ls}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca2beb1a",
   "metadata": {},
   "source": [
    "## Which lines that have `water` also have `marine` or `freshwater`? "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c199983",
   "metadata": {},
   "source": [
    "## These can be set as `marine water`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 528,
   "id": "bb9d0444",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "34\n",
      "index for the list in python: 383\n",
      "sediment        197\n",
      "marine water    149\n",
      "Name: host, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 528,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_dic = defaultdict()\n",
    "for k,v in ln_meta_non_akk_dic.items():\n",
    "    if 'water' in v and ('marine' in v or 'seawater' in v or 'ocean' in v):\n",
    "        temp_dic[k] = v\n",
    "        #         print(k,v)\n",
    "acc_custom_non_akk = build_acc_host_fromdict(temp_dic, 'biosample_descr_non_akk.tsv', 'marine water')\n",
    "add_info_to_summary(merged_non_akk,'host', acc_custom_non_akk)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5cc441fd",
   "metadata": {},
   "source": [
    "## These can be set as `freshwaters`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 529,
   "id": "77ac3f99",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "169\n",
      "index for the list in python: 349\n",
      "freshwaters     169\n",
      "marine water    149\n",
      "sediment         28\n",
      "Name: host, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 529,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_dic = defaultdict()\n",
    "for k,v in ln_meta_non_akk_dic.items():\n",
    "    if 'freshwater' in v or 'freshwaters' in v:\n",
    "#         print(k,v)\n",
    "        temp_dic[k] = v\n",
    "acc_custom_non_akk = build_acc_host_fromdict(temp_dic, 'biosample_descr_non_akk.tsv', 'freshwaters')\n",
    "add_info_to_summary(merged_non_akk,'host', acc_custom_non_akk)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2abde2c5",
   "metadata": {},
   "source": [
    "## These can be set as `sediment`\n",
    "\n",
    "Here, some samples will go from marine water to sediment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 530,
   "id": "4304a5fa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "33\n",
      "index for the list in python: 366\n",
      "freshwaters     164\n",
      "marine water    149\n",
      "sediment         33\n",
      "Name: host, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 530,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_dic = defaultdict()\n",
    "for k,v in ln_meta_non_akk_dic.items():\n",
    "    if 'sediment' in v:\n",
    "#         print(k,v)\n",
    "        temp_dic[k] = v\n",
    "acc_custom_non_akk = build_acc_host_fromdict(temp_dic, 'biosample_descr_non_akk.tsv', 'sediment')\n",
    "add_info_to_summary(merged_non_akk,'host', acc_custom_non_akk)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08ceeae7",
   "metadata": {},
   "source": [
    "## These can be set as `freshwaters sediment` and `marine water sediment`\n",
    "\n",
    "Here, some samples will go from marine water to sediment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 531,
   "id": "b75c654b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5\n",
      "index for the list in python: 232\n",
      "freshwaters             164\n",
      "marine water            149\n",
      "sediment                 28\n",
      "freshwaters sediment      5\n",
      "Name: host, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 531,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_dic = defaultdict()\n",
    "for k,v in ln_meta_non_akk_dic.items():\n",
    "    if 'sediment' in v  and ('freshwater' in v or 'freshwaters' in v):\n",
    "#         print(k,v)\n",
    "        temp_dic[k] = v\n",
    "acc_custom_non_akk = build_acc_host_fromdict(temp_dic, 'biosample_descr_non_akk.tsv', 'freshwaters sediment')\n",
    "add_info_to_summary(merged_non_akk,'host', acc_custom_non_akk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 532,
   "id": "8ac0b306",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5\n",
      "index for the list in python: 366\n",
      "freshwaters              164\n",
      "marine water             149\n",
      "sediment                  23\n",
      "marine water sediment      5\n",
      "freshwaters sediment       5\n",
      "Name: host, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 532,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_dic = defaultdict()\n",
    "for k,v in ln_meta_non_akk_dic.items():\n",
    "    if 'sediment' in v  and ('seawater' in v or 'marine' in v):\n",
    "#         print(k,v)\n",
    "        temp_dic[k] = v\n",
    "acc_custom_non_akk = build_acc_host_fromdict(temp_dic, 'biosample_descr_non_akk.tsv', 'marine water sediment')\n",
    "add_info_to_summary(merged_non_akk,'host', acc_custom_non_akk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 540,
   "id": "85d20b6c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(freshwaters              164\n",
       " marine water             149\n",
       " sediment                  23\n",
       " marine water sediment      5\n",
       " freshwaters sediment       5\n",
       " Name: host, dtype: int64,\n",
       " 38)"
      ]
     },
     "execution_count": 540,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_non_akk['host'].value_counts(), merged_non_akk['host'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 541,
   "id": "dce17d05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(metagenome    332\n",
       " culture        22\n",
       " Name: type_of_sample, dtype: int64,\n",
       " 30)"
      ]
     },
     "execution_count": 541,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_non_akk['type_of_sample'].value_counts(), merged_non_akk['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 535,
   "id": "21cace09",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_non_akk.to_csv(qc_path.joinpath('merged_non_akk.tsv'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 548,
   "id": "583d9f8d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "38"
      ]
     },
     "execution_count": 548,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nan_type = merged_non_akk[merged_non_akk['type_of_sample'].isna()].loc[:, 'Genome'].str.extract('(GC[AF]_[0-9]{9}\\.[1-9]{1,2})').dropna().values.flatten()\n",
    "nan_host = merged_non_akk[merged_non_akk['host'].isna()].loc[:, 'Genome'].str.extract('(GC[AF]_[0-9]{9}\\.[1-9]{1,2})').dropna().values.flatten()\n",
    "len(set(nan_host) | set(nan_type))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 649,
   "id": "f1fd2e4e",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n",
      "^C\n"
     ]
    }
   ],
   "source": [
    "fh = open('bioproject_descr_miss_biosam_non_akk.tsv', 'a')\n",
    "asms = set(nan_type.tolist() + nan_host.tolist())\n",
    "for idx,row in asm_biosam_bioproj.set_index('assembly_accession').loc[asms, :].iterrows():\n",
    "    bp = row['bioproject']\n",
    "    desc = ! esearch -db bioproject -query $bp | efetch -format docsum |  xtract -pattern DocumentSummary -element Project_Name,Project_Title,Project_Description,Project_Target_Scope\n",
    "    desc_ls = desc.get_spstr().split('\\t')\n",
    "    desc_ls.insert(0, row['bioproject'])\n",
    "    desc_ls.insert(0, row['biosample'])\n",
    "    desc_ls.insert(0, idx)#assembly accession\n",
    "    print('\\t'.join(desc_ls), file=fh)\n",
    "fh.close()    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 582,
   "id": "6d7a72ab",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCF_014651675.1\tSAMD00245577\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000817245.1\tSAMN02996356\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_000243495.3\tSAMN02261400\tPRJNA62139\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000379365.1\tSAMN02256457\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_900141815.1\tSAMN02745181\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_004341915.1\tSAMN10866388\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000172555.1\tSAMN00000631\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_014201715.1\tSAMN14908355\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_900104925.1\tSAMN05444173\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_003054665.1\tSAMN08724138\tPRJNA438699\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_003054655.1\tSAMN08724137\tPRJNA438699\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_004364345.1\tSAMN02745162\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_007992435.1\tSAMD00172670\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_014230085.1\tSAMN15676579\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_900105685.1\tSAMN05444156\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_000739655.1\tSAMD00036768\tPRJDB1469\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_000155695.1\tSAMN02436317\tPRJNA19391\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_902143385.2\tSAMEA5600708\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_011044775.1\tSAMN14102279\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_001613545.1\tSAMD00048637\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000297415.1\tSAMEA2272587\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000172155.1\tSAMN02436175\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_902143375.2\tSAMEA5600707\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000019965.1\tSAMN00000285\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_900167535.1\tSAMN02745166\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_007475525.1\tSAMN08451303\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_014651635.1\tSAMD00245575\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000019665.1\tSAMN02604183\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000173075.1\tSAMN00000678\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_016811075.1\tSAMN17711332\tPRJNA698280\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_017310545.1\tSAMN17073650\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_000739615.1\tSAMD00036766\tPRJDB1467\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_003054695.1\tSAMN08724136\tPRJNA438699\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_000526255.1\tSAMN02584925\tPRJNA185245\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_003054705.1\tSAMN08724139\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_000242935.3\tSAMN02261410\tPRJNA63271\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_011044495.1\tSAMN14143996\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_012913485.1\tSAMN14642216\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_014651675.1\tSAMD00245577\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000817245.1\tSAMN02996356\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_000243495.3\tSAMN02261400\tPRJNA62139\tOpitutaceae bacterium TAV1\tOpitutaceae bacterium TAV1 genome sequencing\tThis strain will be used for comparative analysis.\n",
      "\n",
      "GCF_000379365.1\tSAMN02256457\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_900141815.1\tSAMN02745181\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_004341915.1\tSAMN10866388\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000172555.1\tSAMN00000631\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_014201715.1\tSAMN14908355\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_900104925.1\tSAMN05444173\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_003054665.1\tSAMN08724138\tPRJNA438699\tGenomic and phenotypic characterization of four novel plant-associated Verrucomicrobia indicate a new subdivision with one cultivated member\tGenomic and physiological description of new isolates (Verrucomicrobia) from O.sativa and O.longistaminata. Analysis of physiology and genomic context with focus on microbe-plant interactions.\n",
      "\n",
      "GCA_003054655.1\tSAMN08724137\tPRJNA438699\tGenomic and phenotypic characterization of four novel plant-associated Verrucomicrobia indicate a new subdivision with one cultivated member\tGenomic and physiological description of new isolates (Verrucomicrobia) from O.sativa and O.longistaminata. Analysis of physiology and genomic context with focus on microbe-plant interactions.\n",
      "\n",
      "GCF_004364345.1\tSAMN02745162\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_007992435.1\tSAMD00172670\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_014230085.1\tSAMN15676579\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_900105685.1\tSAMN05444156\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_000739655.1\tSAMD00036768\tPRJDB1469\tVerrucomicrobium sp. BvORR106\tGenome sequence of Verrucomicrobium sp. BvORR106\tA sequencing for draft genome of a Verrucomicrobia bacgterium isolated from main roots of sugar beet was conducted in ordet to analyze the functionality of plant-associated Verrucomicrobia bacteria. The sequence data will be used for clarifying the molecular interaction between a plant and a Verrucomicrobia bacterium\n",
      "\n",
      "GCA_000155695.1\tSAMN02436317\tPRJNA19391\tVerrucomicrobiae bacterium DG1235\tA culturable pelagic Verrucomicrobia isolated from a dinoflagellate\tVerrucomicrobiae bacterium DG1235. The genome analysis of this strain will allow a comprehensive investigation of the metabolic potential and involvement in biogeochemical cycling of this abundant, but underrepresented group of organisms in the marine environment.\n",
      "\n",
      "GCF_902143385.2\tSAMEA5600708\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_011044775.1\tSAMN14102279\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_001613545.1\tSAMD00048637\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000297415.1\tSAMEA2272587\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000172155.1\tSAMN02436175\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_902143375.2\tSAMEA5600707\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000019965.1\tSAMN00000285\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_900167535.1\tSAMN02745166\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_007475525.1\tSAMN08451303\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_014651635.1\tSAMD00245575\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000019665.1\tSAMN02604183\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_000173075.1\tSAMN00000678\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_016811075.1\tSAMN17711332\tPRJNA698280\tVerrucomicrobiaceae bacterium 5K15\tVerrucomicrobiaceae bacterium 5K15 Genome sequencing and assembly\tUsed for the study of Verrucomicrobia\n",
      "\n",
      "GCF_017310545.1\tSAMN17073650\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_000739615.1\tSAMD00036766\tPRJDB1467\tHaloferula sp. BvORR071\tGenome sequence of Haloferula sp. BvORR071\tA sequencing for draft genome of a Verrucomicrobia bacgterium isolated from main roots of sugar beet was conducted in ordet to analyze the functionality of plant-associated Verrucomicrobia bacteria. The sequence data will be used for clarifying the molecular interaction between a plant and a Verrucomicrobia bacterium\n",
      "\n",
      "GCA_003054695.1\tSAMN08724136\tPRJNA438699\tGenomic and phenotypic characterization of four novel plant-associated Verrucomicrobia indicate a new subdivision with one cultivated member\tGenomic and physiological description of new isolates (Verrucomicrobia) from O.sativa and O.longistaminata. Analysis of physiology and genomic context with focus on microbe-plant interactions.\n",
      "\n",
      "GCA_000526255.1\tSAMN02584925\tPRJNA185245\tVerrucomicrobia bacterium LP2A\tVerrucomicrobia bacterium LP2A Genome sequencing and assembly\tRevising methanotrophy: a comprehensive genomic probing of the unexpected genetic and metabolic diversity of aerobic methane consuming bacteria.\n",
      "\n",
      "GCF_003054705.1\tSAMN08724139\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCA_000242935.3\tSAMN02261410\tPRJNA63271\tOpitutaceae bacterium TAV5\tOpitutaceae bacterium TAV5 genome sequencing project\tOpitutaceae bacterium TAV5. This strain will be used for comparative genome analysis.\n",
      "\n",
      "GCF_011044495.1\tSAMN14143996\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n",
      "GCF_012913485.1\tSAMN14642216\tPRJNA224116\tRefseq Prokaryotic Genome Annotation Project\tNCBI has developed an automatic annotation pipeline that combines ab initio gene prediction algorithms with homology based methods. See more details here. Historically RefSeq prokaryotic genomes retained on author submitted annotation. Annotation from different submitters varies in quality resulting in the inconsistent annotation even in closely related genomes. The NCBI Prokaryotic Annotation Pipeline can produce consistent high quality automatic annotation.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "with open('bioproject_descr_miss_biosam_non_akk.tsv', 'r') as fh:\n",
    "        bioproj_desc_miss = fh.readlines()\n",
    "for l in bioproj_desc_miss:\n",
    "    print(l)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44a1f177",
   "metadata": {},
   "source": [
    "## From the above, most but no all the information is from projects referred as *Refseq Prokaryotic Genome Annotation Project*"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f961483e",
   "metadata": {},
   "source": [
    "# BioProject `host`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 588,
   "id": "5ffa68db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni -f /home/mibu/common_names_ncbi.list bioproject_descr_miss_biosam_non_akk.tsv | sort -h | uniq\n",
      "rg -woni -f /home/mibu/amniota_only_sp.list bioproject_descr_miss_biosam_non_akk.tsv | sort -h | uniq\n",
      " \n"
     ]
    },
    {
     "ename": "AssertionError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-588-eeb9a28c4375>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mln_sp_bp\u001b[0m \u001b[0;34m=\u001b[0m  \u001b[0mbuild_line_source\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bioproject_descr_miss_biosam_non_akk.tsv'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msp_pats\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfile_pattern\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mln_comm_sp_bp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mln_sp_bp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0;32massert\u001b[0m \u001b[0mln_comm_sp_bp\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m \u001b[0;32massert\u001b[0m \u001b[0mln_sp_bp\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;31m# acc_comm_sp_bp = build_acc_host(ln_comm_sp_bp, 'bioproject_descr_miss_biosam_non_akk.tsv')\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAssertionError\u001b[0m: "
     ]
    }
   ],
   "source": [
    "ln_comm_sp_bp =  build_line_source('bioproject_descr_miss_biosam_non_akk.tsv', common_pats, file_pattern=True)\n",
    "ln_sp_bp =  build_line_source('bioproject_descr_miss_biosam_non_akk.tsv', sp_pats, file_pattern=True)\n",
    "print(ln_comm_sp_bp, ln_sp_bp)\n",
    "assert ln_comm_sp_bp != ''\n",
    "assert ln_sp_bp != ''\n",
    "# acc_comm_sp_bp = build_acc_host(ln_comm_sp_bp, 'bioproject_descr_miss_biosam_non_akk.tsv')\n",
    "# acc_sp_bp = build_acc_host(ln_sp_bp, 'bioproject_descr_miss_biosam_non_akk.tsv')\n",
    "# add_info_to_summary(merged_non_akk, 'host', acc_comm_sp_bp, acc_sp_bp)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c839c60e",
   "metadata": {},
   "source": [
    "**So no host/species information from bioproject**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e50a3d04",
   "metadata": {},
   "source": [
    "# Set `type_of_sample`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 603,
   "id": "d0cd392c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni 'isolate|isolated|isolates|culture|cultured|cultivation|cultivated' bioproject_descr_miss_biosam_non_akk.tsv | sort -h | uniq\n",
      "'48:cultivated\\n48:isolates\\n49:cultivated\\n49:isolates\\n54:isolated\\n55:isolated\\n70:isolated\\n71:cultivated\\n71:isolates\\n'\n",
      "6\n",
      "index for the list in python: 70\n",
      "{'isolated', 'isolates'} 6\n"
     ]
    }
   ],
   "source": [
    "regex = (\"'isolate|isolated|isolates|monoisolate|culture|cultured|cultivation|cultivated'\")\n",
    "ln_culture_non_akk_bp = build_line_source('bioproject_descr_miss_biosam_non_akk.tsv', regex, file_pattern=False)\n",
    "print(repr(ln_culture_non_akk_bp[:200]))\n",
    "acc_culture_non_akk_bp = build_acc_host(ln_culture_non_akk_bp, 'bioproject_descr_miss_biosam_non_akk.tsv')\n",
    "print(set(acc_culture_non_akk_bp.values()), len(acc_culture_non_akk_bp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 604,
   "id": "060fb7a7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metagenome    332\n",
      "culture        19\n",
      "isolated        3\n",
      "isolates        3\n",
      "Name: type_of_sample, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 604,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "add_info_to_summary(merged_non_akk,'type_of_sample', acc_culture_non_akk_bp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 617,
   "id": "e50bd95d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(metagenome    332\n",
       " culture        25\n",
       " Name: type_of_sample, dtype: int64,\n",
       " 27)"
      ]
     },
     "execution_count": 617,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_to_sp  = {'isolate':'culture', 'isolated':'culture', 'isolates':'culture'}\n",
    "merged_non_akk['type_of_sample'].replace(common_to_sp, inplace=True)\n",
    "merged_non_akk['type_of_sample'].value_counts(), merged_non_akk['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 606,
   "id": "2a5f5ab3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni 'metagenome|metagenomic|microbiome|biome|gut|bucal|respiratory|mud|forestsoil|sludge|water|marine|ocean|sediment|seawater|freshwater|freshwaters|feces|fecal|uncultured' bioproject_descr_miss_biosam_non_akk.tsv | sort -h | uniq\n",
      "'55:marine\\n'\n"
     ]
    }
   ],
   "source": [
    "regex = (\"'metagenome|metagenomic|microbiome|biome|gut|bucal|respiratory|mud|forest\"\n",
    "         \"soil|sludge|water|marine|ocean|sediment|seawater|freshwater|freshwaters|feces|fecal|uncultured'\")\n",
    "ln_meta_non_akk_bp = build_line_source('bioproject_descr_miss_biosam_non_akk.tsv', regex, file_pattern=False)\n",
    "print(repr(ln_meta_non_akk_bp))\n",
    "# acc_meta_non_akk_bp = build_acc_host(ln_meta_non_akk, 'bioproject_descr_miss_biosam_non_akk.tsv')\n",
    "# print(set(acc_meta_non_akk_bp.values()), len(acc_meta_non_akk_bp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 609,
   "id": "83210d60",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 609,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(ln_meta_non_akk_bp.strip().splitlines())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02bb2ec8",
   "metadata": {},
   "source": [
    "Only one hit, can be added manually"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36c3297c",
   "metadata": {},
   "source": [
    "### Assign those samples that are representatives, `relation_to_type_material == assembly from type material` to `type_of_sample` = `culture`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 619,
   "id": "d3615d3a",
   "metadata": {},
   "outputs": [],
   "source": [
    "mask = ~merged_non_akk['relation_to_type_material'].isna()\n",
    "merged_non_akk.loc[mask, 'type_of_sample'] = 'culture'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 621,
   "id": "0389fa45",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_non_akk.to_csv(qc_path.joinpath('merged_non_akk.tsv'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 627,
   "id": "640fa5e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1158218 entries, 0 to 1158217\n",
      "Data columns (total 23 columns):\n",
      " #   Column                     Non-Null Count    Dtype \n",
      "---  ------                     --------------    ----- \n",
      " 0   Genome                     1158163 non-null  object\n",
      " 1   assembly_accession         1158218 non-null  object\n",
      " 2   bioproject                 1158214 non-null  object\n",
      " 3   biosample                  1158213 non-null  object\n",
      " 4   wgs_master                 1104090 non-null  object\n",
      " 5   refseq_category            1158218 non-null  object\n",
      " 6   taxid                      1158218 non-null  int64 \n",
      " 7   species_taxid              1158218 non-null  int64 \n",
      " 8   NCBI_organism_name         1158218 non-null  object\n",
      " 9   NCBI_infraspecific_name    998033 non-null   object\n",
      " 10  isolate                    182630 non-null   object\n",
      " 11  version_status             1158218 non-null  object\n",
      " 12  assembly_level             1158218 non-null  object\n",
      " 13  release_type               1158218 non-null  object\n",
      " 14  genome_rep                 1158218 non-null  object\n",
      " 15  seq_rel_date               1158218 non-null  object\n",
      " 16  asm_name                   1158217 non-null  object\n",
      " 17  submitter                  1158210 non-null  object\n",
      " 18  gbrs_paired_asm            1158218 non-null  object\n",
      " 19  paired_asm_comp            1158218 non-null  object\n",
      " 20  ftp_path                   1158218 non-null  object\n",
      " 21  excluded_from_refseq       726985 non-null   object\n",
      " 22  relation_to_type_material  32556 non-null    object\n",
      "dtypes: int64(2), object(21)\n",
      "memory usage: 203.2+ MB\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array([None, 'assembly from type material',\n",
       "       'assembly from synonym type material',\n",
       "       'assembly from pathotype material',\n",
       "       'assembly designated as neotype', 'assembly designated as reftype'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 627,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_df.info()\n",
    "ncbi_df['relation_to_type_material'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 647,
   "id": "942b56d5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>classification</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>sum_len</th>\n",
       "      <th>N50</th>\n",
       "      <th>16S ribosomal RNA &gt; 1400 nt</th>\n",
       "      <th>Total_tRNAs</th>\n",
       "      <th>host</th>\n",
       "      <th>NCBI_organism_name</th>\n",
       "      <th>NCBI_infraspecific_name</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>type_of_sample</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genome</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_000020225.1_ASM2022v1_genomic</th>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2664102</td>\n",
       "      <td>2664102</td>\n",
       "      <td>3.0</td>\n",
       "      <td>54</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Akkermansia muciniphila ATCC BAA-835</td>\n",
       "      <td>strain=ATCC BAA-835</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>culture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_008000975.1_ASM800097v1_genomic</th>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2664043</td>\n",
       "      <td>2664043</td>\n",
       "      <td>3.0</td>\n",
       "      <td>54</td>\n",
       "      <td>homo sapiens</td>\n",
       "      <td>Akkermansia muciniphila</td>\n",
       "      <td>strain=DSM 22959</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>culture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_017504145.1_ASM1750414v1_genomic</th>\n",
       "      <td>97.96</td>\n",
       "      <td>0.00</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>2664051</td>\n",
       "      <td>2664051</td>\n",
       "      <td>3.0</td>\n",
       "      <td>54</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Akkermansia muciniphila ATCC BAA-835</td>\n",
       "      <td>strain=ATCC BAA-835</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>culture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001683795.1_ASM168379v1_genomic</th>\n",
       "      <td>91.84</td>\n",
       "      <td>4.76</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>3157849</td>\n",
       "      <td>127311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>42</td>\n",
       "      <td>reticulated python</td>\n",
       "      <td>Akkermansia glycaniphila</td>\n",
       "      <td>strain=Pyt</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>culture</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Completeness  Contamination  \\\n",
       "Genome                                                              \n",
       "GCF_000020225.1_ASM2022v1_genomic            97.96           0.00   \n",
       "GCF_008000975.1_ASM800097v1_genomic          97.96           0.00   \n",
       "GCF_017504145.1_ASM1750414v1_genomic         97.96           0.00   \n",
       "GCF_001683795.1_ASM168379v1_genomic          91.84           4.76   \n",
       "\n",
       "                                                                         classification  \\\n",
       "Genome                                                                                    \n",
       "GCF_000020225.1_ASM2022v1_genomic     d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "GCF_008000975.1_ASM800097v1_genomic   d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "GCF_017504145.1_ASM1750414v1_genomic  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "GCF_001683795.1_ASM168379v1_genomic   d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "                                                                  classification_method  \\\n",
       "Genome                                                                                    \n",
       "GCF_000020225.1_ASM2022v1_genomic     taxonomic classification defined by topology a...   \n",
       "GCF_008000975.1_ASM800097v1_genomic   taxonomic classification defined by topology a...   \n",
       "GCF_017504145.1_ASM1750414v1_genomic  taxonomic classification defined by topology a...   \n",
       "GCF_001683795.1_ASM168379v1_genomic   taxonomic classification defined by topology a...   \n",
       "\n",
       "                                      sum_len      N50  \\\n",
       "Genome                                                   \n",
       "GCF_000020225.1_ASM2022v1_genomic     2664102  2664102   \n",
       "GCF_008000975.1_ASM800097v1_genomic   2664043  2664043   \n",
       "GCF_017504145.1_ASM1750414v1_genomic  2664051  2664051   \n",
       "GCF_001683795.1_ASM168379v1_genomic   3157849   127311   \n",
       "\n",
       "                                      16S ribosomal RNA > 1400 nt  \\\n",
       "Genome                                                              \n",
       "GCF_000020225.1_ASM2022v1_genomic                             3.0   \n",
       "GCF_008000975.1_ASM800097v1_genomic                           3.0   \n",
       "GCF_017504145.1_ASM1750414v1_genomic                          3.0   \n",
       "GCF_001683795.1_ASM168379v1_genomic                           NaN   \n",
       "\n",
       "                                      Total_tRNAs                host  \\\n",
       "Genome                                                                  \n",
       "GCF_000020225.1_ASM2022v1_genomic              54                 NaN   \n",
       "GCF_008000975.1_ASM800097v1_genomic            54        homo sapiens   \n",
       "GCF_017504145.1_ASM1750414v1_genomic           54                 NaN   \n",
       "GCF_001683795.1_ASM168379v1_genomic            42  reticulated python   \n",
       "\n",
       "                                                        NCBI_organism_name  \\\n",
       "Genome                                                                       \n",
       "GCF_000020225.1_ASM2022v1_genomic     Akkermansia muciniphila ATCC BAA-835   \n",
       "GCF_008000975.1_ASM800097v1_genomic                Akkermansia muciniphila   \n",
       "GCF_017504145.1_ASM1750414v1_genomic  Akkermansia muciniphila ATCC BAA-835   \n",
       "GCF_001683795.1_ASM168379v1_genomic               Akkermansia glycaniphila   \n",
       "\n",
       "                                     NCBI_infraspecific_name  \\\n",
       "Genome                                                         \n",
       "GCF_000020225.1_ASM2022v1_genomic        strain=ATCC BAA-835   \n",
       "GCF_008000975.1_ASM800097v1_genomic         strain=DSM 22959   \n",
       "GCF_017504145.1_ASM1750414v1_genomic     strain=ATCC BAA-835   \n",
       "GCF_001683795.1_ASM168379v1_genomic               strain=Pyt   \n",
       "\n",
       "                                        relation_to_type_material  \\\n",
       "Genome                                                              \n",
       "GCF_000020225.1_ASM2022v1_genomic     assembly from type material   \n",
       "GCF_008000975.1_ASM800097v1_genomic   assembly from type material   \n",
       "GCF_017504145.1_ASM1750414v1_genomic  assembly from type material   \n",
       "GCF_001683795.1_ASM168379v1_genomic   assembly from type material   \n",
       "\n",
       "                                     type_of_sample  \n",
       "Genome                                               \n",
       "GCF_000020225.1_ASM2022v1_genomic           culture  \n",
       "GCF_008000975.1_ASM800097v1_genomic         culture  \n",
       "GCF_017504145.1_ASM1750414v1_genomic        culture  \n",
       "GCF_001683795.1_ASM168379v1_genomic         culture  "
      ]
     },
     "execution_count": 647,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_nans_type = merged_akk[~merged_akk['relation_to_type_material'].isna()].loc[:, 'Genome'].values.flatten()\n",
    "merged_akk.set_index('Genome').loc[not_nans_type]\n",
    "# not_nans_type\n",
    "# ncbi_df.set_index('assembly_accession').loc[not_nans_type].loc[:, 'relation_to_type_material'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 659,
   "id": "f59e6b3d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>biosample</th>\n",
       "      <th>bioproject</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>358</th>\n",
       "      <td>GCF_000172555.1</td>\n",
       "      <td>SAMN00000631</td>\n",
       "      <td>PRJNA224116</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession     biosample   bioproject\n",
       "358    GCF_000172555.1  SAMN00000631  PRJNA224116"
      ]
     },
     "execution_count": 659,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "asm_biosam_bioproj[asm_biosam_bioproj['assembly_accession'] == 'GCF_000172555.1']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b92a5327",
   "metadata": {},
   "source": [
    "### Sometimes, an assembly may have more than one BioProject. The ones from the ncbi summary table are not neccesarily the most nformative. Check the project id inside `GB_BioProjects`\n",
    "\n",
    "Many times, the Bioproj from the summary tables,  refer to a project about ncbi annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 660,
   "id": "bde696e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PRJNA20839\n"
     ]
    }
   ],
   "source": [
    "! esearch -db assembly -query GCF_000172555.1 | efetch -format docsum |xtract -pattern DocumentSummary -block GB_BioProjects -element BioprojectAccn"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc309320",
   "metadata": {},
   "source": [
    "### Add a colummn with the `GB_BioProjects/BioprojectAccn` to `asm_biosam_bioproj`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 662,
   "id": "041753d5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>biosample</th>\n",
       "      <th>bioproject</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCA_000155695.1</td>\n",
       "      <td>SAMN02436317</td>\n",
       "      <td>PRJNA19391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCA_000242935.3</td>\n",
       "      <td>SAMN02261410</td>\n",
       "      <td>PRJNA63271</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  assembly_accession     biosample  bioproject\n",
       "0    GCA_000155695.1  SAMN02436317  PRJNA19391\n",
       "1    GCA_000242935.3  SAMN02261410  PRJNA63271"
      ]
     },
     "execution_count": 662,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "asm_biosam_bioproj.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 669,
   "id": "8513257f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['PRJNA19391']\n",
      "['PRJNA63271']\n",
      "['PRJNA62139']\n",
      "['PRJEB726']\n",
      "['PRJEB840']\n",
      "['PRJNA185245']\n",
      "['PRJDB1467']\n",
      "['PRJDB1469']\n",
      "['PRJNA279927']\n",
      "['PRJNA297582']\n",
      "['PRJNA279279']\n",
      "['PRJNA321218']\n",
      "['PRJNA238866']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA342151']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA348753']\n",
      "['PRJNA391943']\n",
      "['PRJNA391943']\n",
      "['PRJNA391943']\n",
      "['PRJNA391943']\n",
      "['PRJNA391943']\n",
      "['PRJNA391943']\n",
      "['PRJNA362739']\n",
      "['PRJNA362739']\n",
      "['PRJNA362739']\n",
      "['PRJNA438699']\n",
      "['PRJNA438699']\n",
      "['PRJNA438699']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA386568']\n",
      "['PRJNA449266']\n",
      "['PRJNA449266']\n",
      "['PRJNA454827']\n",
      "['PRJNA417962']\n",
      "['PRJNA434545']\n",
      "['PRJNA434545']\n",
      "['PRJNA448579']\n",
      "['PRJNA448579']\n",
      "['PRJNA448579']\n",
      "['PRJNA448579']\n",
      "['PRJNA514088']\n",
      "['PRJNA531348']\n",
      "['PRJNA453733']\n",
      "['PRJNA453733']\n",
      "['PRJNA453733']\n",
      "['PRJNA453733']\n",
      "['PRJNA453733']\n",
      "['PRJNA453733']\n",
      "['PRJNA512237']\n",
      "['PRJNA492716']\n",
      "['PRJNA492716']\n",
      "['PRJNA480789']\n",
      "['PRJNA363633']\n",
      "['PRJNA512237']\n",
      "['PRJNA544874']\n",
      "['PRJNA525692']\n",
      "['PRJNA525692']\n",
      "['PRJNA428417']\n",
      "['PRJNA428417']\n",
      "['PRJNA604877']\n",
      "['PRJNA605358']\n",
      "['PRJNA480137']\n",
      "['PRJNA480137']\n",
      "['PRJNA480137']\n",
      "['PRJNA480137']\n",
      "['PRJNA480137']\n",
      "['PRJNA480137']\n",
      "['PRJNA480137']\n",
      "['PRJNA480137']\n",
      "['PRJNA480137']\n",
      "['PRJNA480137']\n",
      "['PRJNA598059']\n",
      "['PRJNA545004']\n",
      "['PRJNA602310']\n",
      "['PRJNA602310']\n",
      "['PRJNA522654']\n",
      "['PRJNA522654']\n",
      "['PRJNA623061']\n",
      "['PRJNA482655']\n",
      "['PRJNA641721']\n",
      "['PRJNA630822']\n",
      "['PRJDB6644']\n",
      "['PRJNA613976']\n",
      "['PRJNA649215']\n",
      "['PRJNA649215']\n",
      "['PRJNA574041']\n",
      "['PRJNA574041']\n",
      "['PRJNA574041']\n",
      "['PRJNA562766']\n",
      "['PRJNA562766']\n",
      "['PRJNA562766']\n",
      "['PRJNA562766']\n",
      "['PRJNA562766']\n",
      "['PRJNA562766']\n",
      "['PRJNA562766']\n",
      "['PRJNA562766']\n",
      "['PRJNA562766']\n",
      "['PRJNA576469']\n",
      "['PRJNA576469']\n",
      "['PRJNA576469']\n",
      "['PRJNA627084']\n",
      "['PRJNA627084']\n",
      "['PRJNA488180']\n",
      "['PRJNA623111']\n",
      "['PRJNA640378']\n",
      "['PRJNA640378']\n",
      "['PRJNA640378']\n",
      "['PRJNA640378']\n",
      "['PRJNA640378']\n",
      "['PRJNA640378']\n",
      "['PRJNA640378']\n",
      "['PRJNA640378']\n",
      "['PRJNA663590']\n",
      "['PRJNA663590']\n",
      "['PRJNA688125']\n",
      "['PRJNA688125']\n",
      "['PRJNA629478']\n",
      "['PRJNA629478']\n",
      "['PRJNA638805']\n",
      "['PRJNA592128']\n",
      "['PRJNA592128']\n",
      "['PRJNA592128']\n",
      "['PRJNA592128']\n",
      "['PRJNA592128']\n",
      "['PRJNA592128']\n",
      "['PRJNA592128']\n",
      "['PRJNA592128']\n",
      "['PRJNA698280']\n",
      "['PRJNA523022']\n",
      "['PRJNA523022']\n",
      "['PRJNA523022']\n",
      "['PRJNA523022']\n",
      "['PRJNA523022']\n",
      "['PRJNA690107']\n",
      "['PRJNA690107']\n",
      "['PRJNA629336']\n",
      "['PRJNA704939']\n",
      "['PRJEB26432']\n",
      "['PRJEB26432']\n",
      "['PRJEB26432']\n",
      "['PRJEB26432']\n",
      "['PRJEB31003']\n",
      "['PRJEB31003']\n",
      "['PRJEB31003']\n",
      "['PRJEB31003']\n",
      "['PRJEB31003']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB38681']\n",
      "['PRJEB33338']\n",
      "['PRJEB38020']\n",
      "['PRJEB41592']\n",
      "['PRJEB41592']\n",
      "['PRJEB41592']\n",
      "['PRJEB37358']\n",
      "['PRJEB37358']\n",
      "['PRJEB37358']\n",
      "['PRJEB37358']\n",
      "['PRJEB37358']\n",
      "['PRJEB37358']\n",
      "['PRJEB37358']\n",
      "['PRJEB37358']\n",
      "['PRJEB37358']\n",
      "['PRJEB37358']\n",
      "['PRJEB37358']\n",
      "['PRJNA28995']\n",
      "['PRJNA19989']\n",
      "['PRJNA20089']\n",
      "['PRJNA33365']\n",
      "['PRJNA18333']\n",
      "['PRJNA10620']\n",
      "['PRJNA20839']\n",
      "['PRJNA28531']\n",
      "['PRJEA85607']\n",
      "['PRJNA181370']\n",
      "['PRJNA165235']\n",
      "['PRJEB6027']\n",
      "['PRJNA169117']\n",
      "['PRJNA268324']\n",
      "['PRJNA310109']\n",
      "['PRJNA310109']\n",
      "['PRJNA268325']\n",
      "['PRJDB4647']\n",
      "['PRJNA80821']\n",
      "['PRJNA317592']\n",
      "['PRJNA319155']\n",
      "['PRJNA377666']\n",
      "['PRJNA289613']\n",
      "['PRJNA404023']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA331216']\n",
      "['PRJNA355604']\n",
      "['PRJNA438699']\n",
      "['PRJNA454064']\n",
      "['PRJNA417186']\n",
      "['PRJNA417203']\n",
      "['PRJNA417208']\n",
      "['PRJNA417215']\n",
      "['PRJNA429085']\n",
      "['PRJNA429088']\n",
      "['PRJNA429277']\n",
      "['PRJNA429282']\n",
      "['PRJNA429288']\n",
      "['PRJNA429289']\n",
      "['PRJNA429295']\n",
      "['PRJNA429285']\n",
      "['PRJNA429286']\n",
      "['PRJNA429075']\n",
      "['PRJNA429077']\n",
      "['PRJNA429279']\n",
      "['PRJNA429284']\n",
      "['PRJNA429292']\n",
      "['PRJNA429294']\n",
      "['PRJNA494944']\n",
      "['PRJNA516019']\n",
      "['PRJNA496358']\n",
      "['PRJNA523108']\n",
      "['PRJNA520343']\n",
      "['PRJNA245598']\n",
      "['PRJNA320805']\n",
      "['PRJNA320813']\n",
      "['PRJNA320811']\n",
      "['PRJNA320815']\n",
      "['PRJNA320360']\n",
      "['PRJNA320816']\n",
      "['PRJNA540970']\n",
      "['PRJNA432521']\n",
      "['PRJNA510297']\n",
      "['PRJDB6310']\n",
      "['PRJNA559704']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA544527']\n",
      "['PRJNA486904']\n",
      "['PRJDB8988']\n",
      "['PRJNA434535']\n",
      "['PRJNA434536']\n",
      "['PRJNA434602']\n",
      "['PRJNA434603']\n",
      "['PRJNA434605']\n",
      "['PRJNA434606']\n",
      "['PRJNA434607']\n",
      "['PRJNA434609']\n",
      "['PRJNA434608']\n",
      "['PRJNA434610']\n",
      "['PRJNA434611']\n",
      "['PRJNA463453']\n",
      "['PRJNA473879']\n",
      "['PRJNA473880']\n",
      "['PRJNA473884']\n",
      "['PRJNA473885']\n",
      "['PRJNA473887']\n",
      "['PRJNA473889']\n",
      "['PRJNA473890']\n",
      "['PRJNA473891']\n",
      "['PRJNA473892']\n",
      "['PRJNA603250']\n",
      "['PRJNA607683']\n",
      "['PRJNA606647']\n",
      "['PRJNA626486']\n",
      "['PRJNA632309']\n",
      "['PRJNA649666']\n",
      "['PRJNA649666']\n",
      "['PRJDB10508']\n",
      "['PRJDB10508']\n",
      "['PRJNA664317']\n",
      "['PRJNA688125']\n",
      "['PRJNA688125']\n",
      "['PRJNA688125']\n",
      "['PRJNA688125']\n",
      "['PRJNA688125']\n",
      "['PRJNA688125']\n",
      "['PRJNA688125']\n",
      "['PRJNA680355']\n",
      "['PRJNA683735']\n",
      "['PRJNA683748']\n",
      "['PRJNA684964']\n",
      "['PRJDB7416']\n",
      "['PRJNA628672']\n",
      "['PRJDB7416']\n",
      "['PRJDB7416']\n",
      "['PRJDB7416']\n",
      "['PRJDB7416']\n",
      "['PRJEB16366']\n",
      "['PRJEB16461']\n",
      "['PRJEB18286']\n",
      "['PRJEB19594']\n",
      "['PRJEB21068']\n",
      "['PRJEB21068']\n",
      "['PRJEB21068']\n",
      "['PRJEB21068']\n",
      "['PRJEB32513']\n",
      "['PRJEB32513']\n",
      "['PRJEB34624']\n",
      "['PRJEB34624']\n",
      "['PRJEB34624']\n",
      "['PRJEB34624']\n",
      "['PRJEB34624']\n",
      "['PRJEB34624']\n",
      "['PRJEB34624']\n",
      "['PRJEB34624']\n",
      "['PRJNA288964']\n",
      "['PRJEB15121']\n",
      "['PRJEB21068']\n"
     ]
    }
   ],
   "source": [
    "for i, row in asm_biosam_bioproj.iterrows():\n",
    "    asm = row['assembly_accession']\n",
    "    bp = ! esearch -db assembly -query $asm | efetch -format docsum |xtract -pattern DocumentSummary -block GB_BioProjects -element BioprojectAccn\n",
    "    print(bp)\n",
    "    asm_biosam_bioproj.loc[i, 'GB_BioProjects'] = bp[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 672,
   "id": "3e17ba27",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>biosample</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>GB_BioProjects</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>352</th>\n",
       "      <td>GCF_000019665.1</td>\n",
       "      <td>SAMN02604183</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>PRJNA28995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>GCF_000019965.1</td>\n",
       "      <td>SAMN00000285</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>PRJNA19989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>GCF_000020225.1</td>\n",
       "      <td>SAMN00138213</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>PRJNA20089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>GCF_000025905.1</td>\n",
       "      <td>SAMN02598501</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>PRJNA33365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>GCF_000171235.2</td>\n",
       "      <td>SAMN02256391</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>PRJNA18333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>GCF_902728245.1</td>\n",
       "      <td>SAMEA6101986</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>PRJEB34624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>GCF_902728255.1</td>\n",
       "      <td>SAMEA6101990</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>PRJEB34624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>561</th>\n",
       "      <td>GCF_001683795.1</td>\n",
       "      <td>SAMN03840437</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>PRJNA288964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>562</th>\n",
       "      <td>GCF_900097105.1</td>\n",
       "      <td>SAMEA4378782</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>PRJEB15121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>563</th>\n",
       "      <td>GCF_900184965.1</td>\n",
       "      <td>SAMEA104102608</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>PRJEB21068</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>212 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession       biosample   bioproject GB_BioProjects\n",
       "352    GCF_000019665.1    SAMN02604183  PRJNA224116     PRJNA28995\n",
       "353    GCF_000019965.1    SAMN00000285  PRJNA224116     PRJNA19989\n",
       "354    GCF_000020225.1    SAMN00138213  PRJNA224116     PRJNA20089\n",
       "355    GCF_000025905.1    SAMN02598501  PRJNA224116     PRJNA33365\n",
       "356    GCF_000171235.2    SAMN02256391  PRJNA224116     PRJNA18333\n",
       "..                 ...             ...          ...            ...\n",
       "559    GCF_902728245.1    SAMEA6101986  PRJNA224116     PRJEB34624\n",
       "560    GCF_902728255.1    SAMEA6101990  PRJNA224116     PRJEB34624\n",
       "561    GCF_001683795.1    SAMN03840437  PRJNA224116    PRJNA288964\n",
       "562    GCF_900097105.1    SAMEA4378782  PRJNA224116     PRJEB15121\n",
       "563    GCF_900184965.1  SAMEA104102608  PRJNA224116     PRJEB21068\n",
       "\n",
       "[212 rows x 4 columns]"
      ]
     },
     "execution_count": 672,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "asm_biosam_bioproj[asm_biosam_bioproj['bioproject'] != asm_biosam_bioproj['GB_BioProjects']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 677,
   "id": "c73672d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "nan_type = merged_non_akk[merged_non_akk['type_of_sample'].isna()].loc[:, 'Genome'].str.extract('(GC[AF]_[0-9]{9}\\.[1-9]{1,2})').dropna().values.flatten()\n",
    "nan_host = merged_non_akk[merged_non_akk['host'].isna()].loc[:, 'Genome'].str.extract('(GC[AF]_[0-9]{9}\\.[1-9]{1,2})').dropna().values.flatten()\n",
    "len(set(nan_host) | set(nan_type))\n",
    "fh = open('bioproject_descr_miss_biosam_non_akk.tsv', 'a')\n",
    "asms = set(nan_type.tolist() + nan_host.tolist())\n",
    "for idx,row in asm_biosam_bioproj.set_index('assembly_accession').loc[asms, :].iterrows():\n",
    "    bp = row['GB_BioProjects']\n",
    "    desc = ! esearch -db bioproject -query $bp | efetch -format docsum |  xtract -pattern DocumentSummary -element Project_Name,Project_Title,Project_Description,Project_Target_Scope\n",
    "    desc_ls = desc.get_spstr().split('\\t')\n",
    "    desc_ls.insert(0, row['bioproject'])\n",
    "    desc_ls.insert(0, row['biosample'])\n",
    "    desc_ls.insert(0, idx)#assembly accession\n",
    "    print('\\t'.join(desc_ls), file=fh)\n",
    "fh.close()    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 680,
   "id": "1455445c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni 'metagenome|metagenomic|microbiome|biome|gut|bucal|respiratory|mud|forestsoil|sludge|water|marine|ocean|sediment|seawater|freshwater|freshwaters|feces|fecal|uncultured' bioproject_descr_miss_biosam_non_akk.tsv | sort -h | uniq\n",
      "'55:marine\\n93:marine\\n131:marine\\n'\n",
      "1\n",
      "index for the list in python: 130\n",
      "{'marine'} 1\n"
     ]
    }
   ],
   "source": [
    "regex = (\"'metagenome|metagenomic|microbiome|biome|gut|bucal|respiratory|mud|forest\"\n",
    "         \"soil|sludge|water|marine|ocean|sediment|seawater|freshwater|freshwaters|feces|fecal|uncultured'\")\n",
    "ln_meta_non_akk_bp = build_line_source('bioproject_descr_miss_biosam_non_akk.tsv', regex, file_pattern=False)\n",
    "print(repr(ln_meta_non_akk_bp))\n",
    "acc_meta_non_akk_bp = build_acc_host(ln_meta_non_akk_bp, 'bioproject_descr_miss_biosam_non_akk.tsv')\n",
    "print(set(acc_meta_non_akk_bp.values()), len(acc_meta_non_akk_bp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 681,
   "id": "0e8d895c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(None, {'GCA_000155695.1': 'marine'})"
      ]
     },
     "execution_count": 681,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "acc_meta_non_akk_bp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 682,
   "id": "33cc3b44",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rg -woni 'isolate|isolated|isolates|monoisolate|culture|cultured|cultivation|cultivated' bioproject_descr_miss_biosam_non_akk.tsv | sort -h | uniq\n",
      "'48:cultivated\\n48:isolates\\n49:cultivated\\n49:isolates\\n54:isolated\\n55:isolated\\n70:isolated\\n71:cultivated\\n71:isolates\\n79:Monoisolate\\n86:cultivated\\n86:isolates\\n87:cultivated\\n87:isolates\\n92:isolated\\n92:Mono'\n",
      "35\n",
      "index for the list in python: 151\n",
      "{'monoisolate', 'isolates'} 35\n"
     ]
    }
   ],
   "source": [
    "regex = (\"'isolate|isolated|isolates|monoisolate|culture|cultured|cultivation|cultivated'\")\n",
    "ln_culture_non_akk_bp = build_line_source('bioproject_descr_miss_biosam_non_akk.tsv', regex, file_pattern=False)\n",
    "print(repr(ln_culture_non_akk_bp[:200]))\n",
    "acc_culture_non_akk_bp = build_acc_host(ln_culture_non_akk_bp, 'bioproject_descr_miss_biosam_non_akk.tsv')\n",
    "print(set(acc_culture_non_akk_bp.values()), len(acc_culture_non_akk_bp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 691,
   "id": "18bc78cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "({'GCA_000155695.1',\n",
       "  'GCA_000242935.3',\n",
       "  'GCA_000243495.3',\n",
       "  'GCA_000526255.1',\n",
       "  'GCA_000739615.1',\n",
       "  'GCA_000739655.1',\n",
       "  'GCA_003054655.1',\n",
       "  'GCA_003054665.1',\n",
       "  'GCA_003054695.1',\n",
       "  'GCA_016811075.1',\n",
       "  'GCF_000019665.1',\n",
       "  'GCF_000019965.1',\n",
       "  'GCF_000172155.1',\n",
       "  'GCF_000172555.1',\n",
       "  'GCF_000173075.1',\n",
       "  'GCF_000297415.1',\n",
       "  'GCF_000379365.1',\n",
       "  'GCF_000817245.1',\n",
       "  'GCF_001613545.1',\n",
       "  'GCF_003054705.1',\n",
       "  'GCF_004341915.1',\n",
       "  'GCF_004364345.1',\n",
       "  'GCF_007475525.1',\n",
       "  'GCF_007992435.1',\n",
       "  'GCF_011044495.1',\n",
       "  'GCF_011044775.1',\n",
       "  'GCF_012913485.1',\n",
       "  'GCF_014201715.1',\n",
       "  'GCF_017310545.1',\n",
       "  'GCF_900104925.1',\n",
       "  'GCF_900105685.1',\n",
       "  'GCF_900141815.1',\n",
       "  'GCF_900167535.1',\n",
       "  'GCF_902143375.2',\n",
       "  'GCF_902143385.2'},\n",
       " 35,\n",
       " 35,\n",
       " 38)"
      ]
     },
     "execution_count": 691,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(acc_culture_non_akk_bp.keys()) & asms, len(set(acc_culture_non_akk_bp.keys()) & asms), len(acc_culture_non_akk_bp), len(asms)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eded4179",
   "metadata": {},
   "source": [
    "**So all the assemblies in the dictionary are missing from the table, not need to create a new filtered dictionary**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 694,
   "id": "3cbda999",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(metagenome    321\n",
       " culture        50\n",
       " Name: type_of_sample, dtype: int64,\n",
       " 13)"
      ]
     },
     "execution_count": 694,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_non_akk['type_of_sample'].value_counts(), merged_non_akk['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 695,
   "id": "16ca9b8b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metagenome     321\n",
      "monoisolate     31\n",
      "culture         28\n",
      "isolates         4\n",
      "Name: type_of_sample, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 695,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "add_info_to_summary(merged_non_akk,'type_of_sample', acc_culture_non_akk_bp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 696,
   "id": "cab146fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(metagenome    321\n",
       " culture        63\n",
       " Name: type_of_sample, dtype: int64,\n",
       " 0)"
      ]
     },
     "execution_count": 696,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_to_sp  = {'isolate':'culture', 'isolated':'culture', 'isolates':'culture', 'monoisolate':'culture'}\n",
    "merged_non_akk['type_of_sample'].replace(common_to_sp, inplace=True)\n",
    "merged_non_akk['type_of_sample'].value_counts(), merged_non_akk['type_of_sample'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 697,
   "id": "9c28cd6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_non_akk.to_csv(qc_path.joinpath('merged_non_akk.tsv'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58f0ea5d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
